// permitio/build/module/elements/client.js
import axios from 'axios';
import { AuthenticationApi, Configuration } from '../openapi';
export var ElementsApiErrors;
(function (ElementsApiErrors) {
    ElementsApiErrors["USER_NOT_FOUND"] = "User not found";
    ElementsApiErrors["TENANT_NOT_FOUND"] = "Tenant not found";
    ElementsApiErrors["INVALID_PERMISSION_LEVEL"] = "Invalid user permission level";
    ElementsApiErrors["FORBIDDEN_ACCESS"] = "Forbidden access";
})(ElementsApiErrors || (ElementsApiErrors = {}));
export class ElementsClient {
    config;
    logger;
    authApi;
    constructor(config, logger) {
        this.config = config;
        this.logger = logger;
        const axiosClientConfig = new Configuration({
            basePath: `${this.config.apiUrl}`,
            accessToken: this.config.token,
        });
        this.authApi = new AuthenticationApi(axiosClientConfig);
    }
    async loginAs({ userId, tenantId, }) {
        try {
            const response = await this.authApi.elementsLoginAs({
                userLoginRequestInput: {
                    user_id: userId,
                    tenant_id: tenantId,
                },
            });
            this.logger.debug(`[${response.status}] permit.api.loginAs(${userId})`);
            const res = response.data;
            res['content'] = { url: response.data.redirect_url };
            return res;
        }
        catch (err) {
            if (axios.isAxiosError(err)) {
                this.logger.error(`[${err?.response?.status}] permit.api.loginAs(${userId}), err: ${JSON.stringify(err?.response?.data)}`);
            }
            throw err;
        }
    }
    get elements() {
        return {
            loginAs: this.loginAs.bind(this),
        };
    }
    getMethods() {
        return {
            elements: this.elements,
        };
    }
}
