// permitio/build/main/cache/client.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.LocalCacheClient = void 0;
const axios_1 = __importDefault(require("axios"));
const HTTP_404_NOT_FOUND = 404;
/**
 * The LocalCacheClient is able to fetch the latest cached (i.e: synced)
 * state from the policy agent (i.e: OPA). This client is very performant
 * and DOES NOT go outside the local network (i.e: VPC), in other words,
 * queries made by this client are complete private, and do not reach
 * the permit.io control plane in the cloud.
 */
class LocalCacheClient {
    constructor(config, logger) {
        this.config = config;
        this.logger = logger;
        this.client = axios_1.default.create({
            baseURL: `${this.config.pdp}/`,
            headers: {
                Authorization: `Bearer ${this.config.token}`,
                'Content-Type': 'application/json',
            },
        });
    }
    async isUser(userId) {
        const user = await this.getUser(userId);
        if (user === null) {
            return false;
        }
        return user.id === userId;
    }
    // cached object api
    async getUser(userId) {
        return await this.client
            .get(`local/users/${userId}`)
            .then((response) => {
            return response.data;
        })
            .catch((error) => {
            if (error.response) {
                if (error.response.status !== HTTP_404_NOT_FOUND) {
                    this.logger.error(`unexpected error when calling permit.cache.getUser(${userId}): ${error}`);
                }
            }
            return null;
        });
    }
    async getUsers() {
        return await this.client
            .get(`local/users`)
            .then((response) => {
            return response.data;
        })
            .catch((error) => {
            if (error.response) {
                if (error.response.status !== HTTP_404_NOT_FOUND) {
                    this.logger.error(`unexpected error when calling permit.cache.getUsers(): ${error}`);
                }
            }
            return [];
        });
    }
    async getUserTenants(userId) {
        return await this.client
            .get(`local/users/${userId}/tenants`)
            .then((response) => {
            return response.data;
        })
            .catch((error) => {
            if (error.response) {
                if (error.response.status !== HTTP_404_NOT_FOUND) {
                    this.logger.error(`unexpected error when calling permit.cache.getUserTenants(${userId}): ${error}`);
                }
            }
            return null; // indicate user is not synced
        });
    }
    async getAssignedRoles(userId) {
        return await this.client
            .get(`local/users/${userId}/roles`)
            .then((response) => {
            return response.data;
        })
            .catch((error) => {
            if (error.response) {
                if (error.response.status !== HTTP_404_NOT_FOUND) {
                    this.logger.error(`unexpected error when calling permit.cache.getAssignedRoles(${userId}): ${error}`);
                }
            }
            return null; // indicate user is not synced
        });
    }
    async getRoles() {
        return await this.client
            .get(`local/roles`)
            .then((response) => {
            return response.data;
        })
            .catch((error) => {
            if (error.response) {
                if (error.response.status !== HTTP_404_NOT_FOUND) {
                    this.logger.error(`unexpected error when calling permit.cache.getRoles(): ${error}`);
                }
            }
            return [];
        });
    }
    async getRoleById(roleId) {
        return await this.client
            .get(`local/roles/${roleId}`)
            .then((response) => {
            return response.data;
        })
            .catch((error) => {
            if (error.response) {
                if (error.response.status !== HTTP_404_NOT_FOUND) {
                    this.logger.error(`unexpected error when calling permit.cache.getRoleById(${roleId}): ${error}`);
                }
            }
            return null;
        });
    }
    async getRoleByName(roleName) {
        return await this.client
            .get(`local/roles/by-name/${roleName}`)
            .then((response) => {
            return response.data;
        })
            .catch((error) => {
            if (error.response) {
                if (error.response.status !== HTTP_404_NOT_FOUND) {
                    this.logger.error(`unexpected error when calling permit.cache.getRoleByName(${roleName}): ${error}`);
                }
            }
            return null;
        });
    }
    async updatePolicy() {
        return this.client
            .post('policy-updater/trigger')
            .then((response) => {
            return response.status == 200;
        })
            .catch((error) => {
            this.logger.error(`tried to trigger policy update, got error: ${error}`);
            return false;
        });
    }
    async updatePolicyData() {
        return this.client
            .post('data-updater/trigger')
            .then((response) => {
            return response.status == 200;
        })
            .catch((error) => {
            this.logger.error(`tried to trigger policy update, got error: ${error}`);
            return false;
        });
    }
    async refresh() {
        return this.updatePolicy().then((triggered) => {
            if (!triggered)
                return false;
            return this.updatePolicyData().then((triggered) => {
                return triggered;
            });
        });
    }
    getMethods() {
        return {
            isUser: this.isUser.bind(this),
            getUser: this.getUser.bind(this),
            getUsers: this.getUsers.bind(this),
            getUserTenants: this.getUserTenants.bind(this),
            getAssignedRoles: this.getAssignedRoles.bind(this),
            getRoles: this.getRoles.bind(this),
            getRoleById: this.getRoleById.bind(this),
            getRoleByName: this.getRoleByName.bind(this),
            refresh: this.refresh.bind(this),
        };
    }
}
exports.LocalCacheClient = LocalCacheClient;
