// permitio/build/main/api/api-client.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ApiClient = void 0;
const condition_set_rules_1 = require("./condition-set-rules");
const condition_sets_1 = require("./condition-sets");
const deprecated_1 = require("./deprecated");
const environments_1 = require("./environments");
const projects_1 = require("./projects");
const relationship_tuples_1 = require("./relationship-tuples");
const resource_action_groups_1 = require("./resource-action-groups");
const resource_actions_1 = require("./resource-actions");
const resource_attributes_1 = require("./resource-attributes");
const resource_instances_1 = require("./resource-instances");
const resource_relations_1 = require("./resource-relations");
const resource_roles_1 = require("./resource-roles");
const resources_1 = require("./resources");
const role_assignments_1 = require("./role-assignments");
const roles_1 = require("./roles");
const tenants_1 = require("./tenants");
const users_1 = require("./users");
class ApiClient extends deprecated_1.DeprecatedApiClient {
    /**
     * Constructs a new instance of the ApiClient class with the specified configuration and logger.
     *
     * @param config - The configuration for the Permit API client.
     * @param logger - The logger instance.
     */
    constructor(config, logger) {
        super(config, logger);
        this.conditionSetRules = new condition_set_rules_1.ConditionSetRulesApi(config, logger);
        this.conditionSets = new condition_sets_1.ConditionSetsApi(config, logger);
        this.projects = new projects_1.ProjectsApi(config, logger);
        this.environments = new environments_1.EnvironmentsApi(config, logger);
        this.actionGroups = new resource_action_groups_1.ResourceActionGroupsApi(config, logger);
        this.resourceActions = new resource_actions_1.ResourceActionsApi(config, logger);
        this.resourceAttributes = new resource_attributes_1.ResourceAttributesApi(config, logger);
        this.resourceRoles = new resource_roles_1.ResourceRolesApi(config, logger);
        this.resourceRelations = new resource_relations_1.ResourceRelationsApi(config, logger);
        this.resourceInstances = new resource_instances_1.ResourceInstancesApi(config, logger);
        this.resources = new resources_1.ResourcesApi(config, logger);
        this.roleAssignments = new role_assignments_1.RoleAssignmentsApi(config, logger);
        this.roles = new roles_1.RolesApi(config, logger);
        this.relationshipTuples = new relationship_tuples_1.RelationshipTuplesApi(config, logger);
        this.tenants = new tenants_1.TenantsApi(config, logger);
        this.users = new users_1.UsersApi(config, logger);
    }
}
exports.ApiClient = ApiClient;
