// permitio/build/main/api/condition-sets.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ConditionSetsApi = void 0;
const openapi_1 = require("../openapi");
const base_1 = require("../openapi/base");
const base_2 = require("./base"); // eslint-disable-line @typescript-eslint/no-unused-vars
const context_1 = require("./context"); // eslint-disable-line @typescript-eslint/no-unused-vars
/**
 * The ConditionSetsApi class provides methods for interacting with condition sets using the Permit REST API.
 */
class ConditionSetsApi extends base_2.BasePermitApi {
    /**
     * Creates an instance of the ConditionSetsApi.
     * @param config - The configuration object for the Permit SDK.
     * @param logger - The logger instance for logging.
     */
    constructor(config, logger) {
        super(config, logger);
        this.conditionSets = new openapi_1.ConditionSetsApi(this.openapiClientConfig, base_1.BASE_PATH, this.config.axiosInstance);
    }
    /**
     * Retrieves a list of condition sets.
     *
     * @param pagination The pagination options, @see {@link IPagination}
     * @returns A promise that resolves to an array of condition sets.
     * @throws {@link PermitApiError} If the API returns an error HTTP status code.
     * @throws {@link PermitContextError} If the configured {@link ApiContext} does not match the required endpoint context.
     */
    async list(pagination) {
        const { page = 1, perPage = 100 } = pagination !== null && pagination !== void 0 ? pagination : {};
        await this.ensureAccessLevel(context_1.ApiKeyLevel.ENVIRONMENT_LEVEL_API_KEY);
        await this.ensureContext(context_1.ApiContextLevel.ENVIRONMENT);
        try {
            return (await this.conditionSets.listConditionSets(Object.assign(Object.assign({}, this.config.apiContext.environmentContext), { page,
                perPage }))).data;
        }
        catch (err) {
            this.handleApiError(err);
        }
    }
    /**
     * Retrieves a condition set by its key.
     *
     * @param conditionSetKey The key of the condition set.
     * @returns A promise that resolves to the condition set.
     * @throws {@link PermitApiError} If the API returns an error HTTP status code.
     * @throws {@link PermitContextError} If the configured {@link ApiContext} does not match the required endpoint context.
     */
    async get(conditionSetKey) {
        await this.ensureAccessLevel(context_1.ApiKeyLevel.ENVIRONMENT_LEVEL_API_KEY);
        await this.ensureContext(context_1.ApiContextLevel.ENVIRONMENT);
        try {
            return (await this.conditionSets.getConditionSet(Object.assign(Object.assign({}, this.config.apiContext.environmentContext), { conditionSetId: conditionSetKey }))).data;
        }
        catch (err) {
            this.handleApiError(err);
        }
    }
    /**
     * Retrieves a condition set by its key.
     * Alias for the {@link get} method.
     *
     * @param conditionSetKey The key of the condition set.
     * @returns A promise that resolves to the condition set.
     * @throws {@link PermitApiError} If the API returns an error HTTP status code.
     * @throws {@link PermitContextError} If the configured {@link ApiContext} does not match the required endpoint context.
     */
    async getByKey(conditionSetKey) {
        return await this.get(conditionSetKey);
    }
    /**
     * Retrieves a condition set by its ID.
     * Alias for the {@link get} method.
     *
     * @param conditionSetId The ID of the condition set.
     * @returns A promise that resolves to the condition set.
     * @throws {@link PermitApiError} If the API returns an error HTTP status code.
     * @throws {@link PermitContextError} If the configured {@link ApiContext} does not match the required endpoint context.
     */
    async getById(conditionSetId) {
        return await this.get(conditionSetId);
    }
    /**
     * Creates a new condition set.
     *
     * @param conditionSetData The data for the new condition set.
     * @returns A promise that resolves to the created condition set.
     * @throws {@link PermitApiError} If the API returns an error HTTP status code.
     * @throws {@link PermitContextError} If the configured {@link ApiContext} does not match the required endpoint context.
     */
    async create(conditionSetData) {
        await this.ensureAccessLevel(context_1.ApiKeyLevel.ENVIRONMENT_LEVEL_API_KEY);
        await this.ensureContext(context_1.ApiContextLevel.ENVIRONMENT);
        try {
            return (await this.conditionSets.createConditionSet(Object.assign(Object.assign({}, this.config.apiContext.environmentContext), { conditionSetCreate: conditionSetData }))).data;
        }
        catch (err) {
            this.handleApiError(err);
        }
    }
    /**
     * Updates a condition set.
     *
     * @param conditionSetKey The key of the condition set.
     * @param conditionSetData The updated data for the condition set.
     * @returns A promise that resolves to the updated condition set.
     * @throws {@link PermitApiError} If the API returns an error HTTP status code.
     * @throws {@link PermitContextError} If the configured {@link ApiContext} does not match the required endpoint context.
     */
    async update(conditionSetKey, conditionSetData) {
        await this.ensureAccessLevel(context_1.ApiKeyLevel.ENVIRONMENT_LEVEL_API_KEY);
        await this.ensureContext(context_1.ApiContextLevel.ENVIRONMENT);
        try {
            return (await this.conditionSets.updateConditionSet(Object.assign(Object.assign({}, this.config.apiContext.environmentContext), { conditionSetId: conditionSetKey, conditionSetUpdate: conditionSetData }))).data;
        }
        catch (err) {
            this.handleApiError(err);
        }
    }
    /**
     * Deletes a condition set.
     *
     * @param conditionSetKey The key of the condition set to delete.
     * @returns A promise that resolves when the condition set is deleted.
     * @throws {@link PermitApiError} If the API returns an error HTTP status code.
     * @throws {@link PermitContextError} If the configured {@link ApiContext} does not match the required endpoint context.
     */
    async delete(conditionSetKey) {
        await this.ensureAccessLevel(context_1.ApiKeyLevel.ENVIRONMENT_LEVEL_API_KEY);
        await this.ensureContext(context_1.ApiContextLevel.ENVIRONMENT);
        try {
            await this.conditionSets.deleteConditionSet(Object.assign(Object.assign({}, this.config.apiContext.environmentContext), { conditionSetId: conditionSetKey }));
        }
        catch (err) {
            this.handleApiError(err);
        }
    }
}
exports.ConditionSetsApi = ConditionSetsApi;
