// permitio/build/main/api/index.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.AttributeType = exports.EnvironmentCopyConflictStrategyEnum = exports.ConditionSetType = exports.APIKeyOwnerType = exports.MemberAccessObj = exports.MemberAccessLevel = void 0;
__exportStar(require("./condition-set-rules"), exports);
__exportStar(require("./condition-sets"), exports);
__exportStar(require("./deprecated"), exports);
__exportStar(require("./environments"), exports);
__exportStar(require("./projects"), exports);
__exportStar(require("./resource-action-groups"), exports);
__exportStar(require("./resource-actions"), exports);
__exportStar(require("./resource-attributes"), exports);
__exportStar(require("./resources"), exports);
__exportStar(require("./role-assignments"), exports);
__exportStar(require("./roles"), exports);
__exportStar(require("./tenants"), exports);
__exportStar(require("./users"), exports);
__exportStar(require("./api-client"), exports);
__exportStar(require("./elements"), exports);
// referenced by other exports
var types_1 = require("../openapi/types");
Object.defineProperty(exports, "MemberAccessLevel", { enumerable: true, get: function () { return types_1.MemberAccessLevel; } });
Object.defineProperty(exports, "MemberAccessObj", { enumerable: true, get: function () { return types_1.MemberAccessObj; } });
Object.defineProperty(exports, "APIKeyOwnerType", { enumerable: true, get: function () { return types_1.APIKeyOwnerType; } });
Object.defineProperty(exports, "ConditionSetType", { enumerable: true, get: function () { return types_1.ConditionSetType; } });
Object.defineProperty(exports, "EnvironmentCopyConflictStrategyEnum", { enumerable: true, get: function () { return types_1.EnvironmentCopyConflictStrategyEnum; } });
Object.defineProperty(exports, "AttributeType", { enumerable: true, get: function () { return types_1.AttributeType; } });
