// permitio/build/main/api/resources.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ResourcesApi = void 0;
const openapi_1 = require("../openapi");
const base_1 = require("../openapi/base");
const base_2 = require("./base");
const context_1 = require("./context");
/**
 * The ResourcesApi class provides methods for interacting with Permit Resources.
 */
class ResourcesApi extends base_2.BasePermitApi {
    /**
     * Creates an instance of the ResourcesApi.
     * @param config - The configuration object for the Permit SDK.
     * @param logger - The logger instance for logging.
     */
    constructor(config, logger) {
        super(config, logger);
        this.resources = new openapi_1.ResourcesApi(this.openapiClientConfig, base_1.BASE_PATH, this.config.axiosInstance);
    }
    /**
     * Retrieves a list of resources.
     *
     * @param pagination The pagination options, @see {@link IPagination}
     * @returns A promise that resolves to an array of resources.
     * @throws {@link PermitApiError} If the API returns an error HTTP status code.
     * @throws {@link PermitContextError} If the configured {@link ApiContext} does not match the required endpoint context.
     */
    async list(pagination) {
        const { page = 1, perPage = 100 } = pagination !== null && pagination !== void 0 ? pagination : {};
        await this.ensureAccessLevel(context_1.ApiKeyLevel.ENVIRONMENT_LEVEL_API_KEY);
        await this.ensureContext(context_1.ApiContextLevel.ENVIRONMENT);
        try {
            return (await this.resources.listResources(Object.assign(Object.assign({}, this.config.apiContext.environmentContext), { page,
                perPage }))).data;
        }
        catch (err) {
            this.handleApiError(err);
        }
    }
    /**
     * Retrieves a resource by its key.
     *
     * @param resourceKey The key of the resource.
     * @returns A promise that resolves to the resource.
     * @throws {@link PermitApiError} If the API returns an error HTTP status code.
     * @throws {@link PermitContextError} If the configured {@link ApiContext} does not match the required endpoint context.
     */
    async get(resourceKey) {
        await this.ensureAccessLevel(context_1.ApiKeyLevel.ENVIRONMENT_LEVEL_API_KEY);
        await this.ensureContext(context_1.ApiContextLevel.ENVIRONMENT);
        try {
            return (await this.resources.getResource(Object.assign(Object.assign({}, this.config.apiContext.environmentContext), { resourceId: resourceKey }))).data;
        }
        catch (err) {
            this.handleApiError(err);
        }
    }
    /**
     * Retrieves a resource by its key.
     * Alias for the {@link get} method.
     *
     * @param resourceKey The key of the resource.
     * @returns A promise that resolves to the resource.
     * @throws {@link PermitApiError} If the API returns an error HTTP status code.
     * @throws {@link PermitContextError} If the configured {@link ApiContext} does not match the required endpoint context.
     */
    async getByKey(resourceKey) {
        return await this.get(resourceKey);
    }
    /**
     * Retrieves a resource by its ID.
     * Alias for the {@link get} method.
     *
     * @param resourceId The ID of the resource.
     * @returns A promise that resolves to the resource.
     * @throws {@link PermitApiError} If the API returns an error HTTP status code.
     * @throws {@link PermitContextError} If the configured {@link ApiContext} does not match the required endpoint context.
     */
    async getById(resourceId) {
        return await this.get(resourceId);
    }
    /**
     * Creates a new resource.
     *
     * @param resourceData The data for the new resource.
     * @returns A promise that resolves to the created resource.
     * @throws {@link PermitApiError} If the API returns an error HTTP status code.
     * @throws {@link PermitContextError} If the configured {@link ApiContext} does not match the required endpoint context.
     */
    async create(resourceData) {
        await this.ensureAccessLevel(context_1.ApiKeyLevel.ENVIRONMENT_LEVEL_API_KEY);
        await this.ensureContext(context_1.ApiContextLevel.ENVIRONMENT);
        try {
            return (await this.resources.createResource(Object.assign(Object.assign({}, this.config.apiContext.environmentContext), { resourceCreate: resourceData }))).data;
        }
        catch (err) {
            this.handleApiError(err);
        }
    }
    /**
     * Creates a resource if such a resource does not exists, otherwise completely replaces the resource configuration.
     *
     * @param resourceKey The key of the resource.
     * @param resourceData The updated data for the resource.
     * @returns A promise that resolves to the updated resource.
     * @throws {@link PermitApiError} If the API returns an error HTTP status code.
     * @throws {@link PermitContextError} If the configured {@link ApiContext} does not match the required endpoint context.
     */
    async replace(resourceKey, resourceData) {
        await this.ensureAccessLevel(context_1.ApiKeyLevel.ENVIRONMENT_LEVEL_API_KEY);
        await this.ensureContext(context_1.ApiContextLevel.ENVIRONMENT);
        try {
            return (await this.resources.replaceResource(Object.assign(Object.assign({}, this.config.apiContext.environmentContext), { resourceId: resourceKey, resourceReplace: resourceData }))).data;
        }
        catch (err) {
            this.handleApiError(err);
        }
    }
    /**
     * Updates a resource.
     *
     * @param resourceKey The key of the resource.
     * @param resourceData The updated data for the resource.
     * @returns A promise that resolves to the updated resource.
     * @throws {@link PermitApiError} If the API returns an error HTTP status code.
     * @throws {@link PermitContextError} If the configured {@link ApiContext} does not match the required endpoint context.
     */
    async update(resourceKey, resourceData) {
        await this.ensureAccessLevel(context_1.ApiKeyLevel.ENVIRONMENT_LEVEL_API_KEY);
        await this.ensureContext(context_1.ApiContextLevel.ENVIRONMENT);
        try {
            return (await this.resources.updateResource(Object.assign(Object.assign({}, this.config.apiContext.environmentContext), { resourceId: resourceKey, resourceUpdate: resourceData }))).data;
        }
        catch (err) {
            this.handleApiError(err);
        }
    }
    /**
     * Deletes a resource.
     *
     * @param resourceKey The key of the resource to delete.
     * @returns A promise that resolves when the resource is deleted.
     * @throws {@link PermitApiError} If the API returns an error HTTP status code.
     * @throws {@link PermitContextError} If the configured {@link ApiContext} does not match the required endpoint context.
     */
    async delete(resourceKey) {
        await this.ensureAccessLevel(context_1.ApiKeyLevel.ENVIRONMENT_LEVEL_API_KEY);
        await this.ensureContext(context_1.ApiContextLevel.ENVIRONMENT);
        try {
            await this.resources.deleteResource(Object.assign(Object.assign({}, this.config.apiContext.environmentContext), { resourceId: resourceKey }));
        }
        catch (err) {
            this.handleApiError(err);
        }
    }
}
exports.ResourcesApi = ResourcesApi;
