// permitio/build/main/api/resource-instances.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ResourceInstancesApi = void 0;
const openapi_1 = require("../openapi");
const base_1 = require("../openapi/base");
const base_2 = require("./base");
const context_1 = require("./context");
/**
 * The ResourceInstancesApi class provides methods for interacting with Permit ResourceInstances.
 */
class ResourceInstancesApi extends base_2.BasePermitApi {
    /**
     * Creates an instance of the ResourceInstancesApi.
     * @param config - The configuration object for the Permit SDK.
     * @param logger - The logger instance for logging.
     */
    constructor(config, logger) {
        super(config, logger);
        this.instances = new openapi_1.ResourceInstancesApi(this.openapiClientConfig, base_1.BASE_PATH, this.config.axiosInstance);
    }
    /**
     * Retrieves a list of resource instances.
     *
     * @param pagination The pagination options, @see {@link IPagination}
     * @returns A promise that resolves to an array of resource instances.
     * @throws {@link PermitApiError} If the API returns an error HTTP status code.
     * @throws {@link PermitContextError} If the configured {@link ApiContext} does not match the required endpoint context.
     */
    async list(pagination) {
        const { page = 1, perPage = 100 } = pagination !== null && pagination !== void 0 ? pagination : {};
        await this.ensureAccessLevel(context_1.ApiKeyLevel.ENVIRONMENT_LEVEL_API_KEY);
        await this.ensureContext(context_1.ApiContextLevel.ENVIRONMENT);
        try {
            return (await this.instances.listResourceInstances(Object.assign(Object.assign({}, this.config.apiContext.environmentContext), { page,
                perPage }))).data;
        }
        catch (err) {
            this.handleApiError(err);
        }
    }
    /**
     * Retrieves a instance by its key.
     *
     * @param instanceKey The key of the resource instance.
     * @returns A promise that resolves to the resource instance.
     * @throws {@link PermitApiError} If the API returns an error HTTP status code.
     * @throws {@link PermitContextError} If the configured {@link ApiContext} does not match the required endpoint context.
     */
    async get(instanceKey) {
        await this.ensureAccessLevel(context_1.ApiKeyLevel.ENVIRONMENT_LEVEL_API_KEY);
        await this.ensureContext(context_1.ApiContextLevel.ENVIRONMENT);
        try {
            return (await this.instances.getResourceInstance(Object.assign(Object.assign({}, this.config.apiContext.environmentContext), { instanceId: instanceKey }))).data;
        }
        catch (err) {
            this.handleApiError(err);
        }
    }
    /**
     * Retrieves a instance by its key.
     * Alias for the {@link get} method.
     *
     * @param instanceKey The key of the resource instance.
     * @returns A promise that resolves to the resource instance.
     * @throws {@link PermitApiError} If the API returns an error HTTP status code.
     * @throws {@link PermitContextError} If the configured {@link ApiContext} does not match the required endpoint context.
     */
    async getByKey(instanceKey) {
        return await this.get(instanceKey);
    }
    /**
     * Retrieves a instance by its ID.
     * Alias for the {@link get} method.
     *
     * @param instanceId The ID of the resource instance.
     * @returns A promise that resolves to the resource instance.
     * @throws {@link PermitApiError} If the API returns an error HTTP status code.
     * @throws {@link PermitContextError} If the configured {@link ApiContext} does not match the required endpoint context.
     */
    async getById(instanceId) {
        return await this.get(instanceId);
    }
    /**
     * Creates a new instance.
     *
     * @param instanceData The data for the new instance.
     * @returns A promise that resolves to the created instance.
     * @throws {@link PermitApiError} If the API returns an error HTTP status code.
     * @throws {@link PermitContextError} If the configured {@link ApiContext} does not match the required endpoint context.
     */
    async create(instanceData) {
        await this.ensureAccessLevel(context_1.ApiKeyLevel.ENVIRONMENT_LEVEL_API_KEY);
        await this.ensureContext(context_1.ApiContextLevel.ENVIRONMENT);
        try {
            return (await this.instances.createResourceInstance(Object.assign(Object.assign({}, this.config.apiContext.environmentContext), { resourceInstanceCreate: instanceData }))).data;
        }
        catch (err) {
            this.handleApiError(err);
        }
    }
    /**
     * Updates a instance.
     *
     * @param instanceKey The key of the resource instance.
     * @param instanceData The updated data for the resource instance.
     * @returns A promise that resolves to the updated instance.
     * @throws {@link PermitApiError} If the API returns an error HTTP status code.
     * @throws {@link PermitContextError} If the configured {@link ApiContext} does not match the required endpoint context.
     */
    async update(instanceKey, instanceData) {
        await this.ensureAccessLevel(context_1.ApiKeyLevel.ENVIRONMENT_LEVEL_API_KEY);
        await this.ensureContext(context_1.ApiContextLevel.ENVIRONMENT);
        try {
            return (await this.instances.updateResourceInstance(Object.assign(Object.assign({}, this.config.apiContext.environmentContext), { instanceId: instanceKey, resourceInstanceUpdate: instanceData }))).data;
        }
        catch (err) {
            this.handleApiError(err);
        }
    }
    /**
     * Deletes a instance.
     *
     * @param instanceKey The key of the resource instance to delete.
     * @returns A promise that resolves when the resource instance is deleted.
     * @throws {@link PermitApiError} If the API returns an error HTTP status code.
     * @throws {@link PermitContextError} If the configured {@link ApiContext} does not match the required endpoint context.
     */
    async delete(instanceKey) {
        await this.ensureAccessLevel(context_1.ApiKeyLevel.ENVIRONMENT_LEVEL_API_KEY);
        await this.ensureContext(context_1.ApiContextLevel.ENVIRONMENT);
        try {
            await this.instances.deleteResourceInstance(Object.assign(Object.assign({}, this.config.apiContext.environmentContext), { instanceId: instanceKey }));
        }
        catch (err) {
            this.handleApiError(err);
        }
    }
}
exports.ResourceInstancesApi = ResourceInstancesApi;
