// permitio/build/main/api/resource-attributes.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ResourceAttributesApi = void 0;
const openapi_1 = require("../openapi");
const base_1 = require("../openapi/base");
const base_2 = require("./base"); // eslint-disable-line @typescript-eslint/no-unused-vars
const context_1 = require("./context"); // eslint-disable-line @typescript-eslint/no-unused-vars
/**
 * API client for interacting with the Resource Attributes API.
 */
class ResourceAttributesApi extends base_2.BasePermitApi {
    /**
     * Creates an instance of the ResourceAttributesApi.
     * @param config - The configuration object for the Permit SDK.
     * @param logger - The logger instance for logging.
     */
    constructor(config, logger) {
        super(config, logger);
        this.attributesApi = new openapi_1.ResourceAttributesApi(this.openapiClientConfig, base_1.BASE_PATH, this.config.axiosInstance);
    }
    /**
     * Retrieves a list of all attributes that are defined for a given resource.
     * @param params - pagination and filtering params, @see {@link IListAttributes}
     * @returns A promise that resolves to an array of ResourceAttributeRead objects representing the attributes.
     * @throws {@link PermitApiError} If the API returns an error HTTP status code.
     * @throws {@link PermitContextError} If the configured {@link ApiContext} does not match the required endpoint context.
     */
    async list(params) {
        const { resourceKey, page = 1, perPage = 100 } = params;
        await this.ensureAccessLevel(context_1.ApiKeyLevel.ENVIRONMENT_LEVEL_API_KEY);
        await this.ensureContext(context_1.ApiContextLevel.ENVIRONMENT);
        try {
            return (await this.attributesApi.listResourceAttributes(Object.assign(Object.assign({}, this.config.apiContext.environmentContext), { resourceId: resourceKey, page,
                perPage }))).data;
        }
        catch (err) {
            this.handleApiError(err);
        }
    }
    /**
     * Retrieves an attribute based on the resource key and the attribute key.
     *
     * @param resourceKey - The resource key.
     * @param attributeKey - The attribute key.
     * @returns A promise that resolves to a ResourceAttributeRead object representing the attribute.
     * @throws {@link PermitApiError} If the API returns an error HTTP status code.
     * @throws {@link PermitContextError} If the configured {@link ApiContext} does not match the required endpoint context.
     */
    async get(resourceKey, attributeKey) {
        await this.ensureAccessLevel(context_1.ApiKeyLevel.ENVIRONMENT_LEVEL_API_KEY);
        await this.ensureContext(context_1.ApiContextLevel.ENVIRONMENT);
        try {
            return (await this.attributesApi.getResourceAttribute(Object.assign(Object.assign({}, this.config.apiContext.environmentContext), { resourceId: resourceKey, attributeId: attributeKey }))).data;
        }
        catch (err) {
            this.handleApiError(err);
        }
    }
    /**
     * Retrieves an attribute based on the resource key and the attribute key.
     * Alias for the {@link get} method.
     *
     * @param resourceKey - The resource key.
     * @param attributeKey - The attribute key.
     * @returns A promise that resolves to a ResourceAttributeRead object representing the attribute.
     * @throws {@link PermitApiError} If the API returns an error HTTP status code.
     * @throws {@link PermitContextError} If the configured {@link ApiContext} does not match the required endpoint context.
     */
    async getByKey(resourceKey, attributeKey) {
        return await this.get(resourceKey, attributeKey);
    }
    /**
     * Retrieves an attribute based on the resource ID and the attribute ID.
     * Alias for the {@link get} method.
     *
     * @param resourceId - The resource ID.
     * @param attributeId - The attribute ID.
     * @returns A promise that resolves to a ResourceAttributeRead object representing the attribute.
     * @throws {@link PermitApiError} If the API returns an error HTTP status code.
     * @throws {@link PermitContextError} If the configured {@link ApiContext} does not match the required endpoint context.
     */
    async getById(resourceId, attributeId) {
        return await this.get(resourceId, attributeId);
    }
    /**
     * Creates a new attribute.
     *
     * @param resourceKey - The resource key.
     * @param attributeData - The attribute data.
     * @returns A promise that resolves to a ResourceAttributeRead object representing the created attribute.
     * @throws {@link PermitApiError} If the API returns an error HTTP status code.
     * @throws {@link PermitContextError} If the configured {@link ApiContext} does not match the required endpoint context.
     */
    async create(resourceKey, attributeData) {
        await this.ensureAccessLevel(context_1.ApiKeyLevel.ENVIRONMENT_LEVEL_API_KEY);
        await this.ensureContext(context_1.ApiContextLevel.ENVIRONMENT);
        try {
            return (await this.attributesApi.createResourceAttribute(Object.assign(Object.assign({}, this.config.apiContext.environmentContext), { resourceId: resourceKey, resourceAttributeCreate: attributeData }))).data;
        }
        catch (err) {
            this.handleApiError(err);
        }
    }
    /**
     * Updates an existing attribute.
     *
     * @param resourceKey - The resource key.
     * @param attributeKey - The key of the attribute to modify.
     * @param attributeData - The data for updating the attribute.
     * @returns A promise that resolves to a ResourceAttributeRead object representing the updated attribute.
     * @throws {@link PermitApiError} If the API returns an error HTTP status code.
     * @throws {@link PermitContextError} If the configured {@link ApiContext} does not match the required endpoint context.
     */
    async update(resourceKey, attributeKey, attributeData) {
        await this.ensureAccessLevel(context_1.ApiKeyLevel.ENVIRONMENT_LEVEL_API_KEY);
        await this.ensureContext(context_1.ApiContextLevel.ENVIRONMENT);
        try {
            return (await this.attributesApi.updateResourceAttribute(Object.assign(Object.assign({}, this.config.apiContext.environmentContext), { resourceId: resourceKey, attributeId: attributeKey, resourceAttributeUpdate: attributeData }))).data;
        }
        catch (err) {
            this.handleApiError(err);
        }
    }
    /**
     * Deletes a attribute based on the resource key and attribute key.
     * @param resourceKey - The resource key.
     * @param attributeKey - The attribute key.
     * @returns A promise that resolves when the attribute is successfully deleted.
     * @throws {@link PermitApiError} If the API returns an error HTTP status code.
     * @throws {@link PermitContextError} If the configured {@link ApiContext} does not match the required endpoint context.
     */
    async delete(resourceKey, attributeKey) {
        await this.ensureAccessLevel(context_1.ApiKeyLevel.ENVIRONMENT_LEVEL_API_KEY);
        await this.ensureContext(context_1.ApiContextLevel.ENVIRONMENT);
        try {
            await this.attributesApi.deleteResourceAttribute(Object.assign(Object.assign({}, this.config.apiContext.environmentContext), { resourceId: resourceKey, attributeId: attributeKey }));
        }
        catch (err) {
            this.handleApiError(err);
        }
    }
}
exports.ResourceAttributesApi = ResourceAttributesApi;
