// permitio/build/main/api/resource-roles.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ResourceRolesApi = void 0;
const openapi_1 = require("../openapi");
const base_1 = require("../openapi/base");
const base_2 = require("./base");
const context_1 = require("./context");
/**
 * The ResourceRolesApi class provides methods for interacting with Permit Roles.
 */
class ResourceRolesApi extends base_2.BasePermitApi {
    /**
     * Creates an instance of the ResourceRolesApi.
     * @param config - The configuration object for the Permit SDK.
     * @param logger - The logger instance for logging.
     */
    constructor(config, logger) {
        super(config, logger);
        this.resourceRoles = new openapi_1.ResourceRolesApi(this.openapiClientConfig, base_1.BASE_PATH, this.config.axiosInstance);
        this.roleDerivations = new openapi_1.ImplicitGrantsApi(this.openapiClientConfig, base_1.BASE_PATH, this.config.axiosInstance);
    }
    /**
     * Retrieves a list of resource roles.
     *
     * @param params - pagination and filtering params, @see {@link IListResourceRoles}
     * @returns A promise that resolves to an array of roles.
     * @throws {@link PermitApiError} If the API returns an error HTTP status code.
     * @throws {@link PermitContextError} If the configured {@link ApiContext} does not match the required endpoint context.
     */
    async list(params) {
        const { resourceKey, page = 1, perPage = 100 } = params;
        await this.ensureAccessLevel(context_1.ApiKeyLevel.ENVIRONMENT_LEVEL_API_KEY);
        await this.ensureContext(context_1.ApiContextLevel.ENVIRONMENT);
        try {
            return (await this.resourceRoles.listResourceRoles(Object.assign(Object.assign({}, this.config.apiContext.environmentContext), { page,
                perPage, resourceId: resourceKey }))).data;
        }
        catch (err) {
            this.handleApiError(err);
        }
    }
    /**
     * Retrieves a resource role by its key.
     *
     * @param resourceKey The key of the resource.
     * @param roleKey The key of the role.
     * @returns A promise that resolves to the role.
     * @throws {@link PermitApiError} If the API returns an error HTTP status code.
     * @throws {@link PermitContextError} If the configured {@link ApiContext} does not match the required endpoint context.
     */
    async get(resourceKey, roleKey) {
        await this.ensureAccessLevel(context_1.ApiKeyLevel.ENVIRONMENT_LEVEL_API_KEY);
        await this.ensureContext(context_1.ApiContextLevel.ENVIRONMENT);
        try {
            return (await this.resourceRoles.getResourceRole(Object.assign(Object.assign({}, this.config.apiContext.environmentContext), { resourceId: resourceKey, roleId: roleKey }))).data;
        }
        catch (err) {
            this.handleApiError(err);
        }
    }
    /**
     * Retrieves a resource role by its key.
     * Alias for the {@link get} method.
     *
     * @param resourceKey The key of the resource.
     * @param roleKey The key of the role.
     * @returns A promise that resolves to the role.
     * @throws {@link PermitApiError} If the API returns an error HTTP status code.
     * @throws {@link PermitContextError} If the configured {@link ApiContext} does not match the required endpoint context.
     */
    async getByKey(resourceKey, roleKey) {
        return await this.get(resourceKey, roleKey);
    }
    /**
     * Retrieves a resource role by its ID.
     * Alias for the {@link get} method.
     *
     * @param resourceId The ID of the resource.
     * @param roleId The ID of the role.
     * @returns A promise that resolves to the role.
     * @throws {@link PermitApiError} If the API returns an error HTTP status code.
     * @throws {@link PermitContextError} If the configured {@link ApiContext} does not match the required endpoint context.
     */
    async getById(resourceId, roleId) {
        return await this.get(resourceId, roleId);
    }
    /**
     * Creates a new role.
     *
     * @param resourceKey The key of the resource.
     * @param roleData The data for the new role.
     * @returns A promise that resolves to the created role.
     * @throws {@link PermitApiError} If the API returns an error HTTP status code.
     * @throws {@link PermitContextError} If the configured {@link ApiContext} does not match the required endpoint context.
     */
    async create(resourceKey, roleData) {
        await this.ensureAccessLevel(context_1.ApiKeyLevel.ENVIRONMENT_LEVEL_API_KEY);
        await this.ensureContext(context_1.ApiContextLevel.ENVIRONMENT);
        try {
            return (await this.resourceRoles.createResourceRole(Object.assign(Object.assign({}, this.config.apiContext.environmentContext), { resourceId: resourceKey, resourceRoleCreate: roleData }))).data;
        }
        catch (err) {
            this.handleApiError(err);
        }
    }
    /**
     * Updates a resource role.
     *
     * @param resourceKey The key of the resource.
     * @param roleKey The key of the role.
     * @param roleData The updated data for the role.
     * @returns A promise that resolves to the updated role.
     * @throws {@link PermitApiError} If the API returns an error HTTP status code.
     * @throws {@link PermitContextError} If the configured {@link ApiContext} does not match the required endpoint context.
     */
    async update(resourceKey, roleKey, roleData) {
        await this.ensureAccessLevel(context_1.ApiKeyLevel.ENVIRONMENT_LEVEL_API_KEY);
        await this.ensureContext(context_1.ApiContextLevel.ENVIRONMENT);
        try {
            return (await this.resourceRoles.updateResourceRole(Object.assign(Object.assign({}, this.config.apiContext.environmentContext), { resourceId: resourceKey, roleId: roleKey, resourceRoleUpdate: roleData }))).data;
        }
        catch (err) {
            this.handleApiError(err);
        }
    }
    /**
     * Deletes a resource role.
     *
     * @param resourceKey The key of the resource.
     * @param roleKey The key of the role to delete.
     * @returns A promise that resolves when the role is deleted.
     * @throws {@link PermitApiError} If the API returns an error HTTP status code.
     * @throws {@link PermitContextError} If the configured {@link ApiContext} does not match the required endpoint context.
     */
    async delete(resourceKey, roleKey) {
        await this.ensureAccessLevel(context_1.ApiKeyLevel.ENVIRONMENT_LEVEL_API_KEY);
        await this.ensureContext(context_1.ApiContextLevel.ENVIRONMENT);
        try {
            await this.resourceRoles.deleteResourceRole(Object.assign(Object.assign({}, this.config.apiContext.environmentContext), { resourceId: resourceKey, roleId: roleKey }));
        }
        catch (err) {
            this.handleApiError(err);
        }
    }
    /**
     * Assigns permissions to a resource role.
     *
     * @param resourceKey The key of the resource.
     * @param roleKey - The key of the role.
     * @param permissions - An array of permission keys (<resourceKey:actionKey>) to be assigned to the role.
     * @returns A promise that resolves to a ResourceRoleRead object representing the updated role.
     * @throws {@link PermitApiError} If the API returns an error HTTP status code.
     * @throws {@link PermitContextError} If the configured {@link ApiContext} does not match the required endpoint context.
     */
    async assignPermissions(resourceKey, roleKey, permissions) {
        await this.ensureAccessLevel(context_1.ApiKeyLevel.ENVIRONMENT_LEVEL_API_KEY);
        await this.ensureContext(context_1.ApiContextLevel.ENVIRONMENT);
        try {
            return (await this.resourceRoles.assignPermissionsToResourceRole(Object.assign(Object.assign({}, this.config.apiContext.environmentContext), { resourceId: resourceKey, roleId: roleKey, addRolePermissions: {
                    permissions,
                } }))).data;
        }
        catch (err) {
            this.handleApiError(err);
        }
    }
    /**
     * Removes permissions from a resource role.
     *
     * @param resourceKey The key of the resource.
     * @param roleKey - The key of the role.
     * @param permissions - An array of permission keys (<resourceKey:actionKey>) to be removed from the role.
     * @returns A promise that resolves to a ResourceRoleRead object representing the updated role.
     * @throws {@link PermitApiError} If the API returns an error HTTP status code.
     * @throws {@link PermitContextError} If the configured {@link ApiContext} does not match the required endpoint context.
     */
    async removePermissions(resourceKey, roleKey, permissions) {
        await this.ensureAccessLevel(context_1.ApiKeyLevel.ENVIRONMENT_LEVEL_API_KEY);
        await this.ensureContext(context_1.ApiContextLevel.ENVIRONMENT);
        try {
            return (await this.resourceRoles.removePermissionsFromResourceRole(Object.assign(Object.assign({}, this.config.apiContext.environmentContext), { resourceId: resourceKey, roleId: roleKey, removeRolePermissions: {
                    permissions,
                } }))).data;
        }
        catch (err) {
            this.handleApiError(err);
        }
    }
    /**
     * Create a conditional derivation from another role.
     * The derivation states that users with some other role on a related object will implicitly also be granted this role.
     *
     * @param resourceKey - The key of the resource the role belongs to.
     * @param roleKey - The key of the role.
     * @param derivationRule - A rule when to derived this role from another related role.
     * @returns A DerivedRoleRuleRead object representing the newly created role derivation.
     * @throws PermitApiError - If the API returns an error HTTP status code.
     * @throws PermitContextError - If the configured ApiContext does not match the required endpoint context.
     */
    async createRoleDerivation(resourceKey, roleKey, derivationRule) {
        await this.ensureAccessLevel(context_1.ApiKeyLevel.ENVIRONMENT_LEVEL_API_KEY);
        await this.ensureContext(context_1.ApiContextLevel.ENVIRONMENT);
        try {
            return (await this.roleDerivations.createImplicitGrant(Object.assign(Object.assign({}, this.config.apiContext.environmentContext), { resourceId: resourceKey, roleId: roleKey, derivedRoleRuleCreate: derivationRule }))).data;
        }
        catch (err) {
            this.handleApiError(err);
        }
    }
    /**
     * Delete a role derivation.
     *
     * @param resourceKey - The key of the resource the role belongs to.
     * @param roleKey - The key of the role.
     * @param derivationRule - The details of the derivation rule to delete.
     * @throws PermitApiError - If the API returns an error HTTP status code.
     * @throws PermitContextError - If the configured ApiContext does not match the required endpoint context.
     */
    async deleteRoleDerivation(resourceKey, roleKey, derivationRule) {
        await this.ensureAccessLevel(context_1.ApiKeyLevel.ENVIRONMENT_LEVEL_API_KEY);
        await this.ensureContext(context_1.ApiContextLevel.ENVIRONMENT);
        try {
            return (await this.roleDerivations.deleteImplicitGrant(Object.assign(Object.assign({}, this.config.apiContext.environmentContext), { resourceId: resourceKey, roleId: roleKey, derivedRoleRuleDelete: derivationRule }))).data;
        }
        catch (err) {
            this.handleApiError(err);
        }
    }
    /**
     * Update the optional (ABAC) conditions when to derive this role from other roles.
     *
     * @param resourceKey - The key of the resource the role belongs to.
     * @param roleKey - The key of the role.
     * @param conditions - The conditions object.
     * @returns The updated PermitBackendSchemasSchemaDerivedRoleDerivedRoleSettings.
     * @throws PermitApiError - If the API returns an error HTTP status code.
     * @throws PermitContextError - If the configured ApiContext does not match the required endpoint context.
     */
    async updateRoleDerivationConditions(resourceKey, roleKey, conditions) {
        await this.ensureAccessLevel(context_1.ApiKeyLevel.ENVIRONMENT_LEVEL_API_KEY);
        await this.ensureContext(context_1.ApiContextLevel.ENVIRONMENT);
        try {
            return (await this.roleDerivations.updateImplicitGrantsConditions(Object.assign(Object.assign({}, this.config.apiContext.environmentContext), { resourceId: resourceKey, roleId: roleKey, permitBackendSchemasSchemaDerivedRoleDerivedRoleSettings: conditions }))).data;
        }
        catch (err) {
            this.handleApiError(err);
        }
    }
}
exports.ResourceRolesApi = ResourceRolesApi;
