// permitio/build/main/api/tenants.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.TenantsApi = void 0;
const openapi_1 = require("../openapi");
const base_1 = require("../openapi/base");
const base_2 = require("./base");
const context_1 = require("./context");
/**
 * The TenantsApi class provides methods for interacting with Permit Tenants.
 */
class TenantsApi extends base_2.BasePermitApi {
    /**
     * Creates an instance of the TenantsApi.
     * @param config - The configuration object for the Permit SDK.
     * @param logger - The logger instance for logging.
     */
    constructor(config, logger) {
        super(config, logger);
        this.tenants = new openapi_1.TenantsApi(this.openapiClientConfig, base_1.BASE_PATH, this.config.axiosInstance);
    }
    /**
     * Retrieves a list of tenants.
     *
     * @param pagination The pagination options, @see {@link IPagination}
     * @returns A promise that resolves to an array of tenants.
     * @throws {@link PermitApiError} If the API returns an error HTTP status code.
     * @throws {@link PermitContextError} If the configured {@link ApiContext} does not match the required endpoint context.
     */
    async list(pagination) {
        const { page = 1, perPage = 100 } = pagination !== null && pagination !== void 0 ? pagination : {};
        await this.ensureAccessLevel(context_1.ApiKeyLevel.ENVIRONMENT_LEVEL_API_KEY);
        await this.ensureContext(context_1.ApiContextLevel.ENVIRONMENT);
        try {
            return (await this.tenants.listTenants(Object.assign(Object.assign({}, this.config.apiContext.environmentContext), { page,
                perPage }))).data;
        }
        catch (err) {
            this.handleApiError(err);
        }
    }
    /**
     * Retrieves a list of users for a given tenant.
     *
     * @param params - pagination and filtering params.
     * @returns A promise that resolves to a PaginatedResultUserRead object containing the list of tenant users.
     * @throws {@link PermitApiError} If the API returns an error HTTP status code.
     * @throws {@link PermitContextError} If the configured {@link ApiContext} does not match the required endpoint context.
     */
    async listTenantUsers({ tenantKey, page = 1, perPage = 100, }) {
        await this.ensureAccessLevel(context_1.ApiKeyLevel.ENVIRONMENT_LEVEL_API_KEY);
        await this.ensureContext(context_1.ApiContextLevel.ENVIRONMENT);
        try {
            return (await this.tenants.listTenantUsers(Object.assign(Object.assign({}, this.config.apiContext.environmentContext), { tenantId: tenantKey, page,
                perPage }))).data;
        }
        catch (err) {
            this.handleApiError(err);
        }
    }
    /**
     * Retrieves a tenant by its key.
     *
     * @param tenantKey The key of the tenant.
     * @returns A promise that resolves to the tenant.
     * @throws {@link PermitApiError} If the API returns an error HTTP status code.
     * @throws {@link PermitContextError} If the configured {@link ApiContext} does not match the required endpoint context.
     */
    async get(tenantKey) {
        await this.ensureAccessLevel(context_1.ApiKeyLevel.ENVIRONMENT_LEVEL_API_KEY);
        await this.ensureContext(context_1.ApiContextLevel.ENVIRONMENT);
        try {
            return (await this.tenants.getTenant(Object.assign(Object.assign({}, this.config.apiContext.environmentContext), { tenantId: tenantKey }))).data;
        }
        catch (err) {
            this.handleApiError(err);
        }
    }
    /**
     * Retrieves a tenant by its key.
     * Alias for the {@link get} method.
     *
     * @param tenantKey The key of the tenant.
     * @returns A promise that resolves to the tenant.
     * @throws {@link PermitApiError} If the API returns an error HTTP status code.
     * @throws {@link PermitContextError} If the configured {@link ApiContext} does not match the required endpoint context.
     */
    async getByKey(tenantKey) {
        return await this.get(tenantKey);
    }
    /**
     * Retrieves a tenant by its ID.
     * Alias for the {@link get} method.
     *
     * @param tenantId The ID of the tenant.
     * @returns A promise that resolves to the tenant.
     * @throws {@link PermitApiError} If the API returns an error HTTP status code.
     * @throws {@link PermitContextError} If the configured {@link ApiContext} does not match the required endpoint context.
     */
    async getById(tenantId) {
        return await this.get(tenantId);
    }
    /**
     * Creates a new tenant.
     *
     * @param tenantData The data for the new tenant.
     * @returns A promise that resolves to the created tenant.
     * @throws {@link PermitApiError} If the API returns an error HTTP status code.
     * @throws {@link PermitContextError} If the configured {@link ApiContext} does not match the required endpoint context.
     */
    async create(tenantData) {
        await this.ensureAccessLevel(context_1.ApiKeyLevel.ENVIRONMENT_LEVEL_API_KEY);
        await this.ensureContext(context_1.ApiContextLevel.ENVIRONMENT);
        try {
            return (await this.tenants.createTenant(Object.assign(Object.assign({}, this.config.apiContext.environmentContext), { tenantCreate: tenantData }))).data;
        }
        catch (err) {
            this.handleApiError(err);
        }
    }
    /**
     * Updates a tenant.
     *
     * @param tenantKey The key of the tenant.
     * @param tenantData The updated data for the tenant.
     * @returns A promise that resolves to the updated tenant.
     * @throws {@link PermitApiError} If the API returns an error HTTP status code.
     * @throws {@link PermitContextError} If the configured {@link ApiContext} does not match the required endpoint context.
     */
    async update(tenantKey, tenantData) {
        await this.ensureAccessLevel(context_1.ApiKeyLevel.ENVIRONMENT_LEVEL_API_KEY);
        await this.ensureContext(context_1.ApiContextLevel.ENVIRONMENT);
        try {
            return (await this.tenants.updateTenant(Object.assign(Object.assign({}, this.config.apiContext.environmentContext), { tenantId: tenantKey, tenantUpdate: tenantData }))).data;
        }
        catch (err) {
            this.handleApiError(err);
        }
    }
    /**
     * Deletes a tenant.
     *
     * @param tenantKey The key of the tenant to delete.
     * @returns A promise that resolves when the tenant is deleted.
     * @throws {@link PermitApiError} If the API returns an error HTTP status code.
     * @throws {@link PermitContextError} If the configured {@link ApiContext} does not match the required endpoint context.
     */
    async delete(tenantKey) {
        await this.ensureAccessLevel(context_1.ApiKeyLevel.ENVIRONMENT_LEVEL_API_KEY);
        await this.ensureContext(context_1.ApiContextLevel.ENVIRONMENT);
        try {
            await this.tenants.deleteTenant(Object.assign(Object.assign({}, this.config.apiContext.environmentContext), { tenantId: tenantKey }));
        }
        catch (err) {
            this.handleApiError(err);
        }
    }
    /**
     * Deletes a user from a given tenant (also removes all roles granted to the user in that tenant).
     *
     * @param tenantKey - The key of the tenant from which the user will be deleted.
     * @param userKey - The key of the user to be deleted.
     * @returns A promise that resolves when the user is successfully deleted.
     * @throws {@link PermitApiError} If the API returns an error HTTP status code.
     * @throws {@link PermitContextError} If the configured {@link ApiContext} does not match the required endpoint context.
     */
    async deleteTenantUser(tenantKey, userKey) {
        await this.ensureAccessLevel(context_1.ApiKeyLevel.ENVIRONMENT_LEVEL_API_KEY);
        await this.ensureContext(context_1.ApiContextLevel.ENVIRONMENT);
        try {
            await this.tenants.deleteTenantUser(Object.assign(Object.assign({}, this.config.apiContext.environmentContext), { tenantId: tenantKey, userId: userKey }));
        }
        catch (err) {
            this.handleApiError(err);
        }
    }
}
exports.TenantsApi = TenantsApi;
