// permitio/build/module/api/api-client.js
import { ConditionSetRulesApi } from './condition-set-rules';
import { ConditionSetsApi } from './condition-sets';
import { DeprecatedApiClient } from './deprecated';
import { EnvironmentsApi } from './environments';
import { ProjectsApi } from './projects';
import { RelationshipTuplesApi } from './relationship-tuples';
import { ResourceActionGroupsApi } from './resource-action-groups';
import { ResourceActionsApi } from './resource-actions';
import { ResourceAttributesApi } from './resource-attributes';
import { ResourceInstancesApi } from './resource-instances';
import { ResourceRelationsApi } from './resource-relations';
import { ResourceRolesApi } from './resource-roles';
import { ResourcesApi } from './resources';
import { RoleAssignmentsApi } from './role-assignments';
import { RolesApi } from './roles';
import { TenantsApi } from './tenants';
import { UsersApi } from './users';
export class ApiClient extends DeprecatedApiClient {
    /**
     * API for managing condition set rules.
     * @see {@link https://api.permit.io/v2/redoc#tag/Condition-Set-Rules}
     */
    conditionSetRules;
    /**
     * API for managing condition sets.
     * @see {@link https://api.permit.io/v2/redoc#tag/Condition-Sets}
     */
    conditionSets;
    /**
     * API for managing projects.
     * @see {@link https://api.permit.io/v2/redoc#tag/Projects}
     */
    projects;
    /**
     * API for managing environments.
     * @see {@link https://api.permit.io/v2/redoc#tag/Environments}
     */
    environments;
    /**
     * API for managing resource action groups.
     * @see {@link https://api.permit.io/v2/redoc#tag/Resource-Action-Groups}
     */
    actionGroups;
    /**
     * API for managing resource actions.
     * @see {@link https://api.permit.io/v2/redoc#tag/Resource-Actions}
     */
    resourceActions;
    /**
     * API for managing resource attributes.
     * @see {@link https://api.permit.io/v2/redoc#tag/Resource-Attributes}
     */
    resourceAttributes;
    /**
     * API for managing resource roles.
     * @see {@link https://api.permit.io/v2/redoc#tag/Resource-Roles}
     */
    resourceRoles;
    /**
     * API for managing resource relations.
     * @see {@link https://api.permit.io/v2/redoc#tag/Resource-Relations}
     */
    resourceRelations;
    /**
     * API for managing resource instances.
     * @see {@link https://api.permit.io/v2/redoc#tag/Resource-Instances}
     */
    resourceInstances;
    /**
     * API for managing resources.
     * @see {@link https://api.permit.io/v2/redoc#tag/Resources}
     */
    resources;
    /**
     * API for managing role assignments.
     * @see {@link https://api.permit.io/v2/redoc#tag/Role-Assignments}
     */
    roleAssignments;
    /**
     * API for managing relationship tuples.
     * @see {@link https://api.permit.io/v2/redoc#tag/Relationship-Tuples}
     */
    relationshipTuples;
    /**
     * API for managing roles.
     * @see {@link https://api.permit.io/v2/redoc#tag/Roles}
     */
    roles;
    /**
     * API for managing tenants.
     * @see {@link https://api.permit.io/v2/redoc#tag/Tenants}
     */
    tenants;
    /**
     * API for managing users.
     * @see {@link https://api.permit.io/v2/redoc#tag/Users}
     */
    users;
    /**
     * Constructs a new instance of the ApiClient class with the specified configuration and logger.
     *
     * @param config - The configuration for the Permit API client.
     * @param logger - The logger instance.
     */
    constructor(config, logger) {
        super(config, logger);
        this.conditionSetRules = new ConditionSetRulesApi(config, logger);
        this.conditionSets = new ConditionSetsApi(config, logger);
        this.projects = new ProjectsApi(config, logger);
        this.environments = new EnvironmentsApi(config, logger);
        this.actionGroups = new ResourceActionGroupsApi(config, logger);
        this.resourceActions = new ResourceActionsApi(config, logger);
        this.resourceAttributes = new ResourceAttributesApi(config, logger);
        this.resourceRoles = new ResourceRolesApi(config, logger);
        this.resourceRelations = new ResourceRelationsApi(config, logger);
        this.resourceInstances = new ResourceInstancesApi(config, logger);
        this.resources = new ResourcesApi(config, logger);
        this.roleAssignments = new RoleAssignmentsApi(config, logger);
        this.roles = new RolesApi(config, logger);
        this.relationshipTuples = new RelationshipTuplesApi(config, logger);
        this.tenants = new TenantsApi(config, logger);
        this.users = new UsersApi(config, logger);
    }
}
