// permitio/build/module/api/condition-set-rules.js
import { ConditionSetRulesApi as AutogenConditionSetRulesApi, } from '../openapi';
import { BASE_PATH } from '../openapi/base';
import { BasePermitApi } from './base'; // eslint-disable-line @typescript-eslint/no-unused-vars
import { ApiContextLevel, ApiKeyLevel } from './context'; // eslint-disable-line @typescript-eslint/no-unused-vars
export class ConditionSetRulesApi extends BasePermitApi {
    setRules;
    /**
     * Creates an instance of the ConditionSetRulesApi.
     * @param config - The configuration object for the Permit SDK.
     * @param logger - The logger instance for logging.
     */
    constructor(config, logger) {
        super(config, logger);
        this.setRules = new AutogenConditionSetRulesApi(this.openapiClientConfig, BASE_PATH, this.config.axiosInstance);
    }
    /**
     * Retrieves a list of condition set rules based on the specified parameters.
     *
     * @param params - parameters for filtering and pagination, @see {@link IListConditionSetRules}
     * @returns A promise that resolves to an array of condition set rules.
     * @throws {@link PermitApiError} If the API returns an error HTTP status code.
     * @throws {@link PermitContextError} If the configured {@link ApiContext} does not match the required endpoint context.
     */
    async list(params) {
        await this.ensureAccessLevel(ApiKeyLevel.ENVIRONMENT_LEVEL_API_KEY);
        await this.ensureContext(ApiContextLevel.ENVIRONMENT);
        const { userSetKey, permissionKey, resourceSetKey, page = 1, perPage = 100 } = params;
        try {
            return (await this.setRules.listSetPermissions({
                ...this.config.apiContext.environmentContext,
                userSet: userSetKey,
                permission: permissionKey,
                resourceSet: resourceSetKey,
                page,
                perPage,
            })).data;
        }
        catch (err) {
            this.handleApiError(err);
        }
    }
    /**
     * Creates a new condition set rule.
     *
     * @param rule - The condition set rule to create.
     * @returns A promise that resolves to the created condition set rule.
     * @throws {@link PermitApiError} If the API returns an error HTTP status code.
     * @throws {@link PermitContextError} If the configured {@link ApiContext} does not match the required endpoint context.
     */
    async create(rule) {
        await this.ensureAccessLevel(ApiKeyLevel.ENVIRONMENT_LEVEL_API_KEY);
        await this.ensureContext(ApiContextLevel.ENVIRONMENT);
        try {
            return (await this.setRules.assignSetPermissions({
                ...this.config.apiContext.environmentContext,
                conditionSetRuleCreate: rule,
            })).data[0];
        }
        catch (err) {
            this.handleApiError(err);
        }
    }
    /**
     * Deletes a condition set rule.
     *
     * @param rule - The condition set rule to delete.
     * @returns A promise that resolves when the condition set rule is deleted.
     * @throws {@link PermitApiError} If the API returns an error HTTP status code.
     * @throws {@link PermitContextError} If the configured {@link ApiContext} does not match the required endpoint context.
     */
    async delete(rule) {
        await this.ensureAccessLevel(ApiKeyLevel.ENVIRONMENT_LEVEL_API_KEY);
        await this.ensureContext(ApiContextLevel.ENVIRONMENT);
        try {
            return (await this.setRules.unassignSetPermissions({
                ...this.config.apiContext.environmentContext,
                conditionSetRuleRemove: rule,
            })).data;
        }
        catch (err) {
            this.handleApiError(err);
        }
    }
}
