// permitio/build/module/api/condition-sets.js
import { ConditionSetsApi as AutogenConditionSetsApi, } from '../openapi';
import { BASE_PATH } from '../openapi/base';
import { BasePermitApi } from './base'; // eslint-disable-line @typescript-eslint/no-unused-vars
import { ApiContextLevel, ApiKeyLevel } from './context'; // eslint-disable-line @typescript-eslint/no-unused-vars
/**
 * The ConditionSetsApi class provides methods for interacting with condition sets using the Permit REST API.
 */
export class ConditionSetsApi extends BasePermitApi {
    conditionSets;
    /**
     * Creates an instance of the ConditionSetsApi.
     * @param config - The configuration object for the Permit SDK.
     * @param logger - The logger instance for logging.
     */
    constructor(config, logger) {
        super(config, logger);
        this.conditionSets = new AutogenConditionSetsApi(this.openapiClientConfig, BASE_PATH, this.config.axiosInstance);
    }
    /**
     * Retrieves a list of condition sets.
     *
     * @param pagination The pagination options, @see {@link IPagination}
     * @returns A promise that resolves to an array of condition sets.
     * @throws {@link PermitApiError} If the API returns an error HTTP status code.
     * @throws {@link PermitContextError} If the configured {@link ApiContext} does not match the required endpoint context.
     */
    async list(pagination) {
        const { page = 1, perPage = 100 } = pagination ?? {};
        await this.ensureAccessLevel(ApiKeyLevel.ENVIRONMENT_LEVEL_API_KEY);
        await this.ensureContext(ApiContextLevel.ENVIRONMENT);
        try {
            return (await this.conditionSets.listConditionSets({
                ...this.config.apiContext.environmentContext,
                page,
                perPage,
            })).data;
        }
        catch (err) {
            this.handleApiError(err);
        }
    }
    /**
     * Retrieves a condition set by its key.
     *
     * @param conditionSetKey The key of the condition set.
     * @returns A promise that resolves to the condition set.
     * @throws {@link PermitApiError} If the API returns an error HTTP status code.
     * @throws {@link PermitContextError} If the configured {@link ApiContext} does not match the required endpoint context.
     */
    async get(conditionSetKey) {
        await this.ensureAccessLevel(ApiKeyLevel.ENVIRONMENT_LEVEL_API_KEY);
        await this.ensureContext(ApiContextLevel.ENVIRONMENT);
        try {
            return (await this.conditionSets.getConditionSet({
                ...this.config.apiContext.environmentContext,
                conditionSetId: conditionSetKey,
            })).data;
        }
        catch (err) {
            this.handleApiError(err);
        }
    }
    /**
     * Retrieves a condition set by its key.
     * Alias for the {@link get} method.
     *
     * @param conditionSetKey The key of the condition set.
     * @returns A promise that resolves to the condition set.
     * @throws {@link PermitApiError} If the API returns an error HTTP status code.
     * @throws {@link PermitContextError} If the configured {@link ApiContext} does not match the required endpoint context.
     */
    async getByKey(conditionSetKey) {
        return await this.get(conditionSetKey);
    }
    /**
     * Retrieves a condition set by its ID.
     * Alias for the {@link get} method.
     *
     * @param conditionSetId The ID of the condition set.
     * @returns A promise that resolves to the condition set.
     * @throws {@link PermitApiError} If the API returns an error HTTP status code.
     * @throws {@link PermitContextError} If the configured {@link ApiContext} does not match the required endpoint context.
     */
    async getById(conditionSetId) {
        return await this.get(conditionSetId);
    }
    /**
     * Creates a new condition set.
     *
     * @param conditionSetData The data for the new condition set.
     * @returns A promise that resolves to the created condition set.
     * @throws {@link PermitApiError} If the API returns an error HTTP status code.
     * @throws {@link PermitContextError} If the configured {@link ApiContext} does not match the required endpoint context.
     */
    async create(conditionSetData) {
        await this.ensureAccessLevel(ApiKeyLevel.ENVIRONMENT_LEVEL_API_KEY);
        await this.ensureContext(ApiContextLevel.ENVIRONMENT);
        try {
            return (await this.conditionSets.createConditionSet({
                ...this.config.apiContext.environmentContext,
                conditionSetCreate: conditionSetData,
            })).data;
        }
        catch (err) {
            this.handleApiError(err);
        }
    }
    /**
     * Updates a condition set.
     *
     * @param conditionSetKey The key of the condition set.
     * @param conditionSetData The updated data for the condition set.
     * @returns A promise that resolves to the updated condition set.
     * @throws {@link PermitApiError} If the API returns an error HTTP status code.
     * @throws {@link PermitContextError} If the configured {@link ApiContext} does not match the required endpoint context.
     */
    async update(conditionSetKey, conditionSetData) {
        await this.ensureAccessLevel(ApiKeyLevel.ENVIRONMENT_LEVEL_API_KEY);
        await this.ensureContext(ApiContextLevel.ENVIRONMENT);
        try {
            return (await this.conditionSets.updateConditionSet({
                ...this.config.apiContext.environmentContext,
                conditionSetId: conditionSetKey,
                conditionSetUpdate: conditionSetData,
            })).data;
        }
        catch (err) {
            this.handleApiError(err);
        }
    }
    /**
     * Deletes a condition set.
     *
     * @param conditionSetKey The key of the condition set to delete.
     * @returns A promise that resolves when the condition set is deleted.
     * @throws {@link PermitApiError} If the API returns an error HTTP status code.
     * @throws {@link PermitContextError} If the configured {@link ApiContext} does not match the required endpoint context.
     */
    async delete(conditionSetKey) {
        await this.ensureAccessLevel(ApiKeyLevel.ENVIRONMENT_LEVEL_API_KEY);
        await this.ensureContext(ApiContextLevel.ENVIRONMENT);
        try {
            await this.conditionSets.deleteConditionSet({
                ...this.config.apiContext.environmentContext,
                conditionSetId: conditionSetKey,
            });
        }
        catch (err) {
            this.handleApiError(err);
        }
    }
}
