// permitio/build/module/api/elements.js
import { AuthenticationApi } from '../openapi';
import { BASE_PATH } from '../openapi/base';
import { BasePermitApi } from './base';
/**
 * Represents the possible error messages returned by the Elements API.
 */
export var ElementsApiErrors;
(function (ElementsApiErrors) {
    ElementsApiErrors["USER_NOT_FOUND"] = "User not found";
    ElementsApiErrors["TENANT_NOT_FOUND"] = "Tenant not found";
    ElementsApiErrors["INVALID_PERMISSION_LEVEL"] = "Invalid user permission level";
    ElementsApiErrors["FORBIDDEN_ACCESS"] = "Forbidden access";
})(ElementsApiErrors || (ElementsApiErrors = {}));
/**
 * API client for interacting with the Permit Elements API.
 */
export class ElementsClient extends BasePermitApi {
    authApi;
    /**
     * Creates an instance of the ElementsClient.
     * @param config - The configuration object for the Permit SDK.
     * @param logger - The logger instance for logging.
     */
    constructor(config, logger) {
        super(config, logger);
        this.authApi = new AuthenticationApi(this.openapiClientConfig, BASE_PATH, this.config.axiosInstance);
    }
    /**
     * Logs in as a user in the context of a specific tenant.
     * @param loginData - The login data containing the user key and tenant key.
     * @returns The embedded login authentication session data.
     * @throws {@link PermitApiError} If the API returns an error HTTP status code.
     * @throws {@link PermitContextError} If the configured {@link ApiContext} does not match the required endpoint context.
     */
    async loginAs({ userId, tenantId, }) {
        try {
            const response = await this.authApi.elementsLoginAs({
                userLoginRequestInput: {
                    user_id: userId,
                    tenant_id: tenantId,
                },
            });
            return {
                ...response.data,
                content: { url: response.data.redirect_url },
            };
        }
        catch (err) {
            this.handleApiError(err);
        }
    }
}
