// permitio/build/module/api/projects.js
import { ProjectsApi as AutogenProjectsApi, } from '../openapi';
import { BASE_PATH } from '../openapi/base';
import { BasePermitApi } from './base';
import { ApiContextLevel, ApiKeyLevel } from './context';
/**
 * The ProjectsApi class provides methods for interacting with Permit Projects.
 */
export class ProjectsApi extends BasePermitApi {
    projects;
    /**
     * Creates an instance of the ProjectsApi.
     * @param config - The configuration object for the Permit SDK.
     * @param logger - The logger instance for logging.
     */
    constructor(config, logger) {
        super(config, logger);
        this.projects = new AutogenProjectsApi(this.openapiClientConfig, BASE_PATH, this.config.axiosInstance);
    }
    /**
     * Retrieves a list of projects.
     *
     * @param pagination The pagination options, @see {@link IPagination}
     * @returns A promise that resolves to an array of projects.
     * @throws {@link PermitApiError} If the API returns an error HTTP status code.
     * @throws {@link PermitContextError} If the configured {@link ApiContext} does not match the required endpoint context.
     */
    async list(pagination) {
        const { page = 1, perPage = 100 } = pagination ?? {};
        await this.ensureAccessLevel(ApiKeyLevel.ENVIRONMENT_LEVEL_API_KEY);
        await this.ensureContext(ApiContextLevel.ORGANIZATION);
        try {
            return (await this.projects.listProjects({
                page,
                perPage,
            })).data;
        }
        catch (err) {
            this.handleApiError(err);
        }
    }
    /**
     * Retrieves a project by its key.
     *
     * @param projectKey The key of the project.
     * @returns A promise that resolves to the project.
     * @throws {@link PermitApiError} If the API returns an error HTTP status code.
     * @throws {@link PermitContextError} If the configured {@link ApiContext} does not match the required endpoint context.
     */
    async get(projectKey) {
        await this.ensureAccessLevel(ApiKeyLevel.ENVIRONMENT_LEVEL_API_KEY);
        await this.ensureContext(ApiContextLevel.ORGANIZATION);
        try {
            return (await this.projects.getProject({
                projId: projectKey,
            })).data;
        }
        catch (err) {
            this.handleApiError(err);
        }
    }
    /**
     * Retrieves a project by its key.
     * Alias for the {@link get} method.
     *
     * @param projectKey The key of the project.
     * @returns A promise that resolves to the project.
     * @throws {@link PermitApiError} If the API returns an error HTTP status code.
     * @throws {@link PermitContextError} If the configured {@link ApiContext} does not match the required endpoint context.
     */
    async getByKey(projectKey) {
        return await this.get(projectKey);
    }
    /**
     * Retrieves a project by its ID.
     * Alias for the {@link get} method.
     *
     * @param projectId The ID of the project.
     * @returns A promise that resolves to the project.
     * @throws {@link PermitApiError} If the API returns an error HTTP status code.
     * @throws {@link PermitContextError} If the configured {@link ApiContext} does not match the required endpoint context.
     */
    async getById(projectId) {
        return await this.get(projectId);
    }
    /**
     * Creates a new project.
     *
     * @param projectData The data for the new project.
     * @returns A promise that resolves to the created project.
     * @throws {@link PermitApiError} If the API returns an error HTTP status code.
     * @throws {@link PermitContextError} If the configured {@link ApiContext} does not match the required endpoint context.
     */
    async create(projectData) {
        await this.ensureAccessLevel(ApiKeyLevel.ORGANIZATION_LEVEL_API_KEY);
        await this.ensureContext(ApiContextLevel.ORGANIZATION);
        try {
            return (await this.projects.createProject({
                projectCreate: projectData,
            })).data;
        }
        catch (err) {
            this.handleApiError(err);
        }
    }
    /**
     * Updates a project.
     *
     * @param projectKey The key of the project.
     * @param projectData The updated data for the project.
     * @returns A promise that resolves to the updated project.
     * @throws {@link PermitApiError} If the API returns an error HTTP status code.
     * @throws {@link PermitContextError} If the configured {@link ApiContext} does not match the required endpoint context.
     */
    async update(projectKey, projectData) {
        await this.ensureAccessLevel(ApiKeyLevel.PROJECT_LEVEL_API_KEY);
        await this.ensureContext(ApiContextLevel.ORGANIZATION);
        try {
            return (await this.projects.updateProject({
                projId: projectKey,
                projectUpdate: projectData,
            })).data;
        }
        catch (err) {
            this.handleApiError(err);
        }
    }
    /**
     * Deletes a project.
     *
     * @param projectKey The key of the project to delete.
     * @returns A promise that resolves when the project is deleted.
     * @throws {@link PermitApiError} If the API returns an error HTTP status code.
     * @throws {@link PermitContextError} If the configured {@link ApiContext} does not match the required endpoint context.
     */
    async delete(projectKey) {
        await this.ensureAccessLevel(ApiKeyLevel.PROJECT_LEVEL_API_KEY);
        await this.ensureContext(ApiContextLevel.ORGANIZATION);
        try {
            await this.projects.deleteProject({
                projId: projectKey,
            });
        }
        catch (err) {
            this.handleApiError(err);
        }
    }
}
