// permitio/build/module/api/resource-actions.js
import { ResourceActionsApi as AutogenResourceActionsApi, } from '../openapi';
import { BASE_PATH } from '../openapi/base';
import { BasePermitApi } from './base'; // eslint-disable-line @typescript-eslint/no-unused-vars
import { ApiContextLevel, ApiKeyLevel } from './context'; // eslint-disable-line @typescript-eslint/no-unused-vars
/**
 * API client for interacting with the Resource Actions API.
 */
export class ResourceActionsApi extends BasePermitApi {
    actionsApi;
    /**
     * Creates an instance of the ResourceActionsApi.
     * @param config - The configuration object for the Permit SDK.
     * @param logger - The logger instance for logging.
     */
    constructor(config, logger) {
        super(config, logger);
        this.actionsApi = new AutogenResourceActionsApi(this.openapiClientConfig, BASE_PATH, this.config.axiosInstance);
    }
    /**
     * Retrieves a list of all actions that are defined for a given resource.
     * @param params - pagination and filtering params, @see {@link IListActions}
     * @returns A promise that resolves to an array of ResourceActionRead objects representing the actions.
     * @throws {@link PermitApiError} If the API returns an error HTTP status code.
     * @throws {@link PermitContextError} If the configured {@link ApiContext} does not match the required endpoint context.
     */
    async list(params) {
        const { resourceKey, page = 1, perPage = 100 } = params;
        await this.ensureAccessLevel(ApiKeyLevel.ENVIRONMENT_LEVEL_API_KEY);
        await this.ensureContext(ApiContextLevel.ENVIRONMENT);
        try {
            return (await this.actionsApi.listResourceActions({
                ...this.config.apiContext.environmentContext,
                resourceId: resourceKey,
                page,
                perPage,
            })).data;
        }
        catch (err) {
            this.handleApiError(err);
        }
    }
    /**
     * Retrieves an action based on the resource key and the action key.
     *
     * @param resourceKey - The resource key.
     * @param actionKey - The action key.
     * @returns A promise that resolves to a ResourceActionRead object representing the action.
     * @throws {@link PermitApiError} If the API returns an error HTTP status code.
     * @throws {@link PermitContextError} If the configured {@link ApiContext} does not match the required endpoint context.
     */
    async get(resourceKey, actionKey) {
        await this.ensureAccessLevel(ApiKeyLevel.ENVIRONMENT_LEVEL_API_KEY);
        await this.ensureContext(ApiContextLevel.ENVIRONMENT);
        try {
            return (await this.actionsApi.getResourceAction({
                ...this.config.apiContext.environmentContext,
                resourceId: resourceKey,
                actionId: actionKey,
            })).data;
        }
        catch (err) {
            this.handleApiError(err);
        }
    }
    /**
     * Retrieves an action based on the resource key and the action key.
     * Alias for the {@link get} method.
     *
     * @param resourceKey - The resource key.
     * @param actionKey - The action key.
     * @returns A promise that resolves to a ResourceActionRead object representing the action.
     * @throws {@link PermitApiError} If the API returns an error HTTP status code.
     * @throws {@link PermitContextError} If the configured {@link ApiContext} does not match the required endpoint context.
     */
    async getByKey(resourceKey, actionKey) {
        return await this.get(resourceKey, actionKey);
    }
    /**
     * Retrieves an action based on the resource ID and the action ID.
     * Alias for the {@link get} method.
     *
     * @param resourceId - The resource ID.
     * @param actionId - The action ID.
     * @returns A promise that resolves to a ResourceActionRead object representing the action.
     * @throws {@link PermitApiError} If the API returns an error HTTP status code.
     * @throws {@link PermitContextError} If the configured {@link ApiContext} does not match the required endpoint context.
     */
    async getById(resourceId, actionId) {
        return await this.get(resourceId, actionId);
    }
    /**
     * Creates a new action.
     *
     * @param resourceKey - The resource key.
     * @param actionData - The action data.
     * @returns A promise that resolves to a ResourceActionRead object representing the created action.
     * @throws {@link PermitApiError} If the API returns an error HTTP status code.
     * @throws {@link PermitContextError} If the configured {@link ApiContext} does not match the required endpoint context.
     */
    async create(resourceKey, actionData) {
        await this.ensureAccessLevel(ApiKeyLevel.ENVIRONMENT_LEVEL_API_KEY);
        await this.ensureContext(ApiContextLevel.ENVIRONMENT);
        try {
            return (await this.actionsApi.createResourceAction({
                ...this.config.apiContext.environmentContext,
                resourceId: resourceKey,
                resourceActionCreate: actionData,
            })).data;
        }
        catch (err) {
            this.handleApiError(err);
        }
    }
    /**
     * Updates an existing action.
     *
     * @param resourceKey - The resource key.
     * @param actionKey - The key of the action to modify.
     * @param actionData - The data for updating the action.
     * @returns A promise that resolves to a ResourceActionRead object representing the updated action.
     * @throws {@link PermitApiError} If the API returns an error HTTP status code.
     * @throws {@link PermitContextError} If the configured {@link ApiContext} does not match the required endpoint context.
     */
    async update(resourceKey, actionKey, actionData) {
        await this.ensureAccessLevel(ApiKeyLevel.ENVIRONMENT_LEVEL_API_KEY);
        await this.ensureContext(ApiContextLevel.ENVIRONMENT);
        try {
            return (await this.actionsApi.updateResourceAction({
                ...this.config.apiContext.environmentContext,
                resourceId: resourceKey,
                actionId: actionKey,
                resourceActionUpdate: actionData,
            })).data;
        }
        catch (err) {
            this.handleApiError(err);
        }
    }
    /**
     * Deletes a action based on the resource key and action key.
     * @param resourceKey - The resource key.
     * @param actionKey - The action key.
     * @returns A promise that resolves when the action is successfully deleted.
     * @throws {@link PermitApiError} If the API returns an error HTTP status code.
     * @throws {@link PermitContextError} If the configured {@link ApiContext} does not match the required endpoint context.
     */
    async delete(resourceKey, actionKey) {
        await this.ensureAccessLevel(ApiKeyLevel.ENVIRONMENT_LEVEL_API_KEY);
        await this.ensureContext(ApiContextLevel.ENVIRONMENT);
        try {
            await this.actionsApi.deleteResourceAction({
                ...this.config.apiContext.environmentContext,
                resourceId: resourceKey,
                actionId: actionKey,
            });
        }
        catch (err) {
            this.handleApiError(err);
        }
    }
}
