// permitio/build/main/openapi/api/activity-log-api.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Permit.io API
 *  Authorization as a service
 *
 * The version of the OpenAPI document: 2.0.0
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.ActivityLogApi = exports.ActivityLogApiFactory = exports.ActivityLogApiFp = exports.ActivityLogApiAxiosParamCreator = void 0;
const axios_1 = __importDefault(require("axios"));
// Some imports not used depending on template conditions
// @ts-ignore
const common_1 = require("../common");
// @ts-ignore
const base_1 = require("../base");
/**
 * ActivityLogApi - axios parameter creator
 * @export
 */
const ActivityLogApiAxiosParamCreator = function (configuration) {
    return {
        /**
         * Fetches all API History events for given filters.
         * @summary List Activity Events
         * @param {Array<string>} [actorId] Actor ID to search
         * @param {string} [actorDisplayName] Actor display name to search
         * @param {Array<string>} [actorType] Actor type to search (member, user, api_key)
         * @param {Array<string>} [projectId] Projects to include in search
         * @param {Array<string>} [envId] Environments to include in search
         * @param {number} [timestampFrom] Retrieve only events after this timestamp (seconds since epoch)
         * @param {number} [timestampUntil] Retrieve only events before this timestamp (seconds since epoch)
         * @param {number} [page] Page number of the results to fetch, starting at 1.
         * @param {number} [perPage] The number of results per page (max 100).
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        listActivityEvents: async (actorId, actorDisplayName, actorType, projectId, envId, timestampFrom, timestampUntil, page, perPage, options = {}) => {
            const localVarPath = `/v2/activity`;
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new URL(localVarPath, common_1.DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = Object.assign(Object.assign({ method: 'GET' }, baseOptions), options);
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            // authentication HTTPBearer required
            // http bearer authentication required
            await (0, common_1.setBearerAuthToObject)(localVarHeaderParameter, configuration);
            if (actorId) {
                localVarQueryParameter['actor_id'] = actorId;
            }
            if (actorDisplayName !== undefined) {
                localVarQueryParameter['actor_display_name'] = actorDisplayName;
            }
            if (actorType) {
                localVarQueryParameter['actor_type'] = actorType;
            }
            if (projectId) {
                localVarQueryParameter['project_id'] = projectId;
            }
            if (envId) {
                localVarQueryParameter['env_id'] = envId;
            }
            if (timestampFrom !== undefined) {
                localVarQueryParameter['timestamp_from'] = timestampFrom;
            }
            if (timestampUntil !== undefined) {
                localVarQueryParameter['timestamp_until'] = timestampUntil;
            }
            if (page !== undefined) {
                localVarQueryParameter['page'] = page;
            }
            if (perPage !== undefined) {
                localVarQueryParameter['per_page'] = perPage;
            }
            (0, common_1.setSearchParams)(localVarUrlObj, localVarQueryParameter);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = Object.assign(Object.assign(Object.assign({}, localVarHeaderParameter), headersFromBaseOptions), options.headers);
            return {
                url: (0, common_1.toPathString)(localVarUrlObj),
                options: localVarRequestOptions,
            };
        },
        /**
         *
         * @summary List Activity Types
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        listActivityTypes: async (options = {}) => {
            const localVarPath = `/v2/activity/types`;
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new URL(localVarPath, common_1.DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = Object.assign(Object.assign({ method: 'GET' }, baseOptions), options);
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            // authentication HTTPBearer required
            // http bearer authentication required
            await (0, common_1.setBearerAuthToObject)(localVarHeaderParameter, configuration);
            (0, common_1.setSearchParams)(localVarUrlObj, localVarQueryParameter);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = Object.assign(Object.assign(Object.assign({}, localVarHeaderParameter), headersFromBaseOptions), options.headers);
            return {
                url: (0, common_1.toPathString)(localVarUrlObj),
                options: localVarRequestOptions,
            };
        },
    };
};
exports.ActivityLogApiAxiosParamCreator = ActivityLogApiAxiosParamCreator;
/**
 * ActivityLogApi - functional programming interface
 * @export
 */
const ActivityLogApiFp = function (configuration) {
    const localVarAxiosParamCreator = (0, exports.ActivityLogApiAxiosParamCreator)(configuration);
    return {
        /**
         * Fetches all API History events for given filters.
         * @summary List Activity Events
         * @param {Array<string>} [actorId] Actor ID to search
         * @param {string} [actorDisplayName] Actor display name to search
         * @param {Array<string>} [actorType] Actor type to search (member, user, api_key)
         * @param {Array<string>} [projectId] Projects to include in search
         * @param {Array<string>} [envId] Environments to include in search
         * @param {number} [timestampFrom] Retrieve only events after this timestamp (seconds since epoch)
         * @param {number} [timestampUntil] Retrieve only events before this timestamp (seconds since epoch)
         * @param {number} [page] Page number of the results to fetch, starting at 1.
         * @param {number} [perPage] The number of results per page (max 100).
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        async listActivityEvents(actorId, actorDisplayName, actorType, projectId, envId, timestampFrom, timestampUntil, page, perPage, options) {
            const localVarAxiosArgs = await localVarAxiosParamCreator.listActivityEvents(actorId, actorDisplayName, actorType, projectId, envId, timestampFrom, timestampUntil, page, perPage, options);
            return (0, common_1.createRequestFunction)(localVarAxiosArgs, axios_1.default, base_1.BASE_PATH, configuration);
        },
        /**
         *
         * @summary List Activity Types
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        async listActivityTypes(options) {
            const localVarAxiosArgs = await localVarAxiosParamCreator.listActivityTypes(options);
            return (0, common_1.createRequestFunction)(localVarAxiosArgs, axios_1.default, base_1.BASE_PATH, configuration);
        },
    };
};
exports.ActivityLogApiFp = ActivityLogApiFp;
/**
 * ActivityLogApi - factory interface
 * @export
 */
const ActivityLogApiFactory = function (configuration, basePath, axios) {
    const localVarFp = (0, exports.ActivityLogApiFp)(configuration);
    return {
        /**
         * Fetches all API History events for given filters.
         * @summary List Activity Events
         * @param {Array<string>} [actorId] Actor ID to search
         * @param {string} [actorDisplayName] Actor display name to search
         * @param {Array<string>} [actorType] Actor type to search (member, user, api_key)
         * @param {Array<string>} [projectId] Projects to include in search
         * @param {Array<string>} [envId] Environments to include in search
         * @param {number} [timestampFrom] Retrieve only events after this timestamp (seconds since epoch)
         * @param {number} [timestampUntil] Retrieve only events before this timestamp (seconds since epoch)
         * @param {number} [page] Page number of the results to fetch, starting at 1.
         * @param {number} [perPage] The number of results per page (max 100).
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        listActivityEvents(actorId, actorDisplayName, actorType, projectId, envId, timestampFrom, timestampUntil, page, perPage, options) {
            return localVarFp
                .listActivityEvents(actorId, actorDisplayName, actorType, projectId, envId, timestampFrom, timestampUntil, page, perPage, options)
                .then((request) => request(axios, basePath));
        },
        /**
         *
         * @summary List Activity Types
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        listActivityTypes(options) {
            return localVarFp.listActivityTypes(options).then((request) => request(axios, basePath));
        },
    };
};
exports.ActivityLogApiFactory = ActivityLogApiFactory;
/**
 * ActivityLogApi - object-oriented interface
 * @export
 * @class ActivityLogApi
 * @extends {BaseAPI}
 */
class ActivityLogApi extends base_1.BaseAPI {
    /**
     * Fetches all API History events for given filters.
     * @summary List Activity Events
     * @param {ActivityLogApiListActivityEventsRequest} requestParameters Request parameters.
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     * @memberof ActivityLogApi
     */
    listActivityEvents(requestParameters = {}, options) {
        return (0, exports.ActivityLogApiFp)(this.configuration)
            .listActivityEvents(requestParameters.actorId, requestParameters.actorDisplayName, requestParameters.actorType, requestParameters.projectId, requestParameters.envId, requestParameters.timestampFrom, requestParameters.timestampUntil, requestParameters.page, requestParameters.perPage, options)
            .then((request) => request(this.axios, this.basePath));
    }
    /**
     *
     * @summary List Activity Types
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     * @memberof ActivityLogApi
     */
    listActivityTypes(options) {
        return (0, exports.ActivityLogApiFp)(this.configuration)
            .listActivityTypes(options)
            .then((request) => request(this.axios, this.basePath));
    }
}
exports.ActivityLogApi = ActivityLogApi;
