// permitio/build/main/openapi/api/decision-logs-ingress-api.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Permit.io API
 *  Authorization as a service
 *
 * The version of the OpenAPI document: 2.0.0
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.DecisionLogsIngressApi = exports.DecisionLogsIngressApiFactory = exports.DecisionLogsIngressApiFp = exports.DecisionLogsIngressApiAxiosParamCreator = void 0;
const axios_1 = __importDefault(require("axios"));
// Some imports not used depending on template conditions
// @ts-ignore
const common_1 = require("../common");
// @ts-ignore
const base_1 = require("../base");
/**
 * DecisionLogsIngressApi - axios parameter creator
 * @export
 */
const DecisionLogsIngressApiAxiosParamCreator = function (configuration) {
    return {
        /**
         * This ingress endpoint captures OPA decision logs.
         * @summary OPA Decision Logs Ingress
         * @param {Array<object>} requestBody
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        insertOpaDecisionLogs: async (requestBody, options = {}) => {
            // verify required parameter 'requestBody' is not null or undefined
            (0, common_1.assertParamExists)('insertOpaDecisionLogs', 'requestBody', requestBody);
            const localVarPath = `/v2/decision-logs/ingress`;
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new URL(localVarPath, common_1.DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = Object.assign(Object.assign({ method: 'POST' }, baseOptions), options);
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            // authentication HTTPBearer required
            // http bearer authentication required
            await (0, common_1.setBearerAuthToObject)(localVarHeaderParameter, configuration);
            localVarHeaderParameter['Content-Type'] = 'application/json';
            (0, common_1.setSearchParams)(localVarUrlObj, localVarQueryParameter);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = Object.assign(Object.assign(Object.assign({}, localVarHeaderParameter), headersFromBaseOptions), options.headers);
            localVarRequestOptions.data = (0, common_1.serializeDataIfNeeded)(requestBody, localVarRequestOptions, configuration);
            return {
                url: (0, common_1.toPathString)(localVarUrlObj),
                options: localVarRequestOptions,
            };
        },
    };
};
exports.DecisionLogsIngressApiAxiosParamCreator = DecisionLogsIngressApiAxiosParamCreator;
/**
 * DecisionLogsIngressApi - functional programming interface
 * @export
 */
const DecisionLogsIngressApiFp = function (configuration) {
    const localVarAxiosParamCreator = (0, exports.DecisionLogsIngressApiAxiosParamCreator)(configuration);
    return {
        /**
         * This ingress endpoint captures OPA decision logs.
         * @summary OPA Decision Logs Ingress
         * @param {Array<object>} requestBody
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        async insertOpaDecisionLogs(requestBody, options) {
            const localVarAxiosArgs = await localVarAxiosParamCreator.insertOpaDecisionLogs(requestBody, options);
            return (0, common_1.createRequestFunction)(localVarAxiosArgs, axios_1.default, base_1.BASE_PATH, configuration);
        },
    };
};
exports.DecisionLogsIngressApiFp = DecisionLogsIngressApiFp;
/**
 * DecisionLogsIngressApi - factory interface
 * @export
 */
const DecisionLogsIngressApiFactory = function (configuration, basePath, axios) {
    const localVarFp = (0, exports.DecisionLogsIngressApiFp)(configuration);
    return {
        /**
         * This ingress endpoint captures OPA decision logs.
         * @summary OPA Decision Logs Ingress
         * @param {Array<object>} requestBody
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        insertOpaDecisionLogs(requestBody, options) {
            return localVarFp
                .insertOpaDecisionLogs(requestBody, options)
                .then((request) => request(axios, basePath));
        },
    };
};
exports.DecisionLogsIngressApiFactory = DecisionLogsIngressApiFactory;
/**
 * DecisionLogsIngressApi - object-oriented interface
 * @export
 * @class DecisionLogsIngressApi
 * @extends {BaseAPI}
 */
class DecisionLogsIngressApi extends base_1.BaseAPI {
    /**
     * This ingress endpoint captures OPA decision logs.
     * @summary OPA Decision Logs Ingress
     * @param {DecisionLogsIngressApiInsertOpaDecisionLogsRequest} requestParameters Request parameters.
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     * @memberof DecisionLogsIngressApi
     */
    insertOpaDecisionLogs(requestParameters, options) {
        return (0, exports.DecisionLogsIngressApiFp)(this.configuration)
            .insertOpaDecisionLogs(requestParameters.requestBody, options)
            .then((request) => request(this.axios, this.basePath));
    }
}
exports.DecisionLogsIngressApi = DecisionLogsIngressApi;
