// permitio/build/main/openapi/api/decision-logs-api.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Permit.io API
 *  Authorization as a service
 *
 * The version of the OpenAPI document: 2.0.0
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.DecisionLogsApi = exports.DecisionLogsApiFactory = exports.DecisionLogsApiFp = exports.DecisionLogsApiAxiosParamCreator = void 0;
const axios_1 = __importDefault(require("axios"));
// Some imports not used depending on template conditions
// @ts-ignore
const common_1 = require("../common");
// @ts-ignore
const base_1 = require("../base");
/**
 * DecisionLogsApi - axios parameter creator
 * @export
 */
const DecisionLogsApiAxiosParamCreator = function (configuration) {
    return {
        /**
         *
         * @summary List PDP Decisions
         * @param {string} projId Either the unique id of the project, or the URL-friendly key of the project (i.e: the \&quot;slug\&quot;).
         * @param {string} envId Either the unique id of the environment, or the URL-friendly key of the environment (i.e: the \&quot;slug\&quot;).
         * @param {string} pdpId The unique id of the pdp
         * @param {Array<string>} [users] List of user IDs to filter by
         * @param {boolean} [approved] Filter by approved decisions
         * @param {Array<string>} [resources] Filter by resources
         * @param {number} [timestampFrom] Filter by timestamp from
         * @param {number} [timestampTo] Filter by timestamp to
         * @param {number} [page] Page number of the results to fetch, starting at 1.
         * @param {number} [perPage] The number of results per page (max 100).
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        listPdpDecisionLogs: async (projId, envId, pdpId, users, approved, resources, timestampFrom, timestampTo, page, perPage, options = {}) => {
            // verify required parameter 'projId' is not null or undefined
            (0, common_1.assertParamExists)('listPdpDecisionLogs', 'projId', projId);
            // verify required parameter 'envId' is not null or undefined
            (0, common_1.assertParamExists)('listPdpDecisionLogs', 'envId', envId);
            // verify required parameter 'pdpId' is not null or undefined
            (0, common_1.assertParamExists)('listPdpDecisionLogs', 'pdpId', pdpId);
            const localVarPath = `/v2/pdps/{proj_id}/{env_id}/decision-logs/{pdp_id}`
                .replace(`{${'proj_id'}}`, encodeURIComponent(String(projId)))
                .replace(`{${'env_id'}}`, encodeURIComponent(String(envId)))
                .replace(`{${'pdp_id'}}`, encodeURIComponent(String(pdpId)));
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new URL(localVarPath, common_1.DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = Object.assign(Object.assign({ method: 'GET' }, baseOptions), options);
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            // authentication HTTPBearer required
            // http bearer authentication required
            await (0, common_1.setBearerAuthToObject)(localVarHeaderParameter, configuration);
            if (users) {
                localVarQueryParameter['users'] = users;
            }
            if (approved !== undefined) {
                localVarQueryParameter['approved'] = approved;
            }
            if (resources) {
                localVarQueryParameter['resources'] = resources;
            }
            if (timestampFrom !== undefined) {
                localVarQueryParameter['timestamp_from'] = timestampFrom;
            }
            if (timestampTo !== undefined) {
                localVarQueryParameter['timestamp_to'] = timestampTo;
            }
            if (page !== undefined) {
                localVarQueryParameter['page'] = page;
            }
            if (perPage !== undefined) {
                localVarQueryParameter['per_page'] = perPage;
            }
            (0, common_1.setSearchParams)(localVarUrlObj, localVarQueryParameter);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = Object.assign(Object.assign(Object.assign({}, localVarHeaderParameter), headersFromBaseOptions), options.headers);
            return {
                url: (0, common_1.toPathString)(localVarUrlObj),
                options: localVarRequestOptions,
            };
        },
    };
};
exports.DecisionLogsApiAxiosParamCreator = DecisionLogsApiAxiosParamCreator;
/**
 * DecisionLogsApi - functional programming interface
 * @export
 */
const DecisionLogsApiFp = function (configuration) {
    const localVarAxiosParamCreator = (0, exports.DecisionLogsApiAxiosParamCreator)(configuration);
    return {
        /**
         *
         * @summary List PDP Decisions
         * @param {string} projId Either the unique id of the project, or the URL-friendly key of the project (i.e: the \&quot;slug\&quot;).
         * @param {string} envId Either the unique id of the environment, or the URL-friendly key of the environment (i.e: the \&quot;slug\&quot;).
         * @param {string} pdpId The unique id of the pdp
         * @param {Array<string>} [users] List of user IDs to filter by
         * @param {boolean} [approved] Filter by approved decisions
         * @param {Array<string>} [resources] Filter by resources
         * @param {number} [timestampFrom] Filter by timestamp from
         * @param {number} [timestampTo] Filter by timestamp to
         * @param {number} [page] Page number of the results to fetch, starting at 1.
         * @param {number} [perPage] The number of results per page (max 100).
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        async listPdpDecisionLogs(projId, envId, pdpId, users, approved, resources, timestampFrom, timestampTo, page, perPage, options) {
            const localVarAxiosArgs = await localVarAxiosParamCreator.listPdpDecisionLogs(projId, envId, pdpId, users, approved, resources, timestampFrom, timestampTo, page, perPage, options);
            return (0, common_1.createRequestFunction)(localVarAxiosArgs, axios_1.default, base_1.BASE_PATH, configuration);
        },
    };
};
exports.DecisionLogsApiFp = DecisionLogsApiFp;
/**
 * DecisionLogsApi - factory interface
 * @export
 */
const DecisionLogsApiFactory = function (configuration, basePath, axios) {
    const localVarFp = (0, exports.DecisionLogsApiFp)(configuration);
    return {
        /**
         *
         * @summary List PDP Decisions
         * @param {string} projId Either the unique id of the project, or the URL-friendly key of the project (i.e: the \&quot;slug\&quot;).
         * @param {string} envId Either the unique id of the environment, or the URL-friendly key of the environment (i.e: the \&quot;slug\&quot;).
         * @param {string} pdpId The unique id of the pdp
         * @param {Array<string>} [users] List of user IDs to filter by
         * @param {boolean} [approved] Filter by approved decisions
         * @param {Array<string>} [resources] Filter by resources
         * @param {number} [timestampFrom] Filter by timestamp from
         * @param {number} [timestampTo] Filter by timestamp to
         * @param {number} [page] Page number of the results to fetch, starting at 1.
         * @param {number} [perPage] The number of results per page (max 100).
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        listPdpDecisionLogs(projId, envId, pdpId, users, approved, resources, timestampFrom, timestampTo, page, perPage, options) {
            return localVarFp
                .listPdpDecisionLogs(projId, envId, pdpId, users, approved, resources, timestampFrom, timestampTo, page, perPage, options)
                .then((request) => request(axios, basePath));
        },
    };
};
exports.DecisionLogsApiFactory = DecisionLogsApiFactory;
/**
 * DecisionLogsApi - object-oriented interface
 * @export
 * @class DecisionLogsApi
 * @extends {BaseAPI}
 */
class DecisionLogsApi extends base_1.BaseAPI {
    /**
     *
     * @summary List PDP Decisions
     * @param {DecisionLogsApiListPdpDecisionLogsRequest} requestParameters Request parameters.
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     * @memberof DecisionLogsApi
     */
    listPdpDecisionLogs(requestParameters, options) {
        return (0, exports.DecisionLogsApiFp)(this.configuration)
            .listPdpDecisionLogs(requestParameters.projId, requestParameters.envId, requestParameters.pdpId, requestParameters.users, requestParameters.approved, requestParameters.resources, requestParameters.timestampFrom, requestParameters.timestampTo, requestParameters.page, requestParameters.perPage, options)
            .then((request) => request(this.axios, this.basePath));
    }
}
exports.DecisionLogsApi = DecisionLogsApi;
