// permitio/build/main/openapi/api/instructions-api.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Permit.io API
 *  Authorization as a service
 *
 * The version of the OpenAPI document: 2.0.0
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.InstructionsApi = exports.InstructionsApiFactory = exports.InstructionsApiFp = exports.InstructionsApiAxiosParamCreator = void 0;
const axios_1 = __importDefault(require("axios"));
// Some imports not used depending on template conditions
// @ts-ignore
const common_1 = require("../common");
// @ts-ignore
const base_1 = require("../base");
/**
 * InstructionsApi - axios parameter creator
 * @export
 */
const InstructionsApiAxiosParamCreator = function (configuration) {
    return {
        /**
         * Lists pending organization invites
         * @summary List Language Instructions
         * @param {string} projId Either the unique id of the project, or the URL-friendly key of the project (i.e: the \&quot;slug\&quot;).
         * @param {string} envId Either the unique id of the environment, or the URL-friendly key of the environment (i.e: the \&quot;slug\&quot;).
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        listLanguageInstructions: async (projId, envId, options = {}) => {
            // verify required parameter 'projId' is not null or undefined
            (0, common_1.assertParamExists)('listLanguageInstructions', 'projId', projId);
            // verify required parameter 'envId' is not null or undefined
            (0, common_1.assertParamExists)('listLanguageInstructions', 'envId', envId);
            const localVarPath = `/v2/{proj_id}/{env_id}/get_instructions`
                .replace(`{${'proj_id'}}`, encodeURIComponent(String(projId)))
                .replace(`{${'env_id'}}`, encodeURIComponent(String(envId)));
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new URL(localVarPath, common_1.DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = Object.assign(Object.assign({ method: 'GET' }, baseOptions), options);
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            // authentication HTTPBearer required
            // http bearer authentication required
            await (0, common_1.setBearerAuthToObject)(localVarHeaderParameter, configuration);
            (0, common_1.setSearchParams)(localVarUrlObj, localVarQueryParameter);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = Object.assign(Object.assign(Object.assign({}, localVarHeaderParameter), headersFromBaseOptions), options.headers);
            return {
                url: (0, common_1.toPathString)(localVarUrlObj),
                options: localVarRequestOptions,
            };
        },
    };
};
exports.InstructionsApiAxiosParamCreator = InstructionsApiAxiosParamCreator;
/**
 * InstructionsApi - functional programming interface
 * @export
 */
const InstructionsApiFp = function (configuration) {
    const localVarAxiosParamCreator = (0, exports.InstructionsApiAxiosParamCreator)(configuration);
    return {
        /**
         * Lists pending organization invites
         * @summary List Language Instructions
         * @param {string} projId Either the unique id of the project, or the URL-friendly key of the project (i.e: the \&quot;slug\&quot;).
         * @param {string} envId Either the unique id of the environment, or the URL-friendly key of the environment (i.e: the \&quot;slug\&quot;).
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        async listLanguageInstructions(projId, envId, options) {
            const localVarAxiosArgs = await localVarAxiosParamCreator.listLanguageInstructions(projId, envId, options);
            return (0, common_1.createRequestFunction)(localVarAxiosArgs, axios_1.default, base_1.BASE_PATH, configuration);
        },
    };
};
exports.InstructionsApiFp = InstructionsApiFp;
/**
 * InstructionsApi - factory interface
 * @export
 */
const InstructionsApiFactory = function (configuration, basePath, axios) {
    const localVarFp = (0, exports.InstructionsApiFp)(configuration);
    return {
        /**
         * Lists pending organization invites
         * @summary List Language Instructions
         * @param {string} projId Either the unique id of the project, or the URL-friendly key of the project (i.e: the \&quot;slug\&quot;).
         * @param {string} envId Either the unique id of the environment, or the URL-friendly key of the environment (i.e: the \&quot;slug\&quot;).
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        listLanguageInstructions(projId, envId, options) {
            return localVarFp
                .listLanguageInstructions(projId, envId, options)
                .then((request) => request(axios, basePath));
        },
    };
};
exports.InstructionsApiFactory = InstructionsApiFactory;
/**
 * InstructionsApi - object-oriented interface
 * @export
 * @class InstructionsApi
 * @extends {BaseAPI}
 */
class InstructionsApi extends base_1.BaseAPI {
    /**
     * Lists pending organization invites
     * @summary List Language Instructions
     * @param {InstructionsApiListLanguageInstructionsRequest} requestParameters Request parameters.
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     * @memberof InstructionsApi
     */
    listLanguageInstructions(requestParameters, options) {
        return (0, exports.InstructionsApiFp)(this.configuration)
            .listLanguageInstructions(requestParameters.projId, requestParameters.envId, options)
            .then((request) => request(this.axios, this.basePath));
    }
}
exports.InstructionsApi = InstructionsApi;
