// permitio/build/main/openapi/api/members-api.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Permit.io API
 *  Authorization as a service
 *
 * The version of the OpenAPI document: 2.0.0
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.MembersApi = exports.MembersApiFactory = exports.MembersApiFp = exports.MembersApiAxiosParamCreator = void 0;
const axios_1 = __importDefault(require("axios"));
// Some imports not used depending on template conditions
// @ts-ignore
const common_1 = require("../common");
// @ts-ignore
const base_1 = require("../base");
/**
 * MembersApi - axios parameter creator
 * @export
 */
const MembersApiAxiosParamCreator = function (configuration) {
    return {
        /**
         * Create an organization member if needed, and grant it permissions.  The member can be specified either by ID (for an existing member), or by email (for either an existing member or a new one).  For a new member, an invite will be sent.
         * @summary Create Organization Members
         * @param {OrgMemberCreate} orgMemberCreate
         * @param {string} [inviterName]
         * @param {string} [inviterEmail]
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        createOrganizationMembers: async (orgMemberCreate, inviterName, inviterEmail, options = {}) => {
            // verify required parameter 'orgMemberCreate' is not null or undefined
            (0, common_1.assertParamExists)('createOrganizationMembers', 'orgMemberCreate', orgMemberCreate);
            const localVarPath = `/v2/members`;
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new URL(localVarPath, common_1.DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = Object.assign(Object.assign({ method: 'POST' }, baseOptions), options);
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            // authentication HTTPBearer required
            // http bearer authentication required
            await (0, common_1.setBearerAuthToObject)(localVarHeaderParameter, configuration);
            if (inviterName !== undefined) {
                localVarQueryParameter['inviter_name'] = inviterName;
            }
            if (inviterEmail !== undefined) {
                localVarQueryParameter['inviter_email'] = inviterEmail;
            }
            localVarHeaderParameter['Content-Type'] = 'application/json';
            (0, common_1.setSearchParams)(localVarUrlObj, localVarQueryParameter);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = Object.assign(Object.assign(Object.assign({}, localVarHeaderParameter), headersFromBaseOptions), options.headers);
            localVarRequestOptions.data = (0, common_1.serializeDataIfNeeded)(orgMemberCreate, localVarRequestOptions, configuration);
            return {
                url: (0, common_1.toPathString)(localVarUrlObj),
                options: localVarRequestOptions,
            };
        },
        /**
         * Deletes an account member matching the given id or email address. The member will be removed from the active account in permit.io.  If the member is the only member in its account (organization), returns 400 (bad request), due to nobody remains with access to the account, meaning deletion of the entire account (org). To completely remove an account, call DELETE `/orgs/{org}`.
         * @summary Delete Organization Member
         * @param {string} memberId Either the unique id (UUID) of the account member, or the email address of the account member.
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        deleteOrganizationMember: async (memberId, options = {}) => {
            // verify required parameter 'memberId' is not null or undefined
            (0, common_1.assertParamExists)('deleteOrganizationMember', 'memberId', memberId);
            const localVarPath = `/v2/members/{member_id}`.replace(`{${'member_id'}}`, encodeURIComponent(String(memberId)));
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new URL(localVarPath, common_1.DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = Object.assign(Object.assign({ method: 'DELETE' }, baseOptions), options);
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            // authentication HTTPBearer required
            // http bearer authentication required
            await (0, common_1.setBearerAuthToObject)(localVarHeaderParameter, configuration);
            (0, common_1.setSearchParams)(localVarUrlObj, localVarQueryParameter);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = Object.assign(Object.assign(Object.assign({}, localVarHeaderParameter), headersFromBaseOptions), options.headers);
            return {
                url: (0, common_1.toPathString)(localVarUrlObj),
                options: localVarRequestOptions,
            };
        },
        /**
         * Remove permissions from a member. If the last permissions a member has are removed, the member is also deleted.
         * @summary Delete Organization Permissions
         * @param {OrgMemberRemovePermissions} orgMemberRemovePermissions
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        deleteOrganizationPermissions: async (orgMemberRemovePermissions, options = {}) => {
            // verify required parameter 'orgMemberRemovePermissions' is not null or undefined
            (0, common_1.assertParamExists)('deleteOrganizationPermissions', 'orgMemberRemovePermissions', orgMemberRemovePermissions);
            const localVarPath = `/v2/members`;
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new URL(localVarPath, common_1.DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = Object.assign(Object.assign({ method: 'DELETE' }, baseOptions), options);
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            // authentication HTTPBearer required
            // http bearer authentication required
            await (0, common_1.setBearerAuthToObject)(localVarHeaderParameter, configuration);
            localVarHeaderParameter['Content-Type'] = 'application/json';
            (0, common_1.setSearchParams)(localVarUrlObj, localVarQueryParameter);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = Object.assign(Object.assign(Object.assign({}, localVarHeaderParameter), headersFromBaseOptions), options.headers);
            localVarRequestOptions.data = (0, common_1.serializeDataIfNeeded)(orgMemberRemovePermissions, localVarRequestOptions, configuration);
            return {
                url: (0, common_1.toPathString)(localVarUrlObj),
                options: localVarRequestOptions,
            };
        },
        /**
         * Gets the authenticated account member\'s details.
         * @summary Get the authenticated account member
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        getAuthenticatedMember: async (options = {}) => {
            const localVarPath = `/v2/members/me`;
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new URL(localVarPath, common_1.DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = Object.assign(Object.assign({ method: 'GET' }, baseOptions), options);
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            // authentication HTTPBearer required
            // http bearer authentication required
            await (0, common_1.setBearerAuthToObject)(localVarHeaderParameter, configuration);
            (0, common_1.setSearchParams)(localVarUrlObj, localVarQueryParameter);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = Object.assign(Object.assign(Object.assign({}, localVarHeaderParameter), headersFromBaseOptions), options.headers);
            return {
                url: (0, common_1.toPathString)(localVarUrlObj),
                options: localVarRequestOptions,
            };
        },
        /**
         * Gets a single account member by its id or email address. matching the given member, if no such member exists under the current active account (organization), returns 404.
         * @summary Get Organization Member
         * @param {string} memberId Either the unique id (UUID) of the account member, or the email address of the account member.
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        getOrganizationMember: async (memberId, options = {}) => {
            // verify required parameter 'memberId' is not null or undefined
            (0, common_1.assertParamExists)('getOrganizationMember', 'memberId', memberId);
            const localVarPath = `/v2/members/{member_id}`.replace(`{${'member_id'}}`, encodeURIComponent(String(memberId)));
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new URL(localVarPath, common_1.DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = Object.assign(Object.assign({ method: 'GET' }, baseOptions), options);
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            // authentication HTTPBearer required
            // http bearer authentication required
            await (0, common_1.setBearerAuthToObject)(localVarHeaderParameter, configuration);
            (0, common_1.setSearchParams)(localVarUrlObj, localVarQueryParameter);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = Object.assign(Object.assign(Object.assign({}, localVarHeaderParameter), headersFromBaseOptions), options.headers);
            return {
                url: (0, common_1.toPathString)(localVarUrlObj),
                options: localVarRequestOptions,
            };
        },
        /**
         * Lists all the account members that current active account has access to, optionally filtering by project or environment. The active account/organization is determined by the API Key used or by the authenticated session id.
         * @summary List Organization Members
         * @param {string} [projectId]
         * @param {string} [envId]
         * @param {number} [page] Page number of the results to fetch, starting at 1.
         * @param {number} [perPage] The number of results per page (max 100).
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        listOrganizationMembers: async (projectId, envId, page, perPage, options = {}) => {
            const localVarPath = `/v2/members`;
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new URL(localVarPath, common_1.DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = Object.assign(Object.assign({ method: 'GET' }, baseOptions), options);
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            // authentication HTTPBearer required
            // http bearer authentication required
            await (0, common_1.setBearerAuthToObject)(localVarHeaderParameter, configuration);
            if (projectId !== undefined) {
                localVarQueryParameter['project_id'] = projectId;
            }
            if (envId !== undefined) {
                localVarQueryParameter['env_id'] = envId;
            }
            if (page !== undefined) {
                localVarQueryParameter['page'] = page;
            }
            if (perPage !== undefined) {
                localVarQueryParameter['per_page'] = perPage;
            }
            (0, common_1.setSearchParams)(localVarUrlObj, localVarQueryParameter);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = Object.assign(Object.assign(Object.assign({}, localVarHeaderParameter), headersFromBaseOptions), options.headers);
            return {
                url: (0, common_1.toPathString)(localVarUrlObj),
                options: localVarRequestOptions,
            };
        },
        /**
         * Updates an account member\'s settings.
         * @summary Update Organization Member
         * @param {string} memberId Either the unique id (UUID) of the account member, or the email address of the account member.
         * @param {OrgMemberUpdate} orgMemberUpdate
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        updateOrganizationMember: async (memberId, orgMemberUpdate, options = {}) => {
            // verify required parameter 'memberId' is not null or undefined
            (0, common_1.assertParamExists)('updateOrganizationMember', 'memberId', memberId);
            // verify required parameter 'orgMemberUpdate' is not null or undefined
            (0, common_1.assertParamExists)('updateOrganizationMember', 'orgMemberUpdate', orgMemberUpdate);
            const localVarPath = `/v2/members/{member_id}`.replace(`{${'member_id'}}`, encodeURIComponent(String(memberId)));
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new URL(localVarPath, common_1.DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = Object.assign(Object.assign({ method: 'PATCH' }, baseOptions), options);
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            // authentication HTTPBearer required
            // http bearer authentication required
            await (0, common_1.setBearerAuthToObject)(localVarHeaderParameter, configuration);
            localVarHeaderParameter['Content-Type'] = 'application/json';
            (0, common_1.setSearchParams)(localVarUrlObj, localVarQueryParameter);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = Object.assign(Object.assign(Object.assign({}, localVarHeaderParameter), headersFromBaseOptions), options.headers);
            localVarRequestOptions.data = (0, common_1.serializeDataIfNeeded)(orgMemberUpdate, localVarRequestOptions, configuration);
            return {
                url: (0, common_1.toPathString)(localVarUrlObj),
                options: localVarRequestOptions,
            };
        },
    };
};
exports.MembersApiAxiosParamCreator = MembersApiAxiosParamCreator;
/**
 * MembersApi - functional programming interface
 * @export
 */
const MembersApiFp = function (configuration) {
    const localVarAxiosParamCreator = (0, exports.MembersApiAxiosParamCreator)(configuration);
    return {
        /**
         * Create an organization member if needed, and grant it permissions.  The member can be specified either by ID (for an existing member), or by email (for either an existing member or a new one).  For a new member, an invite will be sent.
         * @summary Create Organization Members
         * @param {OrgMemberCreate} orgMemberCreate
         * @param {string} [inviterName]
         * @param {string} [inviterEmail]
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        async createOrganizationMembers(orgMemberCreate, inviterName, inviterEmail, options) {
            const localVarAxiosArgs = await localVarAxiosParamCreator.createOrganizationMembers(orgMemberCreate, inviterName, inviterEmail, options);
            return (0, common_1.createRequestFunction)(localVarAxiosArgs, axios_1.default, base_1.BASE_PATH, configuration);
        },
        /**
         * Deletes an account member matching the given id or email address. The member will be removed from the active account in permit.io.  If the member is the only member in its account (organization), returns 400 (bad request), due to nobody remains with access to the account, meaning deletion of the entire account (org). To completely remove an account, call DELETE `/orgs/{org}`.
         * @summary Delete Organization Member
         * @param {string} memberId Either the unique id (UUID) of the account member, or the email address of the account member.
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        async deleteOrganizationMember(memberId, options) {
            const localVarAxiosArgs = await localVarAxiosParamCreator.deleteOrganizationMember(memberId, options);
            return (0, common_1.createRequestFunction)(localVarAxiosArgs, axios_1.default, base_1.BASE_PATH, configuration);
        },
        /**
         * Remove permissions from a member. If the last permissions a member has are removed, the member is also deleted.
         * @summary Delete Organization Permissions
         * @param {OrgMemberRemovePermissions} orgMemberRemovePermissions
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        async deleteOrganizationPermissions(orgMemberRemovePermissions, options) {
            const localVarAxiosArgs = await localVarAxiosParamCreator.deleteOrganizationPermissions(orgMemberRemovePermissions, options);
            return (0, common_1.createRequestFunction)(localVarAxiosArgs, axios_1.default, base_1.BASE_PATH, configuration);
        },
        /**
         * Gets the authenticated account member\'s details.
         * @summary Get the authenticated account member
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        async getAuthenticatedMember(options) {
            const localVarAxiosArgs = await localVarAxiosParamCreator.getAuthenticatedMember(options);
            return (0, common_1.createRequestFunction)(localVarAxiosArgs, axios_1.default, base_1.BASE_PATH, configuration);
        },
        /**
         * Gets a single account member by its id or email address. matching the given member, if no such member exists under the current active account (organization), returns 404.
         * @summary Get Organization Member
         * @param {string} memberId Either the unique id (UUID) of the account member, or the email address of the account member.
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        async getOrganizationMember(memberId, options) {
            const localVarAxiosArgs = await localVarAxiosParamCreator.getOrganizationMember(memberId, options);
            return (0, common_1.createRequestFunction)(localVarAxiosArgs, axios_1.default, base_1.BASE_PATH, configuration);
        },
        /**
         * Lists all the account members that current active account has access to, optionally filtering by project or environment. The active account/organization is determined by the API Key used or by the authenticated session id.
         * @summary List Organization Members
         * @param {string} [projectId]
         * @param {string} [envId]
         * @param {number} [page] Page number of the results to fetch, starting at 1.
         * @param {number} [perPage] The number of results per page (max 100).
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        async listOrganizationMembers(projectId, envId, page, perPage, options) {
            const localVarAxiosArgs = await localVarAxiosParamCreator.listOrganizationMembers(projectId, envId, page, perPage, options);
            return (0, common_1.createRequestFunction)(localVarAxiosArgs, axios_1.default, base_1.BASE_PATH, configuration);
        },
        /**
         * Updates an account member\'s settings.
         * @summary Update Organization Member
         * @param {string} memberId Either the unique id (UUID) of the account member, or the email address of the account member.
         * @param {OrgMemberUpdate} orgMemberUpdate
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        async updateOrganizationMember(memberId, orgMemberUpdate, options) {
            const localVarAxiosArgs = await localVarAxiosParamCreator.updateOrganizationMember(memberId, orgMemberUpdate, options);
            return (0, common_1.createRequestFunction)(localVarAxiosArgs, axios_1.default, base_1.BASE_PATH, configuration);
        },
    };
};
exports.MembersApiFp = MembersApiFp;
/**
 * MembersApi - factory interface
 * @export
 */
const MembersApiFactory = function (configuration, basePath, axios) {
    const localVarFp = (0, exports.MembersApiFp)(configuration);
    return {
        /**
         * Create an organization member if needed, and grant it permissions.  The member can be specified either by ID (for an existing member), or by email (for either an existing member or a new one).  For a new member, an invite will be sent.
         * @summary Create Organization Members
         * @param {OrgMemberCreate} orgMemberCreate
         * @param {string} [inviterName]
         * @param {string} [inviterEmail]
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        createOrganizationMembers(orgMemberCreate, inviterName, inviterEmail, options) {
            return localVarFp
                .createOrganizationMembers(orgMemberCreate, inviterName, inviterEmail, options)
                .then((request) => request(axios, basePath));
        },
        /**
         * Deletes an account member matching the given id or email address. The member will be removed from the active account in permit.io.  If the member is the only member in its account (organization), returns 400 (bad request), due to nobody remains with access to the account, meaning deletion of the entire account (org). To completely remove an account, call DELETE `/orgs/{org}`.
         * @summary Delete Organization Member
         * @param {string} memberId Either the unique id (UUID) of the account member, or the email address of the account member.
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        deleteOrganizationMember(memberId, options) {
            return localVarFp
                .deleteOrganizationMember(memberId, options)
                .then((request) => request(axios, basePath));
        },
        /**
         * Remove permissions from a member. If the last permissions a member has are removed, the member is also deleted.
         * @summary Delete Organization Permissions
         * @param {OrgMemberRemovePermissions} orgMemberRemovePermissions
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        deleteOrganizationPermissions(orgMemberRemovePermissions, options) {
            return localVarFp
                .deleteOrganizationPermissions(orgMemberRemovePermissions, options)
                .then((request) => request(axios, basePath));
        },
        /**
         * Gets the authenticated account member\'s details.
         * @summary Get the authenticated account member
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        getAuthenticatedMember(options) {
            return localVarFp.getAuthenticatedMember(options).then((request) => request(axios, basePath));
        },
        /**
         * Gets a single account member by its id or email address. matching the given member, if no such member exists under the current active account (organization), returns 404.
         * @summary Get Organization Member
         * @param {string} memberId Either the unique id (UUID) of the account member, or the email address of the account member.
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        getOrganizationMember(memberId, options) {
            return localVarFp
                .getOrganizationMember(memberId, options)
                .then((request) => request(axios, basePath));
        },
        /**
         * Lists all the account members that current active account has access to, optionally filtering by project or environment. The active account/organization is determined by the API Key used or by the authenticated session id.
         * @summary List Organization Members
         * @param {string} [projectId]
         * @param {string} [envId]
         * @param {number} [page] Page number of the results to fetch, starting at 1.
         * @param {number} [perPage] The number of results per page (max 100).
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        listOrganizationMembers(projectId, envId, page, perPage, options) {
            return localVarFp
                .listOrganizationMembers(projectId, envId, page, perPage, options)
                .then((request) => request(axios, basePath));
        },
        /**
         * Updates an account member\'s settings.
         * @summary Update Organization Member
         * @param {string} memberId Either the unique id (UUID) of the account member, or the email address of the account member.
         * @param {OrgMemberUpdate} orgMemberUpdate
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        updateOrganizationMember(memberId, orgMemberUpdate, options) {
            return localVarFp
                .updateOrganizationMember(memberId, orgMemberUpdate, options)
                .then((request) => request(axios, basePath));
        },
    };
};
exports.MembersApiFactory = MembersApiFactory;
/**
 * MembersApi - object-oriented interface
 * @export
 * @class MembersApi
 * @extends {BaseAPI}
 */
class MembersApi extends base_1.BaseAPI {
    /**
     * Create an organization member if needed, and grant it permissions.  The member can be specified either by ID (for an existing member), or by email (for either an existing member or a new one).  For a new member, an invite will be sent.
     * @summary Create Organization Members
     * @param {MembersApiCreateOrganizationMembersRequest} requestParameters Request parameters.
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     * @memberof MembersApi
     */
    createOrganizationMembers(requestParameters, options) {
        return (0, exports.MembersApiFp)(this.configuration)
            .createOrganizationMembers(requestParameters.orgMemberCreate, requestParameters.inviterName, requestParameters.inviterEmail, options)
            .then((request) => request(this.axios, this.basePath));
    }
    /**
     * Deletes an account member matching the given id or email address. The member will be removed from the active account in permit.io.  If the member is the only member in its account (organization), returns 400 (bad request), due to nobody remains with access to the account, meaning deletion of the entire account (org). To completely remove an account, call DELETE `/orgs/{org}`.
     * @summary Delete Organization Member
     * @param {MembersApiDeleteOrganizationMemberRequest} requestParameters Request parameters.
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     * @memberof MembersApi
     */
    deleteOrganizationMember(requestParameters, options) {
        return (0, exports.MembersApiFp)(this.configuration)
            .deleteOrganizationMember(requestParameters.memberId, options)
            .then((request) => request(this.axios, this.basePath));
    }
    /**
     * Remove permissions from a member. If the last permissions a member has are removed, the member is also deleted.
     * @summary Delete Organization Permissions
     * @param {MembersApiDeleteOrganizationPermissionsRequest} requestParameters Request parameters.
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     * @memberof MembersApi
     */
    deleteOrganizationPermissions(requestParameters, options) {
        return (0, exports.MembersApiFp)(this.configuration)
            .deleteOrganizationPermissions(requestParameters.orgMemberRemovePermissions, options)
            .then((request) => request(this.axios, this.basePath));
    }
    /**
     * Gets the authenticated account member\'s details.
     * @summary Get the authenticated account member
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     * @memberof MembersApi
     */
    getAuthenticatedMember(options) {
        return (0, exports.MembersApiFp)(this.configuration)
            .getAuthenticatedMember(options)
            .then((request) => request(this.axios, this.basePath));
    }
    /**
     * Gets a single account member by its id or email address. matching the given member, if no such member exists under the current active account (organization), returns 404.
     * @summary Get Organization Member
     * @param {MembersApiGetOrganizationMemberRequest} requestParameters Request parameters.
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     * @memberof MembersApi
     */
    getOrganizationMember(requestParameters, options) {
        return (0, exports.MembersApiFp)(this.configuration)
            .getOrganizationMember(requestParameters.memberId, options)
            .then((request) => request(this.axios, this.basePath));
    }
    /**
     * Lists all the account members that current active account has access to, optionally filtering by project or environment. The active account/organization is determined by the API Key used or by the authenticated session id.
     * @summary List Organization Members
     * @param {MembersApiListOrganizationMembersRequest} requestParameters Request parameters.
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     * @memberof MembersApi
     */
    listOrganizationMembers(requestParameters = {}, options) {
        return (0, exports.MembersApiFp)(this.configuration)
            .listOrganizationMembers(requestParameters.projectId, requestParameters.envId, requestParameters.page, requestParameters.perPage, options)
            .then((request) => request(this.axios, this.basePath));
    }
    /**
     * Updates an account member\'s settings.
     * @summary Update Organization Member
     * @param {MembersApiUpdateOrganizationMemberRequest} requestParameters Request parameters.
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     * @memberof MembersApi
     */
    updateOrganizationMember(requestParameters, options) {
        return (0, exports.MembersApiFp)(this.configuration)
            .updateOrganizationMember(requestParameters.memberId, requestParameters.orgMemberUpdate, options)
            .then((request) => request(this.axios, this.basePath));
    }
}
exports.MembersApi = MembersApi;
