// permitio/build/main/openapi/api/policy-synchronizer-administration-api.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Permit.io API
 *  Authorization as a service
 *
 * The version of the OpenAPI document: 2.0.0
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.PolicySynchronizerAdministrationApi = exports.PolicySynchronizerAdministrationApiFactory = exports.PolicySynchronizerAdministrationApiFp = exports.PolicySynchronizerAdministrationApiAxiosParamCreator = void 0;
const axios_1 = __importDefault(require("axios"));
// Some imports not used depending on template conditions
// @ts-ignore
const common_1 = require("../common");
// @ts-ignore
const base_1 = require("../base");
/**
 * PolicySynchronizerAdministrationApi - axios parameter creator
 * @export
 */
const PolicySynchronizerAdministrationApiAxiosParamCreator = function (configuration) {
    return {
        /**
         * Push policy synchronizer update to all environments in the vc organization
         * @summary Regenerate environments
         * @param {Message} message
         * @param {number} [chunkSize] Number of environments to regenerate in a chunk
         * @param {number} [chunkCooldown] Cooldown between chunks in seconds
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        policySynchronizerAdministration: async (message, chunkSize, chunkCooldown, options = {}) => {
            // verify required parameter 'message' is not null or undefined
            (0, common_1.assertParamExists)('policySynchronizerAdministration', 'message', message);
            const localVarPath = `/v2/internal_sysadmin/policy-sync/regenerate-environments`;
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new URL(localVarPath, common_1.DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = Object.assign(Object.assign({ method: 'PUT' }, baseOptions), options);
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            // authentication HTTPBearer required
            // http bearer authentication required
            await (0, common_1.setBearerAuthToObject)(localVarHeaderParameter, configuration);
            if (chunkSize !== undefined) {
                localVarQueryParameter['chunk_size'] = chunkSize;
            }
            if (chunkCooldown !== undefined) {
                localVarQueryParameter['chunk_cooldown'] = chunkCooldown;
            }
            localVarHeaderParameter['Content-Type'] = 'application/json';
            (0, common_1.setSearchParams)(localVarUrlObj, localVarQueryParameter);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = Object.assign(Object.assign(Object.assign({}, localVarHeaderParameter), headersFromBaseOptions), options.headers);
            localVarRequestOptions.data = (0, common_1.serializeDataIfNeeded)(message, localVarRequestOptions, configuration);
            return {
                url: (0, common_1.toPathString)(localVarUrlObj),
                options: localVarRequestOptions,
            };
        },
    };
};
exports.PolicySynchronizerAdministrationApiAxiosParamCreator = PolicySynchronizerAdministrationApiAxiosParamCreator;
/**
 * PolicySynchronizerAdministrationApi - functional programming interface
 * @export
 */
const PolicySynchronizerAdministrationApiFp = function (configuration) {
    const localVarAxiosParamCreator = (0, exports.PolicySynchronizerAdministrationApiAxiosParamCreator)(configuration);
    return {
        /**
         * Push policy synchronizer update to all environments in the vc organization
         * @summary Regenerate environments
         * @param {Message} message
         * @param {number} [chunkSize] Number of environments to regenerate in a chunk
         * @param {number} [chunkCooldown] Cooldown between chunks in seconds
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        async policySynchronizerAdministration(message, chunkSize, chunkCooldown, options) {
            const localVarAxiosArgs = await localVarAxiosParamCreator.policySynchronizerAdministration(message, chunkSize, chunkCooldown, options);
            return (0, common_1.createRequestFunction)(localVarAxiosArgs, axios_1.default, base_1.BASE_PATH, configuration);
        },
    };
};
exports.PolicySynchronizerAdministrationApiFp = PolicySynchronizerAdministrationApiFp;
/**
 * PolicySynchronizerAdministrationApi - factory interface
 * @export
 */
const PolicySynchronizerAdministrationApiFactory = function (configuration, basePath, axios) {
    const localVarFp = (0, exports.PolicySynchronizerAdministrationApiFp)(configuration);
    return {
        /**
         * Push policy synchronizer update to all environments in the vc organization
         * @summary Regenerate environments
         * @param {Message} message
         * @param {number} [chunkSize] Number of environments to regenerate in a chunk
         * @param {number} [chunkCooldown] Cooldown between chunks in seconds
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        policySynchronizerAdministration(message, chunkSize, chunkCooldown, options) {
            return localVarFp.policySynchronizerAdministration(message, chunkSize, chunkCooldown, options).then((request) => request(axios, basePath));
        },
    };
};
exports.PolicySynchronizerAdministrationApiFactory = PolicySynchronizerAdministrationApiFactory;
/**
 * PolicySynchronizerAdministrationApi - object-oriented interface
 * @export
 * @class PolicySynchronizerAdministrationApi
 * @extends {BaseAPI}
 */
class PolicySynchronizerAdministrationApi extends base_1.BaseAPI {
    /**
     * Push policy synchronizer update to all environments in the vc organization
     * @summary Regenerate environments
     * @param {PolicySynchronizerAdministrationApiPolicySynchronizerAdministrationRequest} requestParameters Request parameters.
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     * @memberof PolicySynchronizerAdministrationApi
     */
    policySynchronizerAdministration(requestParameters, options) {
        return (0, exports.PolicySynchronizerAdministrationApiFp)(this.configuration).policySynchronizerAdministration(requestParameters.message, requestParameters.chunkSize, requestParameters.chunkCooldown, options).then((request) => request(this.axios, this.basePath));
    }
}
exports.PolicySynchronizerAdministrationApi = PolicySynchronizerAdministrationApi;
