// permitio/build/main/openapi/api/proxy-config-api.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Permit.io API
 *  Authorization as a service
 *
 * The version of the OpenAPI document: 2.0.0
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.ProxyConfigApi = exports.ProxyConfigApiFactory = exports.ProxyConfigApiFp = exports.ProxyConfigApiAxiosParamCreator = void 0;
const axios_1 = __importDefault(require("axios"));
// Some imports not used depending on template conditions
// @ts-ignore
const common_1 = require("../common");
// @ts-ignore
const base_1 = require("../base");
/**
 * ProxyConfigApi - axios parameter creator
 * @export
 */
const ProxyConfigApiAxiosParamCreator = function (configuration) {
    return {
        /**
         * Creates a new proxy config inside the Permit.io system.  If the proxy config is already created: will return 200 instead of 201, and will return the existing proxy config object in the response body.
         * @summary Create Proxy Config
         * @param {string} projId Either the unique id of the project, or the URL-friendly key of the project (i.e: the \&quot;slug\&quot;).
         * @param {string} envId Either the unique id of the environment, or the URL-friendly key of the environment (i.e: the \&quot;slug\&quot;).
         * @param {ProxyConfigCreate} proxyConfigCreate
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        createProxyConfig: async (projId, envId, proxyConfigCreate, options = {}) => {
            // verify required parameter 'projId' is not null or undefined
            (0, common_1.assertParamExists)('createProxyConfig', 'projId', projId);
            // verify required parameter 'envId' is not null or undefined
            (0, common_1.assertParamExists)('createProxyConfig', 'envId', envId);
            // verify required parameter 'proxyConfigCreate' is not null or undefined
            (0, common_1.assertParamExists)('createProxyConfig', 'proxyConfigCreate', proxyConfigCreate);
            const localVarPath = `/v2/facts/{proj_id}/{env_id}/proxy_configs`
                .replace(`{${'proj_id'}}`, encodeURIComponent(String(projId)))
                .replace(`{${'env_id'}}`, encodeURIComponent(String(envId)));
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new URL(localVarPath, common_1.DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = Object.assign(Object.assign({ method: 'POST' }, baseOptions), options);
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            // authentication HTTPBearer required
            // http bearer authentication required
            await (0, common_1.setBearerAuthToObject)(localVarHeaderParameter, configuration);
            localVarHeaderParameter['Content-Type'] = 'application/json';
            (0, common_1.setSearchParams)(localVarUrlObj, localVarQueryParameter);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = Object.assign(Object.assign(Object.assign({}, localVarHeaderParameter), headersFromBaseOptions), options.headers);
            localVarRequestOptions.data = (0, common_1.serializeDataIfNeeded)(proxyConfigCreate, localVarRequestOptions, configuration);
            return {
                url: (0, common_1.toPathString)(localVarUrlObj),
                options: localVarRequestOptions,
            };
        },
        /**
         * Deletes the proxy config and all its related data.
         * @summary Delete Proxy Config
         * @param {string} projId Either the unique id of the project, or the URL-friendly key of the project (i.e: the \&quot;slug\&quot;).
         * @param {string} envId Either the unique id of the environment, or the URL-friendly key of the environment (i.e: the \&quot;slug\&quot;).
         * @param {string} proxyConfigId Either the unique id of the proxy config, or the URL-friendly key of the proxy config (i.e: the \&quot;slug\&quot;).
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        deleteProxyConfig: async (projId, envId, proxyConfigId, options = {}) => {
            // verify required parameter 'projId' is not null or undefined
            (0, common_1.assertParamExists)('deleteProxyConfig', 'projId', projId);
            // verify required parameter 'envId' is not null or undefined
            (0, common_1.assertParamExists)('deleteProxyConfig', 'envId', envId);
            // verify required parameter 'proxyConfigId' is not null or undefined
            (0, common_1.assertParamExists)('deleteProxyConfig', 'proxyConfigId', proxyConfigId);
            const localVarPath = `/v2/facts/{proj_id}/{env_id}/proxy_configs/{proxy_config_id}`
                .replace(`{${'proj_id'}}`, encodeURIComponent(String(projId)))
                .replace(`{${'env_id'}}`, encodeURIComponent(String(envId)))
                .replace(`{${'proxy_config_id'}}`, encodeURIComponent(String(proxyConfigId)));
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new URL(localVarPath, common_1.DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = Object.assign(Object.assign({ method: 'DELETE' }, baseOptions), options);
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            // authentication HTTPBearer required
            // http bearer authentication required
            await (0, common_1.setBearerAuthToObject)(localVarHeaderParameter, configuration);
            (0, common_1.setSearchParams)(localVarUrlObj, localVarQueryParameter);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = Object.assign(Object.assign(Object.assign({}, localVarHeaderParameter), headersFromBaseOptions), options.headers);
            return {
                url: (0, common_1.toPathString)(localVarUrlObj),
                options: localVarRequestOptions,
            };
        },
        /**
         * Gets a proxy config, if such proxy config exists. Otherwise returns 404.
         * @summary Get Proxy Config
         * @param {string} projId Either the unique id of the project, or the URL-friendly key of the project (i.e: the \&quot;slug\&quot;).
         * @param {string} envId Either the unique id of the environment, or the URL-friendly key of the environment (i.e: the \&quot;slug\&quot;).
         * @param {string} proxyConfigId Either the unique id of the proxy config, or the URL-friendly key of the proxy config (i.e: the \&quot;slug\&quot;).
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        getProxyConfig: async (projId, envId, proxyConfigId, options = {}) => {
            // verify required parameter 'projId' is not null or undefined
            (0, common_1.assertParamExists)('getProxyConfig', 'projId', projId);
            // verify required parameter 'envId' is not null or undefined
            (0, common_1.assertParamExists)('getProxyConfig', 'envId', envId);
            // verify required parameter 'proxyConfigId' is not null or undefined
            (0, common_1.assertParamExists)('getProxyConfig', 'proxyConfigId', proxyConfigId);
            const localVarPath = `/v2/facts/{proj_id}/{env_id}/proxy_configs/{proxy_config_id}`
                .replace(`{${'proj_id'}}`, encodeURIComponent(String(projId)))
                .replace(`{${'env_id'}}`, encodeURIComponent(String(envId)))
                .replace(`{${'proxy_config_id'}}`, encodeURIComponent(String(proxyConfigId)));
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new URL(localVarPath, common_1.DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = Object.assign(Object.assign({ method: 'GET' }, baseOptions), options);
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            // authentication HTTPBearer required
            // http bearer authentication required
            await (0, common_1.setBearerAuthToObject)(localVarHeaderParameter, configuration);
            (0, common_1.setSearchParams)(localVarUrlObj, localVarQueryParameter);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = Object.assign(Object.assign(Object.assign({}, localVarHeaderParameter), headersFromBaseOptions), options.headers);
            return {
                url: (0, common_1.toPathString)(localVarUrlObj),
                options: localVarRequestOptions,
            };
        },
        /**
         * Lists all the proxy configs defined within an environment.
         * @summary List Proxy Configs
         * @param {string} projId Either the unique id of the project, or the URL-friendly key of the project (i.e: the \&quot;slug\&quot;).
         * @param {string} envId Either the unique id of the environment, or the URL-friendly key of the environment (i.e: the \&quot;slug\&quot;).
         * @param {number} [page] Page number of the results to fetch, starting at 1.
         * @param {number} [perPage] The number of results per page (max 100).
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        listProxyConfigs: async (projId, envId, page, perPage, options = {}) => {
            // verify required parameter 'projId' is not null or undefined
            (0, common_1.assertParamExists)('listProxyConfigs', 'projId', projId);
            // verify required parameter 'envId' is not null or undefined
            (0, common_1.assertParamExists)('listProxyConfigs', 'envId', envId);
            const localVarPath = `/v2/facts/{proj_id}/{env_id}/proxy_configs`
                .replace(`{${'proj_id'}}`, encodeURIComponent(String(projId)))
                .replace(`{${'env_id'}}`, encodeURIComponent(String(envId)));
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new URL(localVarPath, common_1.DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = Object.assign(Object.assign({ method: 'GET' }, baseOptions), options);
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            // authentication HTTPBearer required
            // http bearer authentication required
            await (0, common_1.setBearerAuthToObject)(localVarHeaderParameter, configuration);
            if (page !== undefined) {
                localVarQueryParameter['page'] = page;
            }
            if (perPage !== undefined) {
                localVarQueryParameter['per_page'] = perPage;
            }
            (0, common_1.setSearchParams)(localVarUrlObj, localVarQueryParameter);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = Object.assign(Object.assign(Object.assign({}, localVarHeaderParameter), headersFromBaseOptions), options.headers);
            return {
                url: (0, common_1.toPathString)(localVarUrlObj),
                options: localVarRequestOptions,
            };
        },
        /**
         * Partially updates the proxy config definition. Fields that will be provided will be completely overwritten.
         * @summary Update Proxy Config
         * @param {string} projId Either the unique id of the project, or the URL-friendly key of the project (i.e: the \&quot;slug\&quot;).
         * @param {string} envId Either the unique id of the environment, or the URL-friendly key of the environment (i.e: the \&quot;slug\&quot;).
         * @param {string} proxyConfigId Either the unique id of the proxy config, or the URL-friendly key of the proxy config (i.e: the \&quot;slug\&quot;).
         * @param {ProxyConfigUpdate} proxyConfigUpdate
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        updateProxyConfig: async (projId, envId, proxyConfigId, proxyConfigUpdate, options = {}) => {
            // verify required parameter 'projId' is not null or undefined
            (0, common_1.assertParamExists)('updateProxyConfig', 'projId', projId);
            // verify required parameter 'envId' is not null or undefined
            (0, common_1.assertParamExists)('updateProxyConfig', 'envId', envId);
            // verify required parameter 'proxyConfigId' is not null or undefined
            (0, common_1.assertParamExists)('updateProxyConfig', 'proxyConfigId', proxyConfigId);
            // verify required parameter 'proxyConfigUpdate' is not null or undefined
            (0, common_1.assertParamExists)('updateProxyConfig', 'proxyConfigUpdate', proxyConfigUpdate);
            const localVarPath = `/v2/facts/{proj_id}/{env_id}/proxy_configs/{proxy_config_id}`
                .replace(`{${'proj_id'}}`, encodeURIComponent(String(projId)))
                .replace(`{${'env_id'}}`, encodeURIComponent(String(envId)))
                .replace(`{${'proxy_config_id'}}`, encodeURIComponent(String(proxyConfigId)));
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new URL(localVarPath, common_1.DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = Object.assign(Object.assign({ method: 'PATCH' }, baseOptions), options);
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            // authentication HTTPBearer required
            // http bearer authentication required
            await (0, common_1.setBearerAuthToObject)(localVarHeaderParameter, configuration);
            localVarHeaderParameter['Content-Type'] = 'application/json';
            (0, common_1.setSearchParams)(localVarUrlObj, localVarQueryParameter);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = Object.assign(Object.assign(Object.assign({}, localVarHeaderParameter), headersFromBaseOptions), options.headers);
            localVarRequestOptions.data = (0, common_1.serializeDataIfNeeded)(proxyConfigUpdate, localVarRequestOptions, configuration);
            return {
                url: (0, common_1.toPathString)(localVarUrlObj),
                options: localVarRequestOptions,
            };
        },
    };
};
exports.ProxyConfigApiAxiosParamCreator = ProxyConfigApiAxiosParamCreator;
/**
 * ProxyConfigApi - functional programming interface
 * @export
 */
const ProxyConfigApiFp = function (configuration) {
    const localVarAxiosParamCreator = (0, exports.ProxyConfigApiAxiosParamCreator)(configuration);
    return {
        /**
         * Creates a new proxy config inside the Permit.io system.  If the proxy config is already created: will return 200 instead of 201, and will return the existing proxy config object in the response body.
         * @summary Create Proxy Config
         * @param {string} projId Either the unique id of the project, or the URL-friendly key of the project (i.e: the \&quot;slug\&quot;).
         * @param {string} envId Either the unique id of the environment, or the URL-friendly key of the environment (i.e: the \&quot;slug\&quot;).
         * @param {ProxyConfigCreate} proxyConfigCreate
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        async createProxyConfig(projId, envId, proxyConfigCreate, options) {
            const localVarAxiosArgs = await localVarAxiosParamCreator.createProxyConfig(projId, envId, proxyConfigCreate, options);
            return (0, common_1.createRequestFunction)(localVarAxiosArgs, axios_1.default, base_1.BASE_PATH, configuration);
        },
        /**
         * Deletes the proxy config and all its related data.
         * @summary Delete Proxy Config
         * @param {string} projId Either the unique id of the project, or the URL-friendly key of the project (i.e: the \&quot;slug\&quot;).
         * @param {string} envId Either the unique id of the environment, or the URL-friendly key of the environment (i.e: the \&quot;slug\&quot;).
         * @param {string} proxyConfigId Either the unique id of the proxy config, or the URL-friendly key of the proxy config (i.e: the \&quot;slug\&quot;).
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        async deleteProxyConfig(projId, envId, proxyConfigId, options) {
            const localVarAxiosArgs = await localVarAxiosParamCreator.deleteProxyConfig(projId, envId, proxyConfigId, options);
            return (0, common_1.createRequestFunction)(localVarAxiosArgs, axios_1.default, base_1.BASE_PATH, configuration);
        },
        /**
         * Gets a proxy config, if such proxy config exists. Otherwise returns 404.
         * @summary Get Proxy Config
         * @param {string} projId Either the unique id of the project, or the URL-friendly key of the project (i.e: the \&quot;slug\&quot;).
         * @param {string} envId Either the unique id of the environment, or the URL-friendly key of the environment (i.e: the \&quot;slug\&quot;).
         * @param {string} proxyConfigId Either the unique id of the proxy config, or the URL-friendly key of the proxy config (i.e: the \&quot;slug\&quot;).
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        async getProxyConfig(projId, envId, proxyConfigId, options) {
            const localVarAxiosArgs = await localVarAxiosParamCreator.getProxyConfig(projId, envId, proxyConfigId, options);
            return (0, common_1.createRequestFunction)(localVarAxiosArgs, axios_1.default, base_1.BASE_PATH, configuration);
        },
        /**
         * Lists all the proxy configs defined within an environment.
         * @summary List Proxy Configs
         * @param {string} projId Either the unique id of the project, or the URL-friendly key of the project (i.e: the \&quot;slug\&quot;).
         * @param {string} envId Either the unique id of the environment, or the URL-friendly key of the environment (i.e: the \&quot;slug\&quot;).
         * @param {number} [page] Page number of the results to fetch, starting at 1.
         * @param {number} [perPage] The number of results per page (max 100).
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        async listProxyConfigs(projId, envId, page, perPage, options) {
            const localVarAxiosArgs = await localVarAxiosParamCreator.listProxyConfigs(projId, envId, page, perPage, options);
            return (0, common_1.createRequestFunction)(localVarAxiosArgs, axios_1.default, base_1.BASE_PATH, configuration);
        },
        /**
         * Partially updates the proxy config definition. Fields that will be provided will be completely overwritten.
         * @summary Update Proxy Config
         * @param {string} projId Either the unique id of the project, or the URL-friendly key of the project (i.e: the \&quot;slug\&quot;).
         * @param {string} envId Either the unique id of the environment, or the URL-friendly key of the environment (i.e: the \&quot;slug\&quot;).
         * @param {string} proxyConfigId Either the unique id of the proxy config, or the URL-friendly key of the proxy config (i.e: the \&quot;slug\&quot;).
         * @param {ProxyConfigUpdate} proxyConfigUpdate
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        async updateProxyConfig(projId, envId, proxyConfigId, proxyConfigUpdate, options) {
            const localVarAxiosArgs = await localVarAxiosParamCreator.updateProxyConfig(projId, envId, proxyConfigId, proxyConfigUpdate, options);
            return (0, common_1.createRequestFunction)(localVarAxiosArgs, axios_1.default, base_1.BASE_PATH, configuration);
        },
    };
};
exports.ProxyConfigApiFp = ProxyConfigApiFp;
/**
 * ProxyConfigApi - factory interface
 * @export
 */
const ProxyConfigApiFactory = function (configuration, basePath, axios) {
    const localVarFp = (0, exports.ProxyConfigApiFp)(configuration);
    return {
        /**
         * Creates a new proxy config inside the Permit.io system.  If the proxy config is already created: will return 200 instead of 201, and will return the existing proxy config object in the response body.
         * @summary Create Proxy Config
         * @param {string} projId Either the unique id of the project, or the URL-friendly key of the project (i.e: the \&quot;slug\&quot;).
         * @param {string} envId Either the unique id of the environment, or the URL-friendly key of the environment (i.e: the \&quot;slug\&quot;).
         * @param {ProxyConfigCreate} proxyConfigCreate
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        createProxyConfig(projId, envId, proxyConfigCreate, options) {
            return localVarFp
                .createProxyConfig(projId, envId, proxyConfigCreate, options)
                .then((request) => request(axios, basePath));
        },
        /**
         * Deletes the proxy config and all its related data.
         * @summary Delete Proxy Config
         * @param {string} projId Either the unique id of the project, or the URL-friendly key of the project (i.e: the \&quot;slug\&quot;).
         * @param {string} envId Either the unique id of the environment, or the URL-friendly key of the environment (i.e: the \&quot;slug\&quot;).
         * @param {string} proxyConfigId Either the unique id of the proxy config, or the URL-friendly key of the proxy config (i.e: the \&quot;slug\&quot;).
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        deleteProxyConfig(projId, envId, proxyConfigId, options) {
            return localVarFp
                .deleteProxyConfig(projId, envId, proxyConfigId, options)
                .then((request) => request(axios, basePath));
        },
        /**
         * Gets a proxy config, if such proxy config exists. Otherwise returns 404.
         * @summary Get Proxy Config
         * @param {string} projId Either the unique id of the project, or the URL-friendly key of the project (i.e: the \&quot;slug\&quot;).
         * @param {string} envId Either the unique id of the environment, or the URL-friendly key of the environment (i.e: the \&quot;slug\&quot;).
         * @param {string} proxyConfigId Either the unique id of the proxy config, or the URL-friendly key of the proxy config (i.e: the \&quot;slug\&quot;).
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        getProxyConfig(projId, envId, proxyConfigId, options) {
            return localVarFp
                .getProxyConfig(projId, envId, proxyConfigId, options)
                .then((request) => request(axios, basePath));
        },
        /**
         * Lists all the proxy configs defined within an environment.
         * @summary List Proxy Configs
         * @param {string} projId Either the unique id of the project, or the URL-friendly key of the project (i.e: the \&quot;slug\&quot;).
         * @param {string} envId Either the unique id of the environment, or the URL-friendly key of the environment (i.e: the \&quot;slug\&quot;).
         * @param {number} [page] Page number of the results to fetch, starting at 1.
         * @param {number} [perPage] The number of results per page (max 100).
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        listProxyConfigs(projId, envId, page, perPage, options) {
            return localVarFp
                .listProxyConfigs(projId, envId, page, perPage, options)
                .then((request) => request(axios, basePath));
        },
        /**
         * Partially updates the proxy config definition. Fields that will be provided will be completely overwritten.
         * @summary Update Proxy Config
         * @param {string} projId Either the unique id of the project, or the URL-friendly key of the project (i.e: the \&quot;slug\&quot;).
         * @param {string} envId Either the unique id of the environment, or the URL-friendly key of the environment (i.e: the \&quot;slug\&quot;).
         * @param {string} proxyConfigId Either the unique id of the proxy config, or the URL-friendly key of the proxy config (i.e: the \&quot;slug\&quot;).
         * @param {ProxyConfigUpdate} proxyConfigUpdate
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        updateProxyConfig(projId, envId, proxyConfigId, proxyConfigUpdate, options) {
            return localVarFp
                .updateProxyConfig(projId, envId, proxyConfigId, proxyConfigUpdate, options)
                .then((request) => request(axios, basePath));
        },
    };
};
exports.ProxyConfigApiFactory = ProxyConfigApiFactory;
/**
 * ProxyConfigApi - object-oriented interface
 * @export
 * @class ProxyConfigApi
 * @extends {BaseAPI}
 */
class ProxyConfigApi extends base_1.BaseAPI {
    /**
     * Creates a new proxy config inside the Permit.io system.  If the proxy config is already created: will return 200 instead of 201, and will return the existing proxy config object in the response body.
     * @summary Create Proxy Config
     * @param {ProxyConfigApiCreateProxyConfigRequest} requestParameters Request parameters.
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     * @memberof ProxyConfigApi
     */
    createProxyConfig(requestParameters, options) {
        return (0, exports.ProxyConfigApiFp)(this.configuration)
            .createProxyConfig(requestParameters.projId, requestParameters.envId, requestParameters.proxyConfigCreate, options)
            .then((request) => request(this.axios, this.basePath));
    }
    /**
     * Deletes the proxy config and all its related data.
     * @summary Delete Proxy Config
     * @param {ProxyConfigApiDeleteProxyConfigRequest} requestParameters Request parameters.
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     * @memberof ProxyConfigApi
     */
    deleteProxyConfig(requestParameters, options) {
        return (0, exports.ProxyConfigApiFp)(this.configuration)
            .deleteProxyConfig(requestParameters.projId, requestParameters.envId, requestParameters.proxyConfigId, options)
            .then((request) => request(this.axios, this.basePath));
    }
    /**
     * Gets a proxy config, if such proxy config exists. Otherwise returns 404.
     * @summary Get Proxy Config
     * @param {ProxyConfigApiGetProxyConfigRequest} requestParameters Request parameters.
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     * @memberof ProxyConfigApi
     */
    getProxyConfig(requestParameters, options) {
        return (0, exports.ProxyConfigApiFp)(this.configuration)
            .getProxyConfig(requestParameters.projId, requestParameters.envId, requestParameters.proxyConfigId, options)
            .then((request) => request(this.axios, this.basePath));
    }
    /**
     * Lists all the proxy configs defined within an environment.
     * @summary List Proxy Configs
     * @param {ProxyConfigApiListProxyConfigsRequest} requestParameters Request parameters.
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     * @memberof ProxyConfigApi
     */
    listProxyConfigs(requestParameters, options) {
        return (0, exports.ProxyConfigApiFp)(this.configuration)
            .listProxyConfigs(requestParameters.projId, requestParameters.envId, requestParameters.page, requestParameters.perPage, options)
            .then((request) => request(this.axios, this.basePath));
    }
    /**
     * Partially updates the proxy config definition. Fields that will be provided will be completely overwritten.
     * @summary Update Proxy Config
     * @param {ProxyConfigApiUpdateProxyConfigRequest} requestParameters Request parameters.
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     * @memberof ProxyConfigApi
     */
    updateProxyConfig(requestParameters, options) {
        return (0, exports.ProxyConfigApiFp)(this.configuration)
            .updateProxyConfig(requestParameters.projId, requestParameters.envId, requestParameters.proxyConfigId, requestParameters.proxyConfigUpdate, options)
            .then((request) => request(this.axios, this.basePath));
    }
}
exports.ProxyConfigApi = ProxyConfigApi;
