// permitio/build/main/tests/fixtures.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.provideTestExecutionContext = exports.handleApiError = exports.printBreak = void 0;
const index_1 = require("../index");
const logger_1 = require("../logger");
const printBreak = () => {
    console.log('\n\n ----------- \n\n');
};
exports.printBreak = printBreak;
function handleApiError(error, message, t) {
    var _a, _b;
    const err = `${message}: status=${(_a = error.response) === null || _a === void 0 ? void 0 : _a.status}, url=${error.request.url}, method=${error.request.method}, details=${(_b = error.response) === null || _b === void 0 ? void 0 : _b.data}`;
    t.context.logger.error(err);
    t.fail(err);
}
exports.handleApiError = handleApiError;
const provideTestExecutionContext = (t) => {
    // config
    const defaultPDPAddress = process.env.CLOUD_PDP === 'true' ? 'https://cloudpdp.api.permit.io' : 'http://localhost:7766';
    const defaultApiAddress = process.env.API_TIER === 'prod' ? 'https://api.permit.io' : 'http://localhost:8000';
    const token = process.env.PDP_API_KEY || '';
    const pdpAddress = process.env.PDP_URL || defaultPDPAddress;
    const apiUrl = process.env.PDP_CONTROL_PLANE || defaultApiAddress;
    if (!token) {
        t.fail('PDP_API_KEY is not configured, test cannot run!');
    }
    t.context.permit = new index_1.Permit({
        token,
        pdp: pdpAddress,
        apiUrl,
        log: {
            level: 'debug',
        },
    });
    t.context.logger = logger_1.LoggerFactory.createLogger(t.context.permit.config);
};
exports.provideTestExecutionContext = provideTestExecutionContext;
