// permitio/build/module/openapi/api/policy-synchronizer-administration-api.js
/* tslint:disable */
/* eslint-disable */
/**
 * Permit.io API
 *  Authorization as a service
 *
 * The version of the OpenAPI document: 2.0.0
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
import globalAxios from 'axios';
// Some imports not used depending on template conditions
// @ts-ignore
import { DUMMY_BASE_URL, assertParamExists, setBearerAuthToObject, setSearchParams, serializeDataIfNeeded, toPathString, createRequestFunction, } from '../common';
// @ts-ignore
import { BASE_PATH, BaseAPI } from '../base';
/**
 * PolicySynchronizerAdministrationApi - axios parameter creator
 * @export
 */
export const PolicySynchronizerAdministrationApiAxiosParamCreator = function (configuration) {
    return {
        /**
         * Push policy synchronizer update to all environments in the vc organization
         * @summary Regenerate environments
         * @param {Message} message
         * @param {number} [chunkSize] Number of environments to regenerate in a chunk
         * @param {number} [chunkCooldown] Cooldown between chunks in seconds
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        policySynchronizerAdministration: async (message, chunkSize, chunkCooldown, options = {}) => {
            // verify required parameter 'message' is not null or undefined
            assertParamExists('policySynchronizerAdministration', 'message', message);
            const localVarPath = `/v2/internal_sysadmin/policy-sync/regenerate-environments`;
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new URL(localVarPath, DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = { method: 'PUT', ...baseOptions, ...options };
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            // authentication HTTPBearer required
            // http bearer authentication required
            await setBearerAuthToObject(localVarHeaderParameter, configuration);
            if (chunkSize !== undefined) {
                localVarQueryParameter['chunk_size'] = chunkSize;
            }
            if (chunkCooldown !== undefined) {
                localVarQueryParameter['chunk_cooldown'] = chunkCooldown;
            }
            localVarHeaderParameter['Content-Type'] = 'application/json';
            setSearchParams(localVarUrlObj, localVarQueryParameter);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = {
                ...localVarHeaderParameter,
                ...headersFromBaseOptions,
                ...options.headers,
            };
            localVarRequestOptions.data = serializeDataIfNeeded(message, localVarRequestOptions, configuration);
            return {
                url: toPathString(localVarUrlObj),
                options: localVarRequestOptions,
            };
        },
    };
};
/**
 * PolicySynchronizerAdministrationApi - functional programming interface
 * @export
 */
export const PolicySynchronizerAdministrationApiFp = function (configuration) {
    const localVarAxiosParamCreator = PolicySynchronizerAdministrationApiAxiosParamCreator(configuration);
    return {
        /**
         * Push policy synchronizer update to all environments in the vc organization
         * @summary Regenerate environments
         * @param {Message} message
         * @param {number} [chunkSize] Number of environments to regenerate in a chunk
         * @param {number} [chunkCooldown] Cooldown between chunks in seconds
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        async policySynchronizerAdministration(message, chunkSize, chunkCooldown, options) {
            const localVarAxiosArgs = await localVarAxiosParamCreator.policySynchronizerAdministration(message, chunkSize, chunkCooldown, options);
            return createRequestFunction(localVarAxiosArgs, globalAxios, BASE_PATH, configuration);
        },
    };
};
/**
 * PolicySynchronizerAdministrationApi - factory interface
 * @export
 */
export const PolicySynchronizerAdministrationApiFactory = function (configuration, basePath, axios) {
    const localVarFp = PolicySynchronizerAdministrationApiFp(configuration);
    return {
        /**
         * Push policy synchronizer update to all environments in the vc organization
         * @summary Regenerate environments
         * @param {Message} message
         * @param {number} [chunkSize] Number of environments to regenerate in a chunk
         * @param {number} [chunkCooldown] Cooldown between chunks in seconds
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        policySynchronizerAdministration(message, chunkSize, chunkCooldown, options) {
            return localVarFp
                .policySynchronizerAdministration(message, chunkSize, chunkCooldown, options)
                .then((request) => request(axios, basePath));
        },
    };
};
/**
 * PolicySynchronizerAdministrationApi - object-oriented interface
 * @export
 * @class PolicySynchronizerAdministrationApi
 * @extends {BaseAPI}
 */
export class PolicySynchronizerAdministrationApi extends BaseAPI {
    /**
     * Push policy synchronizer update to all environments in the vc organization
     * @summary Regenerate environments
     * @param {PolicySynchronizerAdministrationApiPolicySynchronizerAdministrationRequest} requestParameters Request parameters.
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     * @memberof PolicySynchronizerAdministrationApi
     */
    policySynchronizerAdministration(requestParameters, options) {
        return PolicySynchronizerAdministrationApiFp(this.configuration)
            .policySynchronizerAdministration(requestParameters.message, requestParameters.chunkSize, requestParameters.chunkCooldown, options)
            .then((request) => request(this.axios, this.basePath));
    }
}
