// permitio/build/module/openapi/api/scope-configurations-api.js
/* tslint:disable */
/* eslint-disable */
/**
 * Permit.io API
 *  Authorization as a service
 *
 * The version of the OpenAPI document: 2.0.0
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
import globalAxios from 'axios';
// Some imports not used depending on template conditions
// @ts-ignore
import { DUMMY_BASE_URL, assertParamExists, setBearerAuthToObject, setSearchParams, serializeDataIfNeeded, toPathString, createRequestFunction, } from '../common';
// @ts-ignore
import { BASE_PATH, BaseAPI } from '../base';
/**
 * ScopeConfigurationsApi - axios parameter creator
 * @export
 */
export const ScopeConfigurationsApiAxiosParamCreator = function (configuration) {
    return {
        /**
         * Returns the custom OPAL Scope config for given environment, if exists. If not custom config is set, return 404.
         * @summary Get Scope Config
         * @param {string} projId Either the unique id of the project, or the URL-friendly key of the project (i.e: the \&quot;slug\&quot;).
         * @param {string} envId Either the unique id of the environment, or the URL-friendly key of the environment (i.e: the \&quot;slug\&quot;).
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        getScopeConfig: async (projId, envId, options = {}) => {
            // verify required parameter 'projId' is not null or undefined
            assertParamExists('getScopeConfig', 'projId', projId);
            // verify required parameter 'envId' is not null or undefined
            assertParamExists('getScopeConfig', 'envId', envId);
            const localVarPath = `/v2/projects/{proj_id}/{env_id}/opal_scope`
                .replace(`{${'proj_id'}}`, encodeURIComponent(String(projId)))
                .replace(`{${'env_id'}}`, encodeURIComponent(String(envId)));
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new URL(localVarPath, DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = { method: 'GET', ...baseOptions, ...options };
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            // authentication HTTPBearer required
            // http bearer authentication required
            await setBearerAuthToObject(localVarHeaderParameter, configuration);
            setSearchParams(localVarUrlObj, localVarQueryParameter);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = {
                ...localVarHeaderParameter,
                ...headersFromBaseOptions,
                ...options.headers,
            };
            return {
                url: toPathString(localVarUrlObj),
                options: localVarRequestOptions,
            };
        },
        /**
         *
         * @summary Reset Scope Config
         * @param {string} projId Either the unique id of the project, or the URL-friendly key of the project (i.e: the \&quot;slug\&quot;).
         * @param {string} envId Either the unique id of the environment, or the URL-friendly key of the environment (i.e: the \&quot;slug\&quot;).
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        resetScopeConfig: async (projId, envId, options = {}) => {
            // verify required parameter 'projId' is not null or undefined
            assertParamExists('resetScopeConfig', 'projId', projId);
            // verify required parameter 'envId' is not null or undefined
            assertParamExists('resetScopeConfig', 'envId', envId);
            const localVarPath = `/v2/projects/{proj_id}/{env_id}/opal_scope`
                .replace(`{${'proj_id'}}`, encodeURIComponent(String(projId)))
                .replace(`{${'env_id'}}`, encodeURIComponent(String(envId)));
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new URL(localVarPath, DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = { method: 'DELETE', ...baseOptions, ...options };
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            // authentication HTTPBearer required
            // http bearer authentication required
            await setBearerAuthToObject(localVarHeaderParameter, configuration);
            setSearchParams(localVarUrlObj, localVarQueryParameter);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = {
                ...localVarHeaderParameter,
                ...headersFromBaseOptions,
                ...options.headers,
            };
            return {
                url: toPathString(localVarUrlObj),
                options: localVarRequestOptions,
            };
        },
        /**
         *
         * @summary Set Scope Config
         * @param {string} projId Either the unique id of the project, or the URL-friendly key of the project (i.e: the \&quot;slug\&quot;).
         * @param {string} envId Either the unique id of the environment, or the URL-friendly key of the environment (i.e: the \&quot;slug\&quot;).
         * @param {ScopeConfigSet} scopeConfigSet
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        setScopeConfig: async (projId, envId, scopeConfigSet, options = {}) => {
            // verify required parameter 'projId' is not null or undefined
            assertParamExists('setScopeConfig', 'projId', projId);
            // verify required parameter 'envId' is not null or undefined
            assertParamExists('setScopeConfig', 'envId', envId);
            // verify required parameter 'scopeConfigSet' is not null or undefined
            assertParamExists('setScopeConfig', 'scopeConfigSet', scopeConfigSet);
            const localVarPath = `/v2/projects/{proj_id}/{env_id}/opal_scope`
                .replace(`{${'proj_id'}}`, encodeURIComponent(String(projId)))
                .replace(`{${'env_id'}}`, encodeURIComponent(String(envId)));
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new URL(localVarPath, DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = { method: 'PUT', ...baseOptions, ...options };
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            // authentication HTTPBearer required
            // http bearer authentication required
            await setBearerAuthToObject(localVarHeaderParameter, configuration);
            localVarHeaderParameter['Content-Type'] = 'application/json';
            setSearchParams(localVarUrlObj, localVarQueryParameter);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = {
                ...localVarHeaderParameter,
                ...headersFromBaseOptions,
                ...options.headers,
            };
            localVarRequestOptions.data = serializeDataIfNeeded(scopeConfigSet, localVarRequestOptions, configuration);
            return {
                url: toPathString(localVarUrlObj),
                options: localVarRequestOptions,
            };
        },
    };
};
/**
 * ScopeConfigurationsApi - functional programming interface
 * @export
 */
export const ScopeConfigurationsApiFp = function (configuration) {
    const localVarAxiosParamCreator = ScopeConfigurationsApiAxiosParamCreator(configuration);
    return {
        /**
         * Returns the custom OPAL Scope config for given environment, if exists. If not custom config is set, return 404.
         * @summary Get Scope Config
         * @param {string} projId Either the unique id of the project, or the URL-friendly key of the project (i.e: the \&quot;slug\&quot;).
         * @param {string} envId Either the unique id of the environment, or the URL-friendly key of the environment (i.e: the \&quot;slug\&quot;).
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        async getScopeConfig(projId, envId, options) {
            const localVarAxiosArgs = await localVarAxiosParamCreator.getScopeConfig(projId, envId, options);
            return createRequestFunction(localVarAxiosArgs, globalAxios, BASE_PATH, configuration);
        },
        /**
         *
         * @summary Reset Scope Config
         * @param {string} projId Either the unique id of the project, or the URL-friendly key of the project (i.e: the \&quot;slug\&quot;).
         * @param {string} envId Either the unique id of the environment, or the URL-friendly key of the environment (i.e: the \&quot;slug\&quot;).
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        async resetScopeConfig(projId, envId, options) {
            const localVarAxiosArgs = await localVarAxiosParamCreator.resetScopeConfig(projId, envId, options);
            return createRequestFunction(localVarAxiosArgs, globalAxios, BASE_PATH, configuration);
        },
        /**
         *
         * @summary Set Scope Config
         * @param {string} projId Either the unique id of the project, or the URL-friendly key of the project (i.e: the \&quot;slug\&quot;).
         * @param {string} envId Either the unique id of the environment, or the URL-friendly key of the environment (i.e: the \&quot;slug\&quot;).
         * @param {ScopeConfigSet} scopeConfigSet
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        async setScopeConfig(projId, envId, scopeConfigSet, options) {
            const localVarAxiosArgs = await localVarAxiosParamCreator.setScopeConfig(projId, envId, scopeConfigSet, options);
            return createRequestFunction(localVarAxiosArgs, globalAxios, BASE_PATH, configuration);
        },
    };
};
/**
 * ScopeConfigurationsApi - factory interface
 * @export
 */
export const ScopeConfigurationsApiFactory = function (configuration, basePath, axios) {
    const localVarFp = ScopeConfigurationsApiFp(configuration);
    return {
        /**
         * Returns the custom OPAL Scope config for given environment, if exists. If not custom config is set, return 404.
         * @summary Get Scope Config
         * @param {string} projId Either the unique id of the project, or the URL-friendly key of the project (i.e: the \&quot;slug\&quot;).
         * @param {string} envId Either the unique id of the environment, or the URL-friendly key of the environment (i.e: the \&quot;slug\&quot;).
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        getScopeConfig(projId, envId, options) {
            return localVarFp
                .getScopeConfig(projId, envId, options)
                .then((request) => request(axios, basePath));
        },
        /**
         *
         * @summary Reset Scope Config
         * @param {string} projId Either the unique id of the project, or the URL-friendly key of the project (i.e: the \&quot;slug\&quot;).
         * @param {string} envId Either the unique id of the environment, or the URL-friendly key of the environment (i.e: the \&quot;slug\&quot;).
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        resetScopeConfig(projId, envId, options) {
            return localVarFp
                .resetScopeConfig(projId, envId, options)
                .then((request) => request(axios, basePath));
        },
        /**
         *
         * @summary Set Scope Config
         * @param {string} projId Either the unique id of the project, or the URL-friendly key of the project (i.e: the \&quot;slug\&quot;).
         * @param {string} envId Either the unique id of the environment, or the URL-friendly key of the environment (i.e: the \&quot;slug\&quot;).
         * @param {ScopeConfigSet} scopeConfigSet
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        setScopeConfig(projId, envId, scopeConfigSet, options) {
            return localVarFp
                .setScopeConfig(projId, envId, scopeConfigSet, options)
                .then((request) => request(axios, basePath));
        },
    };
};
/**
 * ScopeConfigurationsApi - object-oriented interface
 * @export
 * @class ScopeConfigurationsApi
 * @extends {BaseAPI}
 */
export class ScopeConfigurationsApi extends BaseAPI {
    /**
     * Returns the custom OPAL Scope config for given environment, if exists. If not custom config is set, return 404.
     * @summary Get Scope Config
     * @param {ScopeConfigurationsApiGetScopeConfigRequest} requestParameters Request parameters.
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     * @memberof ScopeConfigurationsApi
     */
    getScopeConfig(requestParameters, options) {
        return ScopeConfigurationsApiFp(this.configuration)
            .getScopeConfig(requestParameters.projId, requestParameters.envId, options)
            .then((request) => request(this.axios, this.basePath));
    }
    /**
     *
     * @summary Reset Scope Config
     * @param {ScopeConfigurationsApiResetScopeConfigRequest} requestParameters Request parameters.
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     * @memberof ScopeConfigurationsApi
     */
    resetScopeConfig(requestParameters, options) {
        return ScopeConfigurationsApiFp(this.configuration)
            .resetScopeConfig(requestParameters.projId, requestParameters.envId, options)
            .then((request) => request(this.axios, this.basePath));
    }
    /**
     *
     * @summary Set Scope Config
     * @param {ScopeConfigurationsApiSetScopeConfigRequest} requestParameters Request parameters.
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     * @memberof ScopeConfigurationsApi
     */
    setScopeConfig(requestParameters, options) {
        return ScopeConfigurationsApiFp(this.configuration)
            .setScopeConfig(requestParameters.projId, requestParameters.envId, requestParameters.scopeConfigSet, options)
            .then((request) => request(this.axios, this.basePath));
    }
}
