// permitio/build/module/openapi/api/users-elements-data-api.js
/* tslint:disable */
/* eslint-disable */
/**
 * Permit.io API
 *  Authorization as a service
 *
 * The version of the OpenAPI document: 2.0.0
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
import globalAxios from 'axios';
// Some imports not used depending on template conditions
// @ts-ignore
import { DUMMY_BASE_URL, assertParamExists, setBearerAuthToObject, setSearchParams, serializeDataIfNeeded, toPathString, createRequestFunction, } from '../common';
// @ts-ignore
import { BASE_PATH, BaseAPI } from '../base';
/**
 * UsersElementsDataApi - axios parameter creator
 * @export
 */
export const UsersElementsDataApiAxiosParamCreator = function (configuration) {
    return {
        /**
         * Assigns a role to the user within the tenant.  The tenant defines the scope of the assignment. In other words, the role is effective only within the tenant.
         * @summary Assign role to user
         * @param {string} projId Either the unique id of the project, or the URL-friendly key of the project (i.e: the \&quot;slug\&quot;).
         * @param {string} envId Either the unique id of the environment, or the URL-friendly key of the environment (i.e: the \&quot;slug\&quot;).
         * @param {string} elementsConfigId Either the unique id of the elements_config, or the URL-friendly key of the elements_config (i.e: the \&quot;slug\&quot;).
         * @param {string} userId Either the unique id of the user, or the URL-friendly key of the user (i.e: the \&quot;slug\&quot;).
         * @param {ElementsUserRoleCreate} elementsUserRoleCreate
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        elementsAssignRoleToUser: async (projId, envId, elementsConfigId, userId, elementsUserRoleCreate, options = {}) => {
            // verify required parameter 'projId' is not null or undefined
            assertParamExists('elementsAssignRoleToUser', 'projId', projId);
            // verify required parameter 'envId' is not null or undefined
            assertParamExists('elementsAssignRoleToUser', 'envId', envId);
            // verify required parameter 'elementsConfigId' is not null or undefined
            assertParamExists('elementsAssignRoleToUser', 'elementsConfigId', elementsConfigId);
            // verify required parameter 'userId' is not null or undefined
            assertParamExists('elementsAssignRoleToUser', 'userId', userId);
            // verify required parameter 'elementsUserRoleCreate' is not null or undefined
            assertParamExists('elementsAssignRoleToUser', 'elementsUserRoleCreate', elementsUserRoleCreate);
            const localVarPath = `/v2/elements/{proj_id}/{env_id}/config/{elements_config_id}/data/users/{user_id}/roles`
                .replace(`{${'proj_id'}}`, encodeURIComponent(String(projId)))
                .replace(`{${'env_id'}}`, encodeURIComponent(String(envId)))
                .replace(`{${'elements_config_id'}}`, encodeURIComponent(String(elementsConfigId)))
                .replace(`{${'user_id'}}`, encodeURIComponent(String(userId)));
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new URL(localVarPath, DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = { method: 'POST', ...baseOptions, ...options };
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            // authentication HTTPBearer required
            // http bearer authentication required
            await setBearerAuthToObject(localVarHeaderParameter, configuration);
            localVarHeaderParameter['Content-Type'] = 'application/json';
            setSearchParams(localVarUrlObj, localVarQueryParameter);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = {
                ...localVarHeaderParameter,
                ...headersFromBaseOptions,
                ...options.headers,
            };
            localVarRequestOptions.data = serializeDataIfNeeded(elementsUserRoleCreate, localVarRequestOptions, configuration);
            return {
                url: toPathString(localVarUrlObj),
                options: localVarRequestOptions,
            };
        },
        /**
         * Creates a new user inside the Permit.io system, from that point forward you may run permission checks on that user.  If the user is already created: will return 200 instead of 201, and will return the existing user object in the response body.
         * @summary Create user
         * @param {string} projId Either the unique id of the project, or the URL-friendly key of the project (i.e: the \&quot;slug\&quot;).
         * @param {string} envId Either the unique id of the environment, or the URL-friendly key of the environment (i.e: the \&quot;slug\&quot;).
         * @param {string} elementsConfigId Either the unique id of the elements_config, or the URL-friendly key of the elements_config (i.e: the \&quot;slug\&quot;).
         * @param {ElementsUserCreate} elementsUserCreate
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        elementsCreateUser: async (projId, envId, elementsConfigId, elementsUserCreate, options = {}) => {
            // verify required parameter 'projId' is not null or undefined
            assertParamExists('elementsCreateUser', 'projId', projId);
            // verify required parameter 'envId' is not null or undefined
            assertParamExists('elementsCreateUser', 'envId', envId);
            // verify required parameter 'elementsConfigId' is not null or undefined
            assertParamExists('elementsCreateUser', 'elementsConfigId', elementsConfigId);
            // verify required parameter 'elementsUserCreate' is not null or undefined
            assertParamExists('elementsCreateUser', 'elementsUserCreate', elementsUserCreate);
            const localVarPath = `/v2/elements/{proj_id}/{env_id}/config/{elements_config_id}/data/users`
                .replace(`{${'proj_id'}}`, encodeURIComponent(String(projId)))
                .replace(`{${'env_id'}}`, encodeURIComponent(String(envId)))
                .replace(`{${'elements_config_id'}}`, encodeURIComponent(String(elementsConfigId)));
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new URL(localVarPath, DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = { method: 'POST', ...baseOptions, ...options };
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            // authentication HTTPBearer required
            // http bearer authentication required
            await setBearerAuthToObject(localVarHeaderParameter, configuration);
            localVarHeaderParameter['Content-Type'] = 'application/json';
            setSearchParams(localVarUrlObj, localVarQueryParameter);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = {
                ...localVarHeaderParameter,
                ...headersFromBaseOptions,
                ...options.headers,
            };
            localVarRequestOptions.data = serializeDataIfNeeded(elementsUserCreate, localVarRequestOptions, configuration);
            return {
                url: toPathString(localVarUrlObj),
                options: localVarRequestOptions,
            };
        },
        /**
         *
         * @summary Delete user
         * @param {string} projId Either the unique id of the project, or the URL-friendly key of the project (i.e: the \&quot;slug\&quot;).
         * @param {string} envId Either the unique id of the environment, or the URL-friendly key of the environment (i.e: the \&quot;slug\&quot;).
         * @param {string} elementsConfigId Either the unique id of the elements_config, or the URL-friendly key of the elements_config (i.e: the \&quot;slug\&quot;).
         * @param {string} userId Either the unique id of the user, or the URL-friendly key of the user (i.e: the \&quot;slug\&quot;).
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        elementsDeleteUser: async (projId, envId, elementsConfigId, userId, options = {}) => {
            // verify required parameter 'projId' is not null or undefined
            assertParamExists('elementsDeleteUser', 'projId', projId);
            // verify required parameter 'envId' is not null or undefined
            assertParamExists('elementsDeleteUser', 'envId', envId);
            // verify required parameter 'elementsConfigId' is not null or undefined
            assertParamExists('elementsDeleteUser', 'elementsConfigId', elementsConfigId);
            // verify required parameter 'userId' is not null or undefined
            assertParamExists('elementsDeleteUser', 'userId', userId);
            const localVarPath = `/v2/elements/{proj_id}/{env_id}/config/{elements_config_id}/data/users/{user_id}`
                .replace(`{${'proj_id'}}`, encodeURIComponent(String(projId)))
                .replace(`{${'env_id'}}`, encodeURIComponent(String(envId)))
                .replace(`{${'elements_config_id'}}`, encodeURIComponent(String(elementsConfigId)))
                .replace(`{${'user_id'}}`, encodeURIComponent(String(userId)));
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new URL(localVarPath, DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = { method: 'DELETE', ...baseOptions, ...options };
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            // authentication HTTPBearer required
            // http bearer authentication required
            await setBearerAuthToObject(localVarHeaderParameter, configuration);
            setSearchParams(localVarUrlObj, localVarQueryParameter);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = {
                ...localVarHeaderParameter,
                ...headersFromBaseOptions,
                ...options.headers,
            };
            return {
                url: toPathString(localVarUrlObj),
                options: localVarRequestOptions,
            };
        },
        /**
         * Lists all the users defined within an environment.
         * @summary List roles
         * @param {string} projId Either the unique id of the project, or the URL-friendly key of the project (i.e: the \&quot;slug\&quot;).
         * @param {string} envId Either the unique id of the environment, or the URL-friendly key of the environment (i.e: the \&quot;slug\&quot;).
         * @param {string} elementsConfigId Either the unique id of the elements_config, or the URL-friendly key of the elements_config (i.e: the \&quot;slug\&quot;).
         * @param {string} [search] Text search for the email field
         * @param {number} [page] Page number of the results to fetch, starting at 1.
         * @param {number} [perPage] The number of results per page (max 100).
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        elementsListRoles: async (projId, envId, elementsConfigId, search, page, perPage, options = {}) => {
            // verify required parameter 'projId' is not null or undefined
            assertParamExists('elementsListRoles', 'projId', projId);
            // verify required parameter 'envId' is not null or undefined
            assertParamExists('elementsListRoles', 'envId', envId);
            // verify required parameter 'elementsConfigId' is not null or undefined
            assertParamExists('elementsListRoles', 'elementsConfigId', elementsConfigId);
            const localVarPath = `/v2/elements/{proj_id}/{env_id}/config/{elements_config_id}/data/roles`
                .replace(`{${'proj_id'}}`, encodeURIComponent(String(projId)))
                .replace(`{${'env_id'}}`, encodeURIComponent(String(envId)))
                .replace(`{${'elements_config_id'}}`, encodeURIComponent(String(elementsConfigId)));
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new URL(localVarPath, DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = { method: 'GET', ...baseOptions, ...options };
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            // authentication HTTPBearer required
            // http bearer authentication required
            await setBearerAuthToObject(localVarHeaderParameter, configuration);
            if (search !== undefined) {
                localVarQueryParameter['search'] = search;
            }
            if (page !== undefined) {
                localVarQueryParameter['page'] = page;
            }
            if (perPage !== undefined) {
                localVarQueryParameter['per_page'] = perPage;
            }
            setSearchParams(localVarUrlObj, localVarQueryParameter);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = {
                ...localVarHeaderParameter,
                ...headersFromBaseOptions,
                ...options.headers,
            };
            return {
                url: toPathString(localVarUrlObj),
                options: localVarRequestOptions,
            };
        },
        /**
         * Lists all the users defined within an environment.
         * @summary List users
         * @param {string} projId Either the unique id of the project, or the URL-friendly key of the project (i.e: the \&quot;slug\&quot;).
         * @param {string} envId Either the unique id of the environment, or the URL-friendly key of the environment (i.e: the \&quot;slug\&quot;).
         * @param {string} elementsConfigId Either the unique id of the elements_config, or the URL-friendly key of the elements_config (i.e: the \&quot;slug\&quot;).
         * @param {string} [search] Text search for the email field
         * @param {number} [page] Page number of the results to fetch, starting at 1.
         * @param {number} [perPage] The number of results per page (max 100).
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        elementsListUsers: async (projId, envId, elementsConfigId, search, page, perPage, options = {}) => {
            // verify required parameter 'projId' is not null or undefined
            assertParamExists('elementsListUsers', 'projId', projId);
            // verify required parameter 'envId' is not null or undefined
            assertParamExists('elementsListUsers', 'envId', envId);
            // verify required parameter 'elementsConfigId' is not null or undefined
            assertParamExists('elementsListUsers', 'elementsConfigId', elementsConfigId);
            const localVarPath = `/v2/elements/{proj_id}/{env_id}/config/{elements_config_id}/data/users`
                .replace(`{${'proj_id'}}`, encodeURIComponent(String(projId)))
                .replace(`{${'env_id'}}`, encodeURIComponent(String(envId)))
                .replace(`{${'elements_config_id'}}`, encodeURIComponent(String(elementsConfigId)));
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new URL(localVarPath, DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = { method: 'GET', ...baseOptions, ...options };
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            // authentication HTTPBearer required
            // http bearer authentication required
            await setBearerAuthToObject(localVarHeaderParameter, configuration);
            if (search !== undefined) {
                localVarQueryParameter['search'] = search;
            }
            if (page !== undefined) {
                localVarQueryParameter['page'] = page;
            }
            if (perPage !== undefined) {
                localVarQueryParameter['per_page'] = perPage;
            }
            setSearchParams(localVarUrlObj, localVarQueryParameter);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = {
                ...localVarHeaderParameter,
                ...headersFromBaseOptions,
                ...options.headers,
            };
            return {
                url: toPathString(localVarUrlObj),
                options: localVarRequestOptions,
            };
        },
        /**
         * Unassigns the role from the user within the tenant.  The tenant defines the scope of the assignment. In other words, the role is effective only within the tenant.  If the role is not actually assigned, will return 404.
         * @summary Unassign role from user
         * @param {string} projId Either the unique id of the project, or the URL-friendly key of the project (i.e: the \&quot;slug\&quot;).
         * @param {string} envId Either the unique id of the environment, or the URL-friendly key of the environment (i.e: the \&quot;slug\&quot;).
         * @param {string} elementsConfigId Either the unique id of the elements_config, or the URL-friendly key of the elements_config (i.e: the \&quot;slug\&quot;).
         * @param {string} userId Either the unique id of the user, or the URL-friendly key of the user (i.e: the \&quot;slug\&quot;).
         * @param {ElementsUserRoleRemove} elementsUserRoleRemove
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        elementsUnassignRoleFromUser: async (projId, envId, elementsConfigId, userId, elementsUserRoleRemove, options = {}) => {
            // verify required parameter 'projId' is not null or undefined
            assertParamExists('elementsUnassignRoleFromUser', 'projId', projId);
            // verify required parameter 'envId' is not null or undefined
            assertParamExists('elementsUnassignRoleFromUser', 'envId', envId);
            // verify required parameter 'elementsConfigId' is not null or undefined
            assertParamExists('elementsUnassignRoleFromUser', 'elementsConfigId', elementsConfigId);
            // verify required parameter 'userId' is not null or undefined
            assertParamExists('elementsUnassignRoleFromUser', 'userId', userId);
            // verify required parameter 'elementsUserRoleRemove' is not null or undefined
            assertParamExists('elementsUnassignRoleFromUser', 'elementsUserRoleRemove', elementsUserRoleRemove);
            const localVarPath = `/v2/elements/{proj_id}/{env_id}/config/{elements_config_id}/data/users/{user_id}/roles`
                .replace(`{${'proj_id'}}`, encodeURIComponent(String(projId)))
                .replace(`{${'env_id'}}`, encodeURIComponent(String(envId)))
                .replace(`{${'elements_config_id'}}`, encodeURIComponent(String(elementsConfigId)))
                .replace(`{${'user_id'}}`, encodeURIComponent(String(userId)));
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new URL(localVarPath, DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = { method: 'DELETE', ...baseOptions, ...options };
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            // authentication HTTPBearer required
            // http bearer authentication required
            await setBearerAuthToObject(localVarHeaderParameter, configuration);
            localVarHeaderParameter['Content-Type'] = 'application/json';
            setSearchParams(localVarUrlObj, localVarQueryParameter);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = {
                ...localVarHeaderParameter,
                ...headersFromBaseOptions,
                ...options.headers,
            };
            localVarRequestOptions.data = serializeDataIfNeeded(elementsUserRoleRemove, localVarRequestOptions, configuration);
            return {
                url: toPathString(localVarUrlObj),
                options: localVarRequestOptions,
            };
        },
        /**
         * Updates the embed_config.
         * @summary Set Config Active
         * @param {string} elementsConfigId Either the unique id of the elements_config, or the URL-friendly key of the elements_config (i.e: the \&quot;slug\&quot;).
         * @param {string} projId Either the unique id of the project, or the URL-friendly key of the project (i.e: the \&quot;slug\&quot;).
         * @param {string} envId Either the unique id of the environment, or the URL-friendly key of the environment (i.e: the \&quot;slug\&quot;).
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        setConfigActive: async (elementsConfigId, projId, envId, options = {}) => {
            // verify required parameter 'elementsConfigId' is not null or undefined
            assertParamExists('setConfigActive', 'elementsConfigId', elementsConfigId);
            // verify required parameter 'projId' is not null or undefined
            assertParamExists('setConfigActive', 'projId', projId);
            // verify required parameter 'envId' is not null or undefined
            assertParamExists('setConfigActive', 'envId', envId);
            const localVarPath = `/v2/elements/{proj_id}/{env_id}/config/{elements_config_id}/data/active`
                .replace(`{${'elements_config_id'}}`, encodeURIComponent(String(elementsConfigId)))
                .replace(`{${'proj_id'}}`, encodeURIComponent(String(projId)))
                .replace(`{${'env_id'}}`, encodeURIComponent(String(envId)));
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new URL(localVarPath, DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = { method: 'POST', ...baseOptions, ...options };
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            // authentication HTTPBearer required
            // http bearer authentication required
            await setBearerAuthToObject(localVarHeaderParameter, configuration);
            setSearchParams(localVarUrlObj, localVarQueryParameter);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = {
                ...localVarHeaderParameter,
                ...headersFromBaseOptions,
                ...options.headers,
            };
            return {
                url: toPathString(localVarUrlObj),
                options: localVarRequestOptions,
            };
        },
    };
};
/**
 * UsersElementsDataApi - functional programming interface
 * @export
 */
export const UsersElementsDataApiFp = function (configuration) {
    const localVarAxiosParamCreator = UsersElementsDataApiAxiosParamCreator(configuration);
    return {
        /**
         * Assigns a role to the user within the tenant.  The tenant defines the scope of the assignment. In other words, the role is effective only within the tenant.
         * @summary Assign role to user
         * @param {string} projId Either the unique id of the project, or the URL-friendly key of the project (i.e: the \&quot;slug\&quot;).
         * @param {string} envId Either the unique id of the environment, or the URL-friendly key of the environment (i.e: the \&quot;slug\&quot;).
         * @param {string} elementsConfigId Either the unique id of the elements_config, or the URL-friendly key of the elements_config (i.e: the \&quot;slug\&quot;).
         * @param {string} userId Either the unique id of the user, or the URL-friendly key of the user (i.e: the \&quot;slug\&quot;).
         * @param {ElementsUserRoleCreate} elementsUserRoleCreate
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        async elementsAssignRoleToUser(projId, envId, elementsConfigId, userId, elementsUserRoleCreate, options) {
            const localVarAxiosArgs = await localVarAxiosParamCreator.elementsAssignRoleToUser(projId, envId, elementsConfigId, userId, elementsUserRoleCreate, options);
            return createRequestFunction(localVarAxiosArgs, globalAxios, BASE_PATH, configuration);
        },
        /**
         * Creates a new user inside the Permit.io system, from that point forward you may run permission checks on that user.  If the user is already created: will return 200 instead of 201, and will return the existing user object in the response body.
         * @summary Create user
         * @param {string} projId Either the unique id of the project, or the URL-friendly key of the project (i.e: the \&quot;slug\&quot;).
         * @param {string} envId Either the unique id of the environment, or the URL-friendly key of the environment (i.e: the \&quot;slug\&quot;).
         * @param {string} elementsConfigId Either the unique id of the elements_config, or the URL-friendly key of the elements_config (i.e: the \&quot;slug\&quot;).
         * @param {ElementsUserCreate} elementsUserCreate
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        async elementsCreateUser(projId, envId, elementsConfigId, elementsUserCreate, options) {
            const localVarAxiosArgs = await localVarAxiosParamCreator.elementsCreateUser(projId, envId, elementsConfigId, elementsUserCreate, options);
            return createRequestFunction(localVarAxiosArgs, globalAxios, BASE_PATH, configuration);
        },
        /**
         *
         * @summary Delete user
         * @param {string} projId Either the unique id of the project, or the URL-friendly key of the project (i.e: the \&quot;slug\&quot;).
         * @param {string} envId Either the unique id of the environment, or the URL-friendly key of the environment (i.e: the \&quot;slug\&quot;).
         * @param {string} elementsConfigId Either the unique id of the elements_config, or the URL-friendly key of the elements_config (i.e: the \&quot;slug\&quot;).
         * @param {string} userId Either the unique id of the user, or the URL-friendly key of the user (i.e: the \&quot;slug\&quot;).
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        async elementsDeleteUser(projId, envId, elementsConfigId, userId, options) {
            const localVarAxiosArgs = await localVarAxiosParamCreator.elementsDeleteUser(projId, envId, elementsConfigId, userId, options);
            return createRequestFunction(localVarAxiosArgs, globalAxios, BASE_PATH, configuration);
        },
        /**
         * Lists all the users defined within an environment.
         * @summary List roles
         * @param {string} projId Either the unique id of the project, or the URL-friendly key of the project (i.e: the \&quot;slug\&quot;).
         * @param {string} envId Either the unique id of the environment, or the URL-friendly key of the environment (i.e: the \&quot;slug\&quot;).
         * @param {string} elementsConfigId Either the unique id of the elements_config, or the URL-friendly key of the elements_config (i.e: the \&quot;slug\&quot;).
         * @param {string} [search] Text search for the email field
         * @param {number} [page] Page number of the results to fetch, starting at 1.
         * @param {number} [perPage] The number of results per page (max 100).
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        async elementsListRoles(projId, envId, elementsConfigId, search, page, perPage, options) {
            const localVarAxiosArgs = await localVarAxiosParamCreator.elementsListRoles(projId, envId, elementsConfigId, search, page, perPage, options);
            return createRequestFunction(localVarAxiosArgs, globalAxios, BASE_PATH, configuration);
        },
        /**
         * Lists all the users defined within an environment.
         * @summary List users
         * @param {string} projId Either the unique id of the project, or the URL-friendly key of the project (i.e: the \&quot;slug\&quot;).
         * @param {string} envId Either the unique id of the environment, or the URL-friendly key of the environment (i.e: the \&quot;slug\&quot;).
         * @param {string} elementsConfigId Either the unique id of the elements_config, or the URL-friendly key of the elements_config (i.e: the \&quot;slug\&quot;).
         * @param {string} [search] Text search for the email field
         * @param {number} [page] Page number of the results to fetch, starting at 1.
         * @param {number} [perPage] The number of results per page (max 100).
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        async elementsListUsers(projId, envId, elementsConfigId, search, page, perPage, options) {
            const localVarAxiosArgs = await localVarAxiosParamCreator.elementsListUsers(projId, envId, elementsConfigId, search, page, perPage, options);
            return createRequestFunction(localVarAxiosArgs, globalAxios, BASE_PATH, configuration);
        },
        /**
         * Unassigns the role from the user within the tenant.  The tenant defines the scope of the assignment. In other words, the role is effective only within the tenant.  If the role is not actually assigned, will return 404.
         * @summary Unassign role from user
         * @param {string} projId Either the unique id of the project, or the URL-friendly key of the project (i.e: the \&quot;slug\&quot;).
         * @param {string} envId Either the unique id of the environment, or the URL-friendly key of the environment (i.e: the \&quot;slug\&quot;).
         * @param {string} elementsConfigId Either the unique id of the elements_config, or the URL-friendly key of the elements_config (i.e: the \&quot;slug\&quot;).
         * @param {string} userId Either the unique id of the user, or the URL-friendly key of the user (i.e: the \&quot;slug\&quot;).
         * @param {ElementsUserRoleRemove} elementsUserRoleRemove
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        async elementsUnassignRoleFromUser(projId, envId, elementsConfigId, userId, elementsUserRoleRemove, options) {
            const localVarAxiosArgs = await localVarAxiosParamCreator.elementsUnassignRoleFromUser(projId, envId, elementsConfigId, userId, elementsUserRoleRemove, options);
            return createRequestFunction(localVarAxiosArgs, globalAxios, BASE_PATH, configuration);
        },
        /**
         * Updates the embed_config.
         * @summary Set Config Active
         * @param {string} elementsConfigId Either the unique id of the elements_config, or the URL-friendly key of the elements_config (i.e: the \&quot;slug\&quot;).
         * @param {string} projId Either the unique id of the project, or the URL-friendly key of the project (i.e: the \&quot;slug\&quot;).
         * @param {string} envId Either the unique id of the environment, or the URL-friendly key of the environment (i.e: the \&quot;slug\&quot;).
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        async setConfigActive(elementsConfigId, projId, envId, options) {
            const localVarAxiosArgs = await localVarAxiosParamCreator.setConfigActive(elementsConfigId, projId, envId, options);
            return createRequestFunction(localVarAxiosArgs, globalAxios, BASE_PATH, configuration);
        },
    };
};
/**
 * UsersElementsDataApi - factory interface
 * @export
 */
export const UsersElementsDataApiFactory = function (configuration, basePath, axios) {
    const localVarFp = UsersElementsDataApiFp(configuration);
    return {
        /**
         * Assigns a role to the user within the tenant.  The tenant defines the scope of the assignment. In other words, the role is effective only within the tenant.
         * @summary Assign role to user
         * @param {string} projId Either the unique id of the project, or the URL-friendly key of the project (i.e: the \&quot;slug\&quot;).
         * @param {string} envId Either the unique id of the environment, or the URL-friendly key of the environment (i.e: the \&quot;slug\&quot;).
         * @param {string} elementsConfigId Either the unique id of the elements_config, or the URL-friendly key of the elements_config (i.e: the \&quot;slug\&quot;).
         * @param {string} userId Either the unique id of the user, or the URL-friendly key of the user (i.e: the \&quot;slug\&quot;).
         * @param {ElementsUserRoleCreate} elementsUserRoleCreate
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        elementsAssignRoleToUser(projId, envId, elementsConfigId, userId, elementsUserRoleCreate, options) {
            return localVarFp
                .elementsAssignRoleToUser(projId, envId, elementsConfigId, userId, elementsUserRoleCreate, options)
                .then((request) => request(axios, basePath));
        },
        /**
         * Creates a new user inside the Permit.io system, from that point forward you may run permission checks on that user.  If the user is already created: will return 200 instead of 201, and will return the existing user object in the response body.
         * @summary Create user
         * @param {string} projId Either the unique id of the project, or the URL-friendly key of the project (i.e: the \&quot;slug\&quot;).
         * @param {string} envId Either the unique id of the environment, or the URL-friendly key of the environment (i.e: the \&quot;slug\&quot;).
         * @param {string} elementsConfigId Either the unique id of the elements_config, or the URL-friendly key of the elements_config (i.e: the \&quot;slug\&quot;).
         * @param {ElementsUserCreate} elementsUserCreate
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        elementsCreateUser(projId, envId, elementsConfigId, elementsUserCreate, options) {
            return localVarFp
                .elementsCreateUser(projId, envId, elementsConfigId, elementsUserCreate, options)
                .then((request) => request(axios, basePath));
        },
        /**
         *
         * @summary Delete user
         * @param {string} projId Either the unique id of the project, or the URL-friendly key of the project (i.e: the \&quot;slug\&quot;).
         * @param {string} envId Either the unique id of the environment, or the URL-friendly key of the environment (i.e: the \&quot;slug\&quot;).
         * @param {string} elementsConfigId Either the unique id of the elements_config, or the URL-friendly key of the elements_config (i.e: the \&quot;slug\&quot;).
         * @param {string} userId Either the unique id of the user, or the URL-friendly key of the user (i.e: the \&quot;slug\&quot;).
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        elementsDeleteUser(projId, envId, elementsConfigId, userId, options) {
            return localVarFp
                .elementsDeleteUser(projId, envId, elementsConfigId, userId, options)
                .then((request) => request(axios, basePath));
        },
        /**
         * Lists all the users defined within an environment.
         * @summary List roles
         * @param {string} projId Either the unique id of the project, or the URL-friendly key of the project (i.e: the \&quot;slug\&quot;).
         * @param {string} envId Either the unique id of the environment, or the URL-friendly key of the environment (i.e: the \&quot;slug\&quot;).
         * @param {string} elementsConfigId Either the unique id of the elements_config, or the URL-friendly key of the elements_config (i.e: the \&quot;slug\&quot;).
         * @param {string} [search] Text search for the email field
         * @param {number} [page] Page number of the results to fetch, starting at 1.
         * @param {number} [perPage] The number of results per page (max 100).
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        elementsListRoles(projId, envId, elementsConfigId, search, page, perPage, options) {
            return localVarFp
                .elementsListRoles(projId, envId, elementsConfigId, search, page, perPage, options)
                .then((request) => request(axios, basePath));
        },
        /**
         * Lists all the users defined within an environment.
         * @summary List users
         * @param {string} projId Either the unique id of the project, or the URL-friendly key of the project (i.e: the \&quot;slug\&quot;).
         * @param {string} envId Either the unique id of the environment, or the URL-friendly key of the environment (i.e: the \&quot;slug\&quot;).
         * @param {string} elementsConfigId Either the unique id of the elements_config, or the URL-friendly key of the elements_config (i.e: the \&quot;slug\&quot;).
         * @param {string} [search] Text search for the email field
         * @param {number} [page] Page number of the results to fetch, starting at 1.
         * @param {number} [perPage] The number of results per page (max 100).
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        elementsListUsers(projId, envId, elementsConfigId, search, page, perPage, options) {
            return localVarFp
                .elementsListUsers(projId, envId, elementsConfigId, search, page, perPage, options)
                .then((request) => request(axios, basePath));
        },
        /**
         * Unassigns the role from the user within the tenant.  The tenant defines the scope of the assignment. In other words, the role is effective only within the tenant.  If the role is not actually assigned, will return 404.
         * @summary Unassign role from user
         * @param {string} projId Either the unique id of the project, or the URL-friendly key of the project (i.e: the \&quot;slug\&quot;).
         * @param {string} envId Either the unique id of the environment, or the URL-friendly key of the environment (i.e: the \&quot;slug\&quot;).
         * @param {string} elementsConfigId Either the unique id of the elements_config, or the URL-friendly key of the elements_config (i.e: the \&quot;slug\&quot;).
         * @param {string} userId Either the unique id of the user, or the URL-friendly key of the user (i.e: the \&quot;slug\&quot;).
         * @param {ElementsUserRoleRemove} elementsUserRoleRemove
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        elementsUnassignRoleFromUser(projId, envId, elementsConfigId, userId, elementsUserRoleRemove, options) {
            return localVarFp
                .elementsUnassignRoleFromUser(projId, envId, elementsConfigId, userId, elementsUserRoleRemove, options)
                .then((request) => request(axios, basePath));
        },
        /**
         * Updates the embed_config.
         * @summary Set Config Active
         * @param {string} elementsConfigId Either the unique id of the elements_config, or the URL-friendly key of the elements_config (i.e: the \&quot;slug\&quot;).
         * @param {string} projId Either the unique id of the project, or the URL-friendly key of the project (i.e: the \&quot;slug\&quot;).
         * @param {string} envId Either the unique id of the environment, or the URL-friendly key of the environment (i.e: the \&quot;slug\&quot;).
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        setConfigActive(elementsConfigId, projId, envId, options) {
            return localVarFp
                .setConfigActive(elementsConfigId, projId, envId, options)
                .then((request) => request(axios, basePath));
        },
    };
};
/**
 * UsersElementsDataApi - object-oriented interface
 * @export
 * @class UsersElementsDataApi
 * @extends {BaseAPI}
 */
export class UsersElementsDataApi extends BaseAPI {
    /**
     * Assigns a role to the user within the tenant.  The tenant defines the scope of the assignment. In other words, the role is effective only within the tenant.
     * @summary Assign role to user
     * @param {UsersElementsDataApiElementsAssignRoleToUserRequest} requestParameters Request parameters.
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     * @memberof UsersElementsDataApi
     */
    elementsAssignRoleToUser(requestParameters, options) {
        return UsersElementsDataApiFp(this.configuration)
            .elementsAssignRoleToUser(requestParameters.projId, requestParameters.envId, requestParameters.elementsConfigId, requestParameters.userId, requestParameters.elementsUserRoleCreate, options)
            .then((request) => request(this.axios, this.basePath));
    }
    /**
     * Creates a new user inside the Permit.io system, from that point forward you may run permission checks on that user.  If the user is already created: will return 200 instead of 201, and will return the existing user object in the response body.
     * @summary Create user
     * @param {UsersElementsDataApiElementsCreateUserRequest} requestParameters Request parameters.
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     * @memberof UsersElementsDataApi
     */
    elementsCreateUser(requestParameters, options) {
        return UsersElementsDataApiFp(this.configuration)
            .elementsCreateUser(requestParameters.projId, requestParameters.envId, requestParameters.elementsConfigId, requestParameters.elementsUserCreate, options)
            .then((request) => request(this.axios, this.basePath));
    }
    /**
     *
     * @summary Delete user
     * @param {UsersElementsDataApiElementsDeleteUserRequest} requestParameters Request parameters.
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     * @memberof UsersElementsDataApi
     */
    elementsDeleteUser(requestParameters, options) {
        return UsersElementsDataApiFp(this.configuration)
            .elementsDeleteUser(requestParameters.projId, requestParameters.envId, requestParameters.elementsConfigId, requestParameters.userId, options)
            .then((request) => request(this.axios, this.basePath));
    }
    /**
     * Lists all the users defined within an environment.
     * @summary List roles
     * @param {UsersElementsDataApiElementsListRolesRequest} requestParameters Request parameters.
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     * @memberof UsersElementsDataApi
     */
    elementsListRoles(requestParameters, options) {
        return UsersElementsDataApiFp(this.configuration)
            .elementsListRoles(requestParameters.projId, requestParameters.envId, requestParameters.elementsConfigId, requestParameters.search, requestParameters.page, requestParameters.perPage, options)
            .then((request) => request(this.axios, this.basePath));
    }
    /**
     * Lists all the users defined within an environment.
     * @summary List users
     * @param {UsersElementsDataApiElementsListUsersRequest} requestParameters Request parameters.
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     * @memberof UsersElementsDataApi
     */
    elementsListUsers(requestParameters, options) {
        return UsersElementsDataApiFp(this.configuration)
            .elementsListUsers(requestParameters.projId, requestParameters.envId, requestParameters.elementsConfigId, requestParameters.search, requestParameters.page, requestParameters.perPage, options)
            .then((request) => request(this.axios, this.basePath));
    }
    /**
     * Unassigns the role from the user within the tenant.  The tenant defines the scope of the assignment. In other words, the role is effective only within the tenant.  If the role is not actually assigned, will return 404.
     * @summary Unassign role from user
     * @param {UsersElementsDataApiElementsUnassignRoleFromUserRequest} requestParameters Request parameters.
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     * @memberof UsersElementsDataApi
     */
    elementsUnassignRoleFromUser(requestParameters, options) {
        return UsersElementsDataApiFp(this.configuration)
            .elementsUnassignRoleFromUser(requestParameters.projId, requestParameters.envId, requestParameters.elementsConfigId, requestParameters.userId, requestParameters.elementsUserRoleRemove, options)
            .then((request) => request(this.axios, this.basePath));
    }
    /**
     * Updates the embed_config.
     * @summary Set Config Active
     * @param {UsersElementsDataApiSetConfigActiveRequest} requestParameters Request parameters.
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     * @memberof UsersElementsDataApi
     */
    setConfigActive(requestParameters, options) {
        return UsersElementsDataApiFp(this.configuration)
            .setConfigActive(requestParameters.elementsConfigId, requestParameters.projId, requestParameters.envId, options)
            .then((request) => request(this.axios, this.basePath));
    }
}
