// permitio/build/module/instrument/appManager.js
import _ from 'lodash';
import { prettyConsoleLog } from '../logger';
import { mapApp } from './appMapper/appMapper';
export class AppManager {
    constructor(config, resourceReporter, logger) {
        this.config = config;
        this.resourceReporter = resourceReporter;
        this.logger = logger;
        this.resources = {};
    }
    report(name, resources, endpoints) {
        const SEPARATOR_LINE = '*************************************************';
        console.log(`<Permit.io Documenting App - ${name}>`);
        console.log(SEPARATOR_LINE);
        prettyConsoleLog('endpoints', endpoints);
        console.log(SEPARATOR_LINE);
        prettyConsoleLog('resources', resources);
        console.log(SEPARATOR_LINE);
    }
    manage(name, app) {
        // TODO support mixed HTTP/HTTPS server (via name merging resources mutual to http and https )
        // If we have auto-mapping - map the app given to us and save its resources
        if (this.config.autoMapping.enable) {
            const { resources, endpoints } = mapApp(app, this.config.autoMapping.ignoredUrlPrefixes, this.logger);
            if (this.config.autoMapping.reviewMode) {
                this.report(name, resources, endpoints);
            }
            else {
                this.resources[name] = resources;
                _.forEach(resources, (resource) => this.resourceReporter.resource(resource));
            }
        }
    }
}
