// permitio/build/main/elements/client.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.ElementsClient = exports.ElementsApiErrors = void 0;
const axios_1 = __importDefault(require("axios"));
const openapi_1 = require("../openapi");
var ElementsApiErrors;
(function (ElementsApiErrors) {
    ElementsApiErrors["USER_NOT_FOUND"] = "User not found";
    ElementsApiErrors["TENANT_NOT_FOUND"] = "Tenant not found";
    ElementsApiErrors["INVALID_PERMISSION_LEVEL"] = "Invalid user permission level";
    ElementsApiErrors["FORBIDDEN_ACCESS"] = "Forbidden access";
})(ElementsApiErrors = exports.ElementsApiErrors || (exports.ElementsApiErrors = {}));
class ElementsClient {
    constructor(config, logger) {
        this.config = config;
        this.logger = logger;
        const axiosClientConfig = new openapi_1.Configuration({
            basePath: `${this.config.apiUrl}`,
            accessToken: this.config.token,
        });
        this.authApi = new openapi_1.AuthenticationApi(axiosClientConfig);
    }
    async loginAs({ userId, tenantId, }) {
        var _a, _b;
        try {
            const response = await this.authApi.elementsLoginAs({
                userLoginRequestInput: {
                    user_id: userId,
                    tenant_id: tenantId,
                },
            });
            this.logger.debug(`[${response.status}] permit.api.loginAs(${userId})`);
            const res = response.data;
            res['content'] = { url: response.data.redirect_url };
            return res;
        }
        catch (err) {
            if (axios_1.default.isAxiosError(err)) {
                this.logger.error(`[${(_a = err === null || err === void 0 ? void 0 : err.response) === null || _a === void 0 ? void 0 : _a.status}] permit.api.loginAs(${userId}), err: ${JSON.stringify((_b = err === null || err === void 0 ? void 0 : err.response) === null || _b === void 0 ? void 0 : _b.data)}`);
            }
            throw err;
        }
    }
    get elements() {
        return {
            loginAs: this.loginAs.bind(this),
        };
    }
    getMethods() {
        return {
            elements: this.elements,
        };
    }
}
exports.ElementsClient = ElementsClient;
