// permitio/build/module/config.js
import globalAxios from 'axios';
import _ from 'lodash';
import { ApiContext } from './api/context';
/**
 * A factory class for the Permit SDK configuration
 */
export class ConfigFactory {
    /**
     * @returns the default SDK configuration
     */
    static defaults() {
        return {
            token: _.get(process.env, 'PERMIT_API_KEY', ''),
            pdp: _.get(process.env, 'PERMIT_PDP_URL', 'http://localhost:7766'),
            apiUrl: _.get(process.env, 'PERMIT_API_URL', 'https://api.permit.io'),
            log: {
                // log level, default is warn (warnings and errors)
                level: _.get(process.env, 'PERMIT_LOG_LEVEL', 'warn'),
                // Label added to logs
                label: _.get(process.env, 'PERMIT_LOG_LABEL', 'Permit.io'),
                // When logging - dump full data to console as JSON
                json: JSON.parse(_.get(process.env, 'PERMIT_LOG_JSON', 'false')),
            },
            multiTenancy: {
                defaultTenant: 'default',
                useDefaultTenantIfEmpty: true,
            },
            timeout: undefined,
            throwOnError: true,
            apiContext: new ApiContext(),
            axiosInstance: globalAxios.create(),
        };
    }
    /**
     * Builds the Permit SDK configuration from the values provided by the SDK user
     * and from the default SDK configuration when no specific values are set.
     *
     * @param options - a partial configuration
     * @returns the SDK configuration (for unset values returns the default config)
     */
    static build(options) {
        const config = _.merge(_.assign({}, ConfigFactory.defaults()), options);
        return config;
    }
}
