// permitio/build/module/index.js
import { ApiClient } from './api/api-client';
import { ElementsClient } from './api/elements';
import { ConfigFactory } from './config';
import { Enforcer } from './enforcement/enforcer';
import { LoggerFactory } from './logger';
import { AxiosLoggingInterceptor } from './utils/http-logger';
// exported interfaces
export * from './api';
export { PermitConnectionError, PermitError, PermitPDPStatusError } from './enforcement/enforcer';
export { ApiContext, PermitContextError, ApiKeyLevel } from './api/context';
export { PermitApiError } from './api/base';
/**
 * The `Permit` class represents the main entry point for interacting with the Permit.io SDK.
 * The SDK constructor expects an object implementing the {@link IPermitConfig} interface.
 *
 * Example usage:
 *
 * ```typescript
 * import { Permit } from 'permitio';
 *
 * const permit = new Permit({
 *   // this is typically the same API Key you would use for the PDP container
 *   token: "[YOUR_API_KEY]",
 *   // in production, you might need to change this url to fit your deployment
 *   pdp: "http://localhost:7766",
 *   ...
 * });
 *
 * // creates (or updates) a user on that can be assigned roles and permissions
 * const { user } = await permit.api.users.sync({
 *   // the user key must be a unique id of the user
 *   key: 'auth0|elon',
 *   // optional params
 *   email: 'elonmusk@tesla.com',
 *   first_name: 'Elon',
 *   last_name: 'Musk',
 *   // user attributes can be used in attribute-based access-control policies
 *   attributes: {
 *     age: 50,
 *     favoriteColor: 'red',
 *   },
 * });
 *
 * // 'document' is the protected resource we are enforcing access to
 * const resource = 'document';
 * // the action the user is trying to do on the resource
 * const action = 'read';
 *
 * const permitted = await permit.check(user, action, resource);
 * if (permitted) {
 *     console.log('User is authorized to read a document.');
 * } else {
 *     console.log('User is not authorized to read a document.');
 * }
 * ```
 */
export class Permit {
    logger;
    enforcer;
    /**
     * Access the SDK configuration using this property.
     * Once the SDK is initialized, the configuration is read-only.
     *
     * Usage example:
     *
     * ```typescript
     * const permit = new Permit(config);
     * const pdpUrl = permit.config.pdp;
     * ```
     */
    config;
    /**
     * Access the Permit REST API using this property.
     *
     * Usage example:
     *
     * ```typescript
     * const permit = new Permit(config);
     * permit.api.roles.create(...);
     * ```
     */
    api;
    /**
     * Access the Permit Elements API using this property.
     *
     * Usage example:
     *
     * ```typescript
     * const permit = new Permit(config);
     * permit.elements.loginAs(user, tenant);
     * ```
     */
    elements;
    /**
     * Constructs a new instance of the {@link Permit} class with the specified configuration.
     *
     * @param config - The configuration for the Permit SDK.
     */
    constructor(config) {
        this.config = ConfigFactory.build(config);
        this.logger = LoggerFactory.createLogger(this.config);
        AxiosLoggingInterceptor.setupInterceptor(this.config.axiosInstance, this.logger);
        this.api = new ApiClient(this.config, this.logger);
        this.enforcer = new Enforcer(this.config, this.logger);
        this.elements = new ElementsClient(this.config, this.logger);
        this.logger.debug(`Permit.io SDK initialized with config:\n${JSON.stringify(this.config, undefined, 2)}`);
    }
    /**
     * Checks if a `user` is authorized to perform an `action` on a `resource` within the specified context.
     *
     * @param user     - The user object representing the user.
     * @param action   - The action to be performed on the resource.
     * @param resource - The resource object representing the resource.
     * @param context  - The context object representing the context in which the action is performed.
     * @returns `true` if the user is authorized, `false` otherwise.
     * @throws {@link PermitConnectionError} if an error occurs while sending the authorization request to the PDP.
     * @throws {@link PermitPDPStatusError} if received a response with unexpected status code from the PDP.
     */
    async check(user, action, resource, context, config) {
        return await this.enforcer.check(user, action, resource, context, config);
    }
    /**
     * Checks multiple requests within the specified context.
     *
     * @param checks   - The check requests.
     * @param context  - The context object representing the context in which the action is performed.
     * @returns array containing `true` if the user is authorized, `false` otherwise for each check request.
     * @throws {@link PermitConnectionError} if an error occurs while sending the authorization request to the PDP.
     * @throws {@link PermitPDPStatusError} if received a response with unexpected status code from the PDP.
     */
    async bulkCheck(checks, context, config) {
        return await this.enforcer.bulkCheck(checks, context, config);
    }
    /**
     * Get all permissions for the specified user.
     *
     * @param user     - The user object representing the user.
     * @param tenants  - The list of tenants to filter the permissions on ( given by roles ).
     * @param resources - The list of resources to filter the permissions on ( given by resource roles ).
     * @param resource_types - The list of resource types to filter the permissions on ( given by resource roles ).
     * @returns object with key as the resource identifier and value as the resource details and permissions.
     * @throws {@link PermitConnectionError} if an error occurs while sending the authorization request to the PDP.
     * @throws {@link PermitPDPStatusError} if received a response with unexpected status code from the PDP.
     */
    async getUserPermissions(user, tenants, resources, resource_types, config) {
        return await this.enforcer.getUserPermissions(user, tenants, resources, resource_types, config);
    }
}
