// permitio/build/main/utils/context.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ContextStore = void 0;
/**
 * Store for contexted passed as part of check() queries
 */
class ContextStore {
    constructor() {
        this.baseContext = {}; // cross-query context (global context)
        this.transforms = [];
    }
    /**
     * add context to the base context
     */
    add(context) {
        this.baseContext = Object.assign(this.baseContext, context);
    }
    registerTransform(transform) {
        this.transforms.push(transform);
    }
    /**
     * merges the global context (this.context) with the context
     * provided for this specific query (context). the specific
     * context overrides the base (global) context.
     */
    getDerivedContext(context) {
        return Object.assign({}, this.baseContext, context);
    }
    transform(initialContext) {
        let context = Object.assign({}, initialContext);
        for (const transform of this.transforms) {
            context = transform(context);
        }
        return context;
    }
}
exports.ContextStore = ContextStore;
