// permitio/build/main/utils/regex.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.matchAll = exports.escapeRegex = void 0;
function escapeRegex(s) {
    return s.replace(/[-\/\\^$*+?.()|[\]{}]/g, '\\$&');
}
exports.escapeRegex = escapeRegex;
function makeRegexGlobal(re) {
    if (re.global) {
        return re;
    }
    let flags = 'g';
    flags += re.ignoreCase ? 'i' : '';
    flags += re.multiline ? 'm' : '';
    return new RegExp(re.source, flags);
}
/**
 * RegExp: Get all matches and capturing groups
 * @param {RegExp} re
 * @param {String} str
 * @returns {Array<Array<String>>}
 */
function matchAll(re, str) {
    const matches = [];
    const regex = makeRegexGlobal(re);
    let groups;
    while ((groups = regex.exec(str)) !== null) {
        const found = Array.from(groups);
        const length = found[0].length;
        matches.push({
            start: groups.index,
            end: groups.index + length - 1,
            length: length,
            groups: found,
        });
    }
    return matches;
}
exports.matchAll = matchAll;
