// permitio/build/main/instrument/appManager.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.AppManager = void 0;
const lodash_1 = __importDefault(require("lodash"));
const logger_1 = require("../logger");
const appMapper_1 = require("./appMapper/appMapper");
class AppManager {
    constructor(config, resourceReporter, logger) {
        this.config = config;
        this.resourceReporter = resourceReporter;
        this.logger = logger;
        this.resources = {};
    }
    report(name, resources, endpoints) {
        const SEPARATOR_LINE = '*************************************************';
        console.log(`<Permit.io Documenting App - ${name}>`);
        console.log(SEPARATOR_LINE);
        logger_1.prettyConsoleLog('endpoints', endpoints);
        console.log(SEPARATOR_LINE);
        logger_1.prettyConsoleLog('resources', resources);
        console.log(SEPARATOR_LINE);
    }
    manage(name, app) {
        // TODO support mixed HTTP/HTTPS server (via name merging resources mutual to http and https )
        // If we have auto-mapping - map the app given to us and save its resources
        if (this.config.autoMapping.enable) {
            const { resources, endpoints } = appMapper_1.mapApp(app, this.config.autoMapping.ignoredUrlPrefixes, this.logger);
            if (this.config.autoMapping.reviewMode) {
                this.report(name, resources, endpoints);
            }
            else {
                this.resources[name] = resources;
                lodash_1.default.forEach(resources, (resource) => this.resourceReporter.resource(resource));
            }
        }
    }
}
exports.AppManager = AppManager;
