// permitio/build/module/resources/reporter.js
import axios from 'axios';
import { ActionDefinition, ResourceDefinition } from './registry';
export class ResourceStub {
    reporter;
    resourceName;
    constructor(reporter, resourceName) {
        this.reporter = reporter;
        this.resourceName = resourceName;
    }
    action(config) {
        const action = new ActionDefinition(config.name, config.title, config.description, config.path, config.attributes || {});
        this.reporter.addActionToResource(this.resourceName, action);
    }
}
/**
 * the ResourceReporter is used to sync resources and actions
 * (the authorization enforcement points) defined by the app to the
 * permit.io control plane in the cloud, so that policy and permissions
 * may be defined by the control plane. The only data permit.io needs to
 * know about the app is the available enforcement point (i.e: resources
 * and actions).
 */
export class ResourceReporter {
    config;
    registry;
    logger;
    initialized = false;
    client = axios.create();
    constructor(config, registry, logger) {
        this.config = config;
        this.registry = registry;
        this.logger = logger;
        this.client = axios.create({
            baseURL: `${this.config.pdpUrl}/`,
            headers: {
                Authorization: `Bearer ${this.config.token}`,
                'Content-Type': 'application/json',
            },
        });
        this.initialized = true; // TODO: remove this
        this.syncResourcesToControlPlane();
    }
    get token() {
        return this.config.token;
    }
    // resources and actions
    addResource(resource) {
        this.registry.addResource(resource);
        this.maybeSyncResource(resource);
        return new ResourceStub(this, resource.name);
    }
    addActionToResource(resourceName, actionDef) {
        const action = this.registry.addActionToResource(resourceName, actionDef);
        if (action) {
            this.maybeSyncAction(action);
        }
    }
    maybeSyncResource(resource) {
        if (this.initialized && !this.registry.isSynced(resource)) {
            this.logger.info(`syncing resource: ${resource.repr()}`);
            this.client
                .put(`cloud/resources/${resource.name}`, resource.dict())
                .then((response) => {
                this.registry.markAsSynced(resource, response.data.id);
            })
                .catch((error) => {
                this.logger.error(`tried to sync resource ${resource.name}, got error: ${error}`);
            });
        }
    }
    maybeSyncAction(action) {
        if (!action.resourceId) {
            return;
        }
        const resourceId = action.resourceId;
        if (this.initialized && !this.registry.isSynced(action)) {
            this.logger.info(`syncing action: ${action.repr()}`);
            this.client
                .put(`cloud/resources/${resourceId}/actions`, action.dict())
                .then((response) => {
                this.registry.markAsSynced(action, response.data.id);
            })
                .catch((error) => {
                this.logger.error(`tried to sync action ${action.name}, got error: ${error}`);
            });
        }
    }
    syncResourcesToControlPlane() {
        // will also sync actions
        for (const resource of this.registry.resourceList) {
            this.maybeSyncResource(resource);
        }
    }
    resource(config) {
        const resource = new ResourceDefinition(config.name, config.type, config.path, config.description, config.actions || [], config.attributes || {});
        return this.addResource(resource);
    }
    action(config) {
        return new ActionDefinition(config.name, config.title, config.description, config.path, config.attributes || {});
    }
    // TODO: currently we use the old api (PUT single resource)
    // due to mismatches with the resource registry
    syncResources(config) {
        const stubs = [];
        for (const resource of config.resources) {
            stubs.push(this.addResource(new ResourceDefinition(resource.type, 'rest', `/resources/${resource.type}`, resource.description, Object.keys(resource.actions).map((actionName) => {
                const action = resource.actions[actionName];
                return new ActionDefinition(actionName, action.title ?? actionName, action.description, action.path, action.attributes || {});
            }), resource.attributes || {})));
        }
        return stubs;
    }
    getMethods() {
        return {
            resource: this.resource.bind(this),
            action: this.action.bind(this),
            syncResources: this.syncResources.bind(this),
        };
    }
}
